---
title: "NEON project"
output: html_document
date: "2024-01-29"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

****jaccard's distance for NMDS instead of bray****

## ALPHA DIVERSITY ##
#Q1 What are the most diverse eukaryotic groups in soil (and maybe which deserve more attention)?
  #Major Eukaryotic groups by alpha diversity across all sites with error bars (Oliverio 2020, Fig 1 A) - Done
#Q2 what is the relative alpha diversity of soil eukaryotes in North America?
  #Relative taxonomic diversity of eukaryotic phyla by site (e.g., Bates 2013 Fig 1 or Thompson 2020 Fig 3 or Mahe 2017 Fig 3)
#Q3 How under explored is soil eukaryote diversity? Maybe do this by group (protists, metazoa, fungi, etc.)
  #Measure of the novelty of sequences based off of their similarity to reference sequences (Mahe 2017, Fig 1)


## BETA DIVERSITY ##
#Q4 How do communities (site compositions) differ by biome assignment?
  #PCoA biome (Bates 2013 Fig 3)
#Q5 How homogenous are communities over large spatial scales?
  #Community Dissimilarity - Phyla (Bates 2013 Fig 2)
#Q6 Is everything everywhere? Prediction: dispersal is constrained by body size patterns (papers?) This one would have to be genus or at least family.
  #PCoA is community dissimilarity driven by body size?  Or another test?


## OTHERS ##
#Q How correlated are N. American soil eukaryotes by group?
  #Heatmap of different groups co-occurrence (?) (Mahe 2017 Fig 4)
#Q Are parasites, consumers, or phototrophs the most diverse eukaryotes in soil?
  #Major Eukaryotic groups by function? (Oliverio 2020, Fig 1 C)
#Q How well does research efforts parallel the diversity hierarchy of soil eukaryotic groups?
  #That paper byron shared with the skew...?


#3. Basic Stats
  #3A. Count sampling events per site
  #3B. Count Unique OTUs and Reads per Group and Site
#4. Sampling Depth Estimation
  #4A. Rarefaction Curves
  #4B. Species Accumulation Curves

Fix OTU table and taxonomy for mctoolsr into one dataframe.
```{r}
#library(janitor)
#setwd("/Users/leenavilonen/Documents/NEON")
#taxonomy<-read.csv('eukDiv_tax_simplified.csv',header=TRUE)
#library(dplyr)
#library(tidyr)
#taxonomy<-unite(taxonomy, col='taxonomy', c('Domain', 'SuperGroup', 'Kingdom', 'Phylum', 'Class', 'Order', 'Genus'), sep=';')
#otu_table<-read.csv('eukDiv_otu.csv',header=TRUE)
#otu_table_withtax <- merge(otu_table,taxonomy,by="X.OTU.ID")
#write.csv(otu_table_withtax,'otu_table_withtax.csv')
```
Make sure to change the csv file to a .txt file. I think it's easier to just do in excel, but you can also save out the file as a .txt file. I also like to check the formatting there too to make sure it all worked out.


On your computer turn both the map and ASV table with taxonomy into text files. Make sure to rename as "#OTU ID". Re input these and then we are good to go.
```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
```

Filter out chro/mito, which are problems with binning from the pipeline
```{r}
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Fungi:mito')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')

```

```{r}
fungi_specnum<-filter_taxa_from_input(input,taxa_to_keep = 'Fungi')
metazoa_specnum<-filter_taxa_from_input(input,taxa_to_keep = 'Metazoa')
Streptophyta_specnum<-filter_taxa_from_input(input,taxa_to_keep = 'Streptophyta')
Cercozoa_specnum<-filter_taxa_from_input(input,taxa_to_keep = 'Cercozoa')
library(vegan)
library(dplyr)
nrow(fungi_specnum$data_loaded)
nrow(metazoa_specnum$data_loaded)
nrow(Streptophyta_specnum$data_loaded)
nrow(Cercozoa_specnum$data_loaded)
```
```{r}
Ascomycota_specnum<-filter_taxa_from_input(fungi_specnum,taxa_to_keep = 'Ascomycota')
Basidiomycota_specnum<-filter_taxa_from_input(fungi_specnum,taxa_to_keep = 'Basidiomycota')
Mucoromycota_specnum<-filter_taxa_from_input(fungi_specnum,taxa_to_keep = 'Mucoromycota')
nrow(Ascomycota_specnum$data_loaded)
nrow(Basidiomycota_specnum$data_loaded)
nrow(Mucoromycota_specnum$data_loaded)
```

```{r}
Nematoda_specnum<-filter_taxa_from_input(metazoa_specnum,taxa_to_keep = 'Nematoda')
Arthropoda_specnum<-filter_taxa_from_input(metazoa_specnum,taxa_to_keep = 'Arthropoda')
Annelida_specnum<-filter_taxa_from_input(metazoa_specnum,taxa_to_keep = 'Annelida')
nrow(Nematoda_specnum$data_loaded)
nrow(Arthropoda_specnum$data_loaded)
nrow(Annelida_specnum$data_loaded)
```

```{r}
library(labdsv)
otuTAB<-input$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
input$data_loaded<-t(hell)
```

```{r}
tax_sum_phyla = summarize_taxonomy(input, level = 3, report_higher_tax = FALSE)
tax_sum_phyla_t<-t(tax_sum_phyla)
library(reshape2)
tax_sum_phyla_t_melt<-melt(tax_sum_phyla_t)
boxplot(tax_sum_phyla_t)
library(ggplot2)
```

```{r}
full<-ggplot(data=tax_sum_phyla_t_melt,aes(x=Var2,y=value,color=factor(Var2))) +
  geom_boxplot()+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Relative Abundance")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
full
```

```{r}
ggsave(filename = "full_species_richness_NEON.pdf",
      plot = full,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
setwd("/Users/leenavilonen/Documents/NEON")
library(mctoolsr)
library(vegan)
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Fungi:mito')
inputfinal <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
```

```{r}
inputTelonemia <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Telonemia')
Telonemia<-specnumber(t(inputTelonemia$data_loaded))
inputStreptophyta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Streptophyta')
Streptophyta<-specnumber(t(inputStreptophyta$data_loaded))
inputSagenista <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Sagenista')
Sagenista<-specnumber(t(inputSagenista$data_loaded))
inputRhodophyta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Rhodophyta')
Rhodophyta<-specnumber(t(inputRhodophyta$data_loaded))
inputRadiolaria <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Radiolaria')
Radiolaria<-specnumber(t(inputRadiolaria$data_loaded))
inputPseudofungi <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Pseudofungi')
Pseudofungi<-specnumber(t(inputPseudofungi$data_loaded))
inputPrasinodermophyta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Prasinodermophyta')
Prasinodermophyta<-specnumber(t(inputPrasinodermophyta$data_loaded))
inputPicozoa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Picozoa')
Picozoa<-specnumber(t(inputPicozoa$data_loaded))
inputPerkinsea <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Perkinsea')
Perkinsea<-specnumber(t(inputPerkinsea$data_loaded))
inputOpalozoa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Opalozoa')
Opalozoa<-specnumber(t(inputOpalozoa$data_loaded))
inputOchrophyta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Ochrophyta')
Ochrophyta<-specnumber(t(inputOchrophyta$data_loaded))
inputMetazoa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Metazoa')
Metazoa<-specnumber(t(inputMetazoa$data_loaded))
inputMetamonada <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Metamonada')
Metamonada<-specnumber(t(inputMetamonada$data_loaded))
inputMesomycetozoa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Mesomycetozoa')
Mesomycetozoa<-specnumber(t(inputMesomycetozoa$data_loaded))
inputLobosa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Lobosa')
Lobosa<-specnumber(t(inputLobosa$data_loaded))
inputKatablepharidophyta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Katablepharidophyta')
Katablepharidophyta<-specnumber(t(inputKatablepharidophyta$data_loaded))
inputHilomonadea <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Hilomonadea')
Hilomonadea<-specnumber(t(inputHilomonadea$data_loaded))
inputHemimastigophora <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Hemimastigophora')
Hemimastigophora<-specnumber(t(inputHemimastigophora$data_loaded))
inputHaptophyta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Haptophyta')
Haptophyta<-specnumber(t(inputHaptophyta$data_loaded))
inputGlaucophyta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Glaucophyta')
Glaucophyta<-specnumber(t(inputGlaucophyta$data_loaded))
inputFungi <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Fungi')
Fungi<-specnumber(t(inputFungi$data_loaded))
inputForaminifera <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Foraminifera')
Foraminifera<-specnumber(t(inputForaminifera$data_loaded))
inputDiscoba <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Discoba')
Discoba<-specnumber(t(inputDiscoba$data_loaded))
inputDinoflagellata <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Dinoflagellata')
Dinoflagellata<-specnumber(t(inputDinoflagellata$data_loaded))
inputCryptophyta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Cryptophyta')
Cryptophyta<-specnumber(t(inputCryptophyta$data_loaded))
inputConosa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Conosa')
Conosa<-specnumber(t(inputConosa$data_loaded))
inputCiliophora <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Ciliophora')
Ciliophora<-specnumber(t(inputCiliophora$data_loaded))
inputChrompodellids <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Chrompodellids')
Chrompodellids<-specnumber(t(inputChrompodellids$data_loaded))
inputChoanoflagellida <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Choanoflagellida')
Choanoflagellida<-specnumber(t(inputChoanoflagellida$data_loaded))
inputChlorophyta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Chlorophyta')
Chlorophyta<-specnumber(t(inputChlorophyta$data_loaded))
inputCercozoa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Cercozoa')
Cercozoa<-specnumber(t(inputCercozoa$data_loaded))
inputCentroheliozoa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Centroheliozoa')
Centroheliozoa<-specnumber(t(inputCentroheliozoa$data_loaded))
inputBreviatea <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Breviatea')
Breviatea<-specnumber(t(inputBreviatea$data_loaded))
inputApusomonadidae <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Apusomonadidae')
Apusomonadidae<-specnumber(t(inputApusomonadidae$data_loaded))
inputApicomplexa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Apicomplexa')
Apicomplexa<-specnumber(t(inputApicomplexa$data_loaded))
```

```{r}
Telonemiadf<-as.data.frame(Telonemia)
colnames(Telonemiadf)[1] ="rich"
site<-rep("Telonemia",times=nrow(Telonemiadf))
Telonemiadffinal<-cbind(Telonemiadf,site)

Streptophytadf<-as.data.frame(Streptophyta)
colnames(Streptophytadf)[1] ="rich"
site<-rep("Streptophyta",times=nrow(Streptophytadf))
Streptophytadffinal<-cbind(Streptophytadf,site)

Sagenistadf<-as.data.frame(Sagenista)
colnames(Sagenistadf)[1] ="rich"
site<-rep("Sagenista",times=nrow(Sagenistadf))
Sagenistadffinal<-cbind(Sagenistadf,site)

Rhodophytadf<-as.data.frame(Rhodophyta)
colnames(Rhodophytadf)[1] ="rich"
site<-rep("Rhodophyta",times=nrow(Rhodophytadf))
Rhodophytadffinal<-cbind(Rhodophytadf,site)

Radiolariadf<-as.data.frame(Radiolaria)
colnames(Radiolariadf)[1] ="rich"
site<-rep("Radiolaria",times=nrow(Radiolariadf))
Radiolariadffinal<-cbind(Radiolariadf,site)

Pseudofungidf<-as.data.frame(Pseudofungi)
colnames(Pseudofungidf)[1] ="rich"
site<-rep("Pseudofungi",times=nrow(Pseudofungidf))
Pseudofungidffinal<-cbind(Pseudofungidf,site)

Prasinodermophytadf<-as.data.frame(Prasinodermophyta)
colnames(Prasinodermophytadf)[1] ="rich"
site<-rep("Prasinodermophyta",times=nrow(Prasinodermophytadf))
Prasinodermophytadffinal<-cbind(Prasinodermophytadf,site)

Picozoadf<-as.data.frame(Picozoa)
colnames(Picozoadf)[1] ="rich"
site<-rep("Picozoa",times=nrow(Picozoadf))
Picozoadffinal<-cbind(Picozoadf,site)

Perkinseadf<-as.data.frame(Perkinsea)
colnames(Perkinseadf)[1] ="rich"
site<-rep("Perkinsea",times=nrow(Perkinseadf))
Perkinseadffinal<-cbind(Perkinseadf,site)

Opalozoadf<-as.data.frame(Opalozoa)
colnames(Opalozoadf)[1] ="rich"
site<-rep("Opalozoa",times=nrow(Opalozoadf))
Opalozoadffinal<-cbind(Opalozoadf,site)

Ochrophytadf<-as.data.frame(Ochrophyta)
colnames(Ochrophytadf)[1] ="rich"
site<-rep("Ochrophyta",times=nrow(Ochrophytadf))
Ochrophytadffinal<-cbind(Ochrophytadf,site)

Metazoadf<-as.data.frame(Metazoa)
colnames(Metazoadf)[1] ="rich"
site<-rep("Metazoa",times=nrow(Metazoadf))
Metazoadffinal<-cbind(Metazoadf,site)

Metamonadadf<-as.data.frame(Metamonada)
colnames(Metamonadadf)[1] ="rich"
site<-rep("Metamonada",times=nrow(Metamonadadf))
Metamonadadffinal<-cbind(Metamonadadf,site)

Mesomycetozoadf<-as.data.frame(Mesomycetozoa)
colnames(Mesomycetozoadf)[1] ="rich"
site<-rep("Mesomycetozoa",times=nrow(Mesomycetozoadf))
Mesomycetozoadffinal<-cbind(Mesomycetozoadf,site)

Lobosadf<-as.data.frame(Lobosa)
colnames(Lobosadf)[1] ="rich"
site<-rep("Lobosa",times=nrow(Lobosadf))
Lobosadffinal<-cbind(Lobosadf,site)

Katablepharidophytadf<-as.data.frame(Katablepharidophyta)
colnames(Katablepharidophytadf)[1] ="rich"
site<-rep("Katablepharidophyta",times=nrow(Katablepharidophytadf))
Katablepharidophytadffinal<-cbind(Katablepharidophytadf,site)

Hilomonadeadf<-as.data.frame(Hilomonadea)
colnames(Hilomonadeadf)[1] ="rich"
site<-rep("Hilomonadea",times=nrow(Hilomonadeadf))
Hilomonadeadffinal<-cbind(Hilomonadeadf,site)

Hemimastigophoradf<-as.data.frame(Hemimastigophora)
colnames(Hemimastigophoradf)[1] ="rich"
site<-rep("Hemimastigophora",times=nrow(Hemimastigophoradf))
Hemimastigophoradffinal<-cbind(Hemimastigophoradf,site)

Haptophytadf<-as.data.frame(Haptophyta)
colnames(Haptophytadf)[1] ="rich"
site<-rep("Haptophyta",times=nrow(Haptophytadf))
Haptophytadffinal<-cbind(Haptophytadf,site)

Glaucophytadf<-as.data.frame(Glaucophyta)
colnames(Glaucophytadf)[1] ="rich"
site<-rep("Glaucophyta",times=nrow(Glaucophytadf))
Glaucophytadffinal<-cbind(Glaucophytadf,site)

Fungidf<-as.data.frame(Fungi)
colnames(Fungidf)[1] ="rich"
site<-rep("Fungi",times=nrow(Fungidf))
Fungidffinal<-cbind(Fungidf,site)

Foraminiferadf<-as.data.frame(Foraminifera)
colnames(Foraminiferadf)[1] ="rich"
site<-rep("Foraminifera",times=nrow(Foraminiferadf))
Foraminiferadffinal<-cbind(Foraminiferadf,site)

Discobadf<-as.data.frame(Discoba)
colnames(Discobadf)[1] ="rich"
site<-rep("Discoba",times=nrow(Discobadf))
Discobadffinal<-cbind(Discobadf,site)

Dinoflagellatadf<-as.data.frame(Dinoflagellata)
colnames(Dinoflagellatadf)[1] ="rich"
site<-rep("Dinoflagellata",times=nrow(Dinoflagellatadf))
Dinoflagellatadffinal<-cbind(Dinoflagellatadf,site)

Cryptophytadf<-as.data.frame(Cryptophyta)
colnames(Cryptophytadf)[1] ="rich"
site<-rep("Cryptophyta",times=nrow(Cryptophytadf))
Cryptophytadffinal<-cbind(Cryptophytadf,site)

Conosadf<-as.data.frame(Conosa)
colnames(Conosadf)[1] ="rich"
site<-rep("Conosa",times=nrow(Conosadf))
Conosadffinal<-cbind(Conosadf,site)

Ciliophoradf<-as.data.frame(Ciliophora)
colnames(Ciliophoradf)[1] ="rich"
site<-rep("Ciliophora",times=nrow(Ciliophoradf))
Ciliophoradffinal<-cbind(Ciliophoradf,site)

Chrompodellidsdf<-as.data.frame(Chrompodellids)
colnames(Chrompodellidsdf)[1] ="rich"
site<-rep("Chrompodellids",times=nrow(Chrompodellidsdf))
Chrompodellidsdffinal<-cbind(Chrompodellidsdf,site)

Choanoflagellidadf<-as.data.frame(Choanoflagellida)
colnames(Choanoflagellidadf)[1] ="rich"
site<-rep("Choanoflagellida",times=nrow(Choanoflagellidadf))
Choanoflagellidadffinal<-cbind(Choanoflagellidadf,site)

Chlorophytadf<-as.data.frame(Chlorophyta)
colnames(Chlorophytadf)[1] ="rich"
site<-rep("Chlorophyta",times=nrow(Chlorophytadf))
Chlorophytadffinal<-cbind(Chlorophytadf,site)

Cercozoadf<-as.data.frame(Cercozoa)
colnames(Cercozoadf)[1] ="rich"
site<-rep("Cercozoa",times=nrow(Cercozoadf))
Cercozoadffinal<-cbind(Cercozoadf,site)

Centroheliozoadf<-as.data.frame(Centroheliozoa)
colnames(Centroheliozoadf)[1] ="rich"
site<-rep("Centroheliozoa",times=nrow(Centroheliozoadf))
Centroheliozoadffinal<-cbind(Centroheliozoadf,site)

Breviateadf<-as.data.frame(Breviatea)
colnames(Breviateadf)[1] ="rich"
site<-rep("Breviatea",times=nrow(Breviateadf))
Breviateadffinal<-cbind(Breviateadf,site)

Apusomonadidaedf<-as.data.frame(Apusomonadidae)
colnames(Apusomonadidaedf)[1] ="rich"
site<-rep("Apusomonadidae",times=nrow(Apusomonadidaedf))
Apusomonadidaedffinal<-cbind(Apusomonadidaedf,site)

Apicomplexadf<-as.data.frame(Apicomplexa)
colnames(Apicomplexadf)[1] ="rich"
site<-rep("Apicomplexa",times=nrow(Apicomplexadf))
Apicomplexadffinal<-cbind(Apicomplexadf,site)
```

```{r}
#now rbind them together
finalbindall<-do.call("rbind",list(Apicomplexadffinal,Apusomonadidaedffinal,Breviateadffinal,Centroheliozoadffinal,Cercozoadffinal,Chlorophytadffinal,Choanoflagellidadffinal,Chrompodellidsdffinal,Ciliophoradffinal,Conosadffinal,Cryptophytadffinal,Dinoflagellatadffinal,Discobadffinal,Foraminiferadffinal,Fungidffinal,Glaucophytadffinal,Haptophytadffinal,Hemimastigophoradffinal,Hilomonadeadffinal,Katablepharidophytadffinal,Lobosadffinal,Mesomycetozoadffinal,Metamonadadffinal,Metazoadffinal,Ochrophytadffinal,Opalozoadffinal,Perkinseadffinal,Picozoadffinal,Prasinodermophytadffinal,Pseudofungidffinal,Radiolariadffinal,Rhodophytadffinal,Sagenistadffinal,Streptophytadffinal,Telonemiadffinal))
```


```{r}
library(ggplot2)
full_2<-ggplot(data=finalbindall,aes(x=site,y=log(rich),color=factor(site))) +
  geom_boxplot()+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("log(Unique ASVs)")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
full_2
```

```{r}
ggsave(filename = "full_species_richness_NEON_unique_ASV.pdf",
      plot = full_2,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
library(patchwork)
combined_full<-full+full_2
combined_full
```

```{r}
ggsave(filename = "full_species_richness_NEON_unique_ASV_combined.pdf",
      plot = combined_full,
      bg = "transparent",
      width = 10, height = 6, units = "in",
      dpi = 600)
```

Metazoa only

```{r}
input <- filter_taxa_from_input(input, taxa_to_keep = 'Metazoa')
```

```{r}
tax_sum_phyla = summarize_taxonomy(input, level = 4, report_higher_tax = FALSE)
tax_sum_phyla_t<-t(tax_sum_phyla)
library(reshape2)
tax_sum_phyla_t_melt<-melt(tax_sum_phyla_t)
boxplot(tax_sum_phyla_t)
library(ggplot2)
```

```{r}
full<-ggplot(data=tax_sum_phyla_t_melt,aes(x=Var2,y=value,color=factor(Var2))) +
  geom_boxplot()+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Relative Abundance")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))

ggsave(filename = "full_species_richness_NEON_metazoa_only.pdf",
      plot = full,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```



```{r}
setwd("/Users/leenavilonen/Documents/NEON")
library(mctoolsr)
library(vegan)
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Fungi:mito')
inputfinal <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
inputfinal <- filter_taxa_from_input(input, taxa_to_keep = 'Metazoa')
```

```{r}
inputUrochordata <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Urochordata')
Urochordata<-specnumber(t(inputUrochordata$data_loaded))

inputTardigrada <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Tardigrada')
Tardigrada<-specnumber(t(inputTardigrada$data_loaded))

inputRotifera <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Rotifera')
Rotifera<-specnumber(t(inputRotifera$data_loaded))

inputPorifera <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Porifera')
Porifera<-specnumber(t(inputPorifera$data_loaded))

inputPlatyhelminthes <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Platyhelminthes')
Platyhelminthes<-specnumber(t(inputPlatyhelminthes$data_loaded))

inputNemertea <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Nemertea')
Nemertea<-specnumber(t(inputNemertea$data_loaded))

inputNematomorpha <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Nematomorpha')
Nematomorpha<-specnumber(t(inputNematomorpha$data_loaded))

inputNematoda <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Nematoda')
Nematoda<-specnumber(t(inputNematoda$data_loaded))

inputMollusca <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Mollusca')
Mollusca<-specnumber(t(inputMollusca$data_loaded))

inputKinorhyncha <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Kinorhyncha')
Kinorhyncha<-specnumber(t(inputKinorhyncha$data_loaded))

inputHemichordata <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Hemichordata')
Hemichordata<-specnumber(t(inputHemichordata$data_loaded))

inputGnathostomulida <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Gnathostomulida')
Gnathostomulida<-specnumber(t(inputGnathostomulida$data_loaded))

inputGastrotricha <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Gastrotricha')
Gastrotricha<-specnumber(t(inputGastrotricha$data_loaded))

inputEntoprocta <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Entoprocta')
Entoprocta<-specnumber(t(inputEntoprocta$data_loaded))

inputEchinodermata <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Echinodermata')
Echinodermata<-specnumber(t(inputEchinodermata$data_loaded))

inputCtenophora <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Ctenophora')
Ctenophora<-specnumber(t(inputCtenophora$data_loaded))

inputCraniata <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Craniata')
Craniata<-specnumber(t(inputCraniata$data_loaded))

inputCnidaria <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Cnidaria')
Cnidaria<-specnumber(t(inputCnidaria$data_loaded))

inputBryozoa <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Bryozoa')
Bryozoa<-specnumber(t(inputBryozoa$data_loaded))

inputBrachiopoda <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Brachiopoda')
Brachiopoda<-specnumber(t(inputBrachiopoda$data_loaded))

inputArthropoda <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Arthropoda')
Arthropoda<-specnumber(t(inputArthropoda$data_loaded))

inputAnnelida <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Annelida')
Annelida<-specnumber(t(inputAnnelida$data_loaded))
```

```{r}
Urochordatadf<-as.data.frame(Urochordata)
colnames(Urochordatadf)[1] ="rich"
site<-rep("Urochordata",times=nrow(Urochordatadf))
Urochordatadffinal<-cbind(Urochordatadf,site)

Tardigradadf<-as.data.frame(Tardigrada)
colnames(Tardigradadf)[1] ="rich"
site<-rep("Tardigrada",times=nrow(Tardigradadf))
Tardigradadffinal<-cbind(Tardigradadf,site)

Rotiferadf<-as.data.frame(Rotifera)
colnames(Rotiferadf)[1] ="rich"
site<-rep("Rotifera",times=nrow(Rotiferadf))
Rotiferadffinal<-cbind(Rotiferadf,site)

Poriferadf<-as.data.frame(Porifera)
colnames(Poriferadf)[1] ="rich"
site<-rep("Porifera",times=nrow(Poriferadf))
Poriferadffinal<-cbind(Poriferadf,site)

Platyhelminthesdf<-as.data.frame(Platyhelminthes)
colnames(Platyhelminthesdf)[1] ="rich"
site<-rep("Platyhelminthes",times=nrow(Platyhelminthesdf))
Platyhelminthesdffinal<-cbind(Platyhelminthesdf,site)

Nemerteadf<-as.data.frame(Nemertea)
colnames(Nemerteadf)[1] ="rich"
site<-rep("Nemertea",times=nrow(Nemerteadf))
Nemerteadffinal<-cbind(Nemerteadf,site)

Nematomorphadf<-as.data.frame(Nematomorpha)
colnames(Nematomorphadf)[1] ="rich"
site<-rep("Nematomorpha",times=nrow(Nematomorphadf))
Nematomorphadffinal<-cbind(Nematomorphadf,site)

Nematodadf<-as.data.frame(Nematoda)
colnames(Nematodadf)[1] ="rich"
site<-rep("Nematoda",times=nrow(Nematodadf))
Nematodadffinal<-cbind(Nematodadf,site)

Molluscadf<-as.data.frame(Mollusca)
colnames(Molluscadf)[1] ="rich"
site<-rep("Mollusca",times=nrow(Molluscadf))
Molluscadffinal<-cbind(Molluscadf,site)

Kinorhynchadf<-as.data.frame(Kinorhyncha)
colnames(Kinorhynchadf)[1] ="rich"
site<-rep("Kinorhyncha",times=nrow(Kinorhynchadf))
Kinorhynchadffinal<-cbind(Kinorhynchadf,site)

Hemichordatadf<-as.data.frame(Hemichordata)
colnames(Hemichordatadf)[1] ="rich"
site<-rep("Hemichordata",times=nrow(Hemichordatadf))
Hemichordatadffinal<-cbind(Hemichordatadf,site)

Gnathostomulidadf<-as.data.frame(Gnathostomulida)
colnames(Gnathostomulidadf)[1] ="rich"
site<-rep("Gnathostomulida",times=nrow(Gnathostomulidadf))
Gnathostomulidadffinal<-cbind(Gnathostomulidadf,site)

Gastrotrichadf<-as.data.frame(Gastrotricha)
colnames(Gastrotrichadf)[1] ="rich"
site<-rep("Gastrotricha",times=nrow(Gastrotrichadf))
Gastrotrichadffinal<-cbind(Gastrotrichadf,site)

Entoproctadf<-as.data.frame(Entoprocta)
colnames(Entoproctadf)[1] ="rich"
site<-rep("Entoprocta",times=nrow(Entoproctadf))
Entoproctadffinal<-cbind(Entoproctadf,site)

Echinodermatadf<-as.data.frame(Echinodermata)
colnames(Echinodermatadf)[1] ="rich"
site<-rep("Echinodermata",times=nrow(Echinodermatadf))
Echinodermatadffinal<-cbind(Echinodermatadf,site)

Ctenophoradf<-as.data.frame(Ctenophora)
colnames(Ctenophoradf)[1] ="rich"
site<-rep("Ctenophora",times=nrow(Ctenophoradf))
Ctenophoradffinal<-cbind(Ctenophoradf,site)

Craniatadf<-as.data.frame(Craniata)
colnames(Craniatadf)[1] ="rich"
site<-rep("Craniata",times=nrow(Craniatadf))
Craniatadffinal<-cbind(Craniatadf,site)

Cnidariadf<-as.data.frame(Cnidaria)
colnames(Cnidariadf)[1] ="rich"
site<-rep("Cnidaria",times=nrow(Cnidariadf))
Cnidariadffinal<-cbind(Cnidariadf,site)

Bryozoadf<-as.data.frame(Bryozoa)
colnames(Bryozoadf)[1] ="rich"
site<-rep("Bryozoa",times=nrow(Bryozoadf))
Bryozoadffinal<-cbind(Bryozoadf,site)

Brachiopodadf<-as.data.frame(Brachiopoda)
colnames(Brachiopodadf)[1] ="rich"
site<-rep("Brachiopoda",times=nrow(Brachiopodadf))
Brachiopodadffinal<-cbind(Brachiopodadf,site)

Arthropodadf<-as.data.frame(Arthropoda)
colnames(Arthropodadf)[1] ="rich"
site<-rep("Arthropoda",times=nrow(Arthropodadf))
Arthropodadffinal<-cbind(Arthropodadf,site)

Annelidadf<-as.data.frame(Annelida)
colnames(Annelidadf)[1] ="rich"
site<-rep("Annelida",times=nrow(Annelidadf))
Annelidadffinal<-cbind(Annelidadf,site)
```

```{r}
#now rbind them together
finalbindmetazoa<-do.call("rbind",list(Annelidadffinal,Arthropodadffinal,Brachiopodadffinal,Bryozoadffinal,Cnidariadffinal,Craniatadffinal,Ctenophoradffinal,Echinodermatadffinal,Entoproctadffinal,Gastrotrichadffinal,Gnathostomulidadffinal,Hemichordatadffinal,Kinorhynchadffinal,Molluscadffinal,Nematodadffinal,Nematomorphadffinal,Nemerteadffinal,Platyhelminthesdffinal,Poriferadffinal,Rotiferadffinal,Tardigradadffinal,Urochordatadffinal))
```


```{r}
library(ggplot2)
full_2_metazoa<-ggplot(data=finalbindmetazoa,aes(x=site,y=log(rich),color=factor(site))) +
  geom_boxplot()+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Unique ASVs")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
full_2_metazoa
```

```{r}
ggsave(filename = "full_species_richness_NEON_unique_ASV_metazoa.pdf",
      plot = full_2_metazoa,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
library(patchwork)
combined_full_metazoa<-full+full_2_metazoa
combined_full_metazoa
```

```{r}
ggsave(filename = "full_species_richness_NEON_unique_ASV_metazoa_combined.pdf",
      plot = combined_full_metazoa,
      bg = "transparent",
      width = 10, height = 6, units = "in",
      dpi = 600)
```


Fungi only

```{r}
input <- filter_taxa_from_input(input, taxa_to_keep = 'Fungi')
```

```{r}
tax_sum_phyla = summarize_taxonomy(input, level = 4, report_higher_tax = FALSE)
tax_sum_phyla_t<-t(tax_sum_phyla)
library(reshape2)
tax_sum_phyla_t_melt<-melt(tax_sum_phyla_t)
boxplot(tax_sum_phyla_t)
library(ggplot2)
```

```{r}
full<-ggplot(data=tax_sum_phyla_t_melt,aes(x=Var2,y=value,color=factor(Var2))) +
  geom_boxplot()+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Relative Abundance")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))

ggsave(filename = "full_species_richness_NEON_fungi_only.pdf",
      plot = full,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```



```{r}
setwd("/Users/leenavilonen/Documents/NEON")
library(mctoolsr)
library(vegan)
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Fungi:mito')
inputfinal <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
inputfinal <- filter_taxa_from_input(input, taxa_to_keep = 'Fungi')
```

```{r}
inputZoopagomycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Zoopagomycota')
Zoopagomycota<-specnumber(t(inputZoopagomycota$data_loaded))

inputMucoromycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Mucoromycota')
Mucoromycota<-specnumber(t(inputMucoromycota$data_loaded))

inputMicrosporidiomycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Microsporidiomycota')
Microsporidiomycota<-specnumber(t(inputMicrosporidiomycota$data_loaded))

inputKickxellomycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Kickxellomycota')
Kickxellomycota<-specnumber(t(inputKickxellomycota$data_loaded))

inputGlomeromycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Glomeromycota')
Glomeromycota<-specnumber(t(inputGlomeromycota$data_loaded))

inputEntomophthoromycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Entomophthoromycota')
Entomophthoromycota<-specnumber(t(inputEntomophthoromycota$data_loaded))

inputCryptomycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Cryptomycota')
Cryptomycota<-specnumber(t(inputCryptomycota$data_loaded))

inputChytridiomycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Chytridiomycota')
Chytridiomycota<-specnumber(t(inputChytridiomycota$data_loaded))

inputBlastocladiomycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Blastocladiomycota')
Blastocladiomycota<-specnumber(t(inputBlastocladiomycota$data_loaded))

inputBasidiomycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Basidiomycota')
Basidiomycota<-specnumber(t(inputBasidiomycota$data_loaded))

inputAscomycota <- filter_taxa_from_input(inputfinal, taxa_to_keep = 'Ascomycota')
Ascomycota<-specnumber(t(inputAscomycota$data_loaded))
```

```{r}
Zoopagomycotadf<-as.data.frame(Zoopagomycota)
colnames(Zoopagomycotadf)[1] ="rich"
site<-rep("Zoopagomycota",times=nrow(Zoopagomycotadf))
Zoopagomycotadffinal<-cbind(Zoopagomycotadf,site)

Mucoromycotadf<-as.data.frame(Mucoromycota)
colnames(Mucoromycotadf)[1] ="rich"
site<-rep("Mucoromycota",times=nrow(Mucoromycotadf))
Mucoromycotadffinal<-cbind(Mucoromycotadf,site)

Microsporidiomycotadf<-as.data.frame(Microsporidiomycota)
colnames(Microsporidiomycotadf)[1] ="rich"
site<-rep("Microsporidiomycota",times=nrow(Microsporidiomycotadf))
Microsporidiomycotadffinal<-cbind(Microsporidiomycotadf,site)

Kickxellomycotadf<-as.data.frame(Kickxellomycota)
colnames(Kickxellomycotadf)[1] ="rich"
site<-rep("Kickxellomycota",times=nrow(Kickxellomycotadf))
Kickxellomycotadffinal<-cbind(Kickxellomycotadf,site)

Glomeromycotadf<-as.data.frame(Glomeromycota)
colnames(Glomeromycotadf)[1] ="rich"
site<-rep("Glomeromycota",times=nrow(Glomeromycotadf))
Glomeromycotadffinal<-cbind(Glomeromycotadf,site)

Entomophthoromycotadf<-as.data.frame(Entomophthoromycota)
colnames(Entomophthoromycotadf)[1] ="rich"
site<-rep("Entomophthoromycota",times=nrow(Entomophthoromycotadf))
Entomophthoromycotadffinal<-cbind(Entomophthoromycotadf,site)

Cryptomycotadf<-as.data.frame(Cryptomycota)
colnames(Cryptomycotadf)[1] ="rich"
site<-rep("Cryptomycota",times=nrow(Cryptomycotadf))
Cryptomycotadffinal<-cbind(Cryptomycotadf,site)

Chytridiomycotadf<-as.data.frame(Chytridiomycota)
colnames(Chytridiomycotadf)[1] ="rich"
site<-rep("Chytridiomycota",times=nrow(Chytridiomycotadf))
Chytridiomycotadffinal<-cbind(Chytridiomycotadf,site)

Blastocladiomycotadf<-as.data.frame(Blastocladiomycota)
colnames(Blastocladiomycotadf)[1] ="rich"
site<-rep("Blastocladiomycota",times=nrow(Blastocladiomycotadf))
Blastocladiomycotadffinal<-cbind(Blastocladiomycotadf,site)

Basidiomycotadf<-as.data.frame(Basidiomycota)
colnames(Basidiomycotadf)[1] ="rich"
site<-rep("Basidiomycota",times=nrow(Basidiomycotadf))
Basidiomycotadffinal<-cbind(Basidiomycotadf,site)

Ascomycotadf<-as.data.frame(Ascomycota)
colnames(Ascomycotadf)[1] ="rich"
site<-rep("Ascomycota",times=nrow(Ascomycotadf))
Ascomycotadffinal<-cbind(Ascomycotadf,site)
```

```{r}
#now rbind them together
finalbindfungi<-do.call("rbind",list(Ascomycotadffinal,Basidiomycotadffinal,Blastocladiomycotadffinal,Chytridiomycotadffinal,Cryptomycotadffinal,Entomophthoromycotadffinal,Glomeromycotadffinal,Kickxellomycotadffinal,Microsporidiomycotadffinal,Mucoromycotadffinal,Zoopagomycotadffinal))
```


```{r}
library(ggplot2)
full_2_fungi<-ggplot(data=finalbindfungi,aes(x=site,y=log(rich),color=factor(site))) +
  geom_boxplot()+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Unique ASVs")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
full_2_fungi
```

```{r}
ggsave(filename = "full_species_richness_NEON_unique_ASV_fungi.pdf",
      plot = full_2_fungi,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
library(patchwork)
combined_full_fungi<-full+full_2_fungi
combined_full_fungi
```

```{r}
ggsave(filename = "full_species_richness_NEON_unique_ASV_fungi_combined.pdf",
      plot = combined_full_fungi,
      bg = "transparent",
      width = 10, height = 6, units = "in",
      dpi = 600)
```


Filter by site
```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
MLBS = filter_data(input, 'Site', keep_vals = 'MLBS')
HARV = filter_data(input, 'Site', keep_vals = 'HARV')
WOOD = filter_data(input, 'Site', keep_vals = 'WOOD')
LAJA = filter_data(input, 'Site', keep_vals = 'LAJA')
UNDE = filter_data(input, 'Site', keep_vals = 'UNDE')
SRER = filter_data(input, 'Site', keep_vals = 'SRER')
DSNY = filter_data(input, 'Site', keep_vals = 'DSNY')
BONA = filter_data(input, 'Site', keep_vals = 'BONA')
GUAN = filter_data(input, 'Site', keep_vals = 'GUAN')
ABBY = filter_data(input, 'Site', keep_vals = 'ABBY')
BARR = filter_data(input, 'Site', keep_vals = 'BARR')
BART = filter_data(input, 'Site', keep_vals = 'BART')
BLAN = filter_data(input, 'Site', keep_vals = 'BLAN')
CLBJ = filter_data(input, 'Site', keep_vals = 'CLBJ')
CPER = filter_data(input, 'Site', keep_vals = 'CPER')
DCFS = filter_data(input, 'Site', keep_vals = 'DCFS')
DEJU = filter_data(input, 'Site', keep_vals = 'DEJU')
DELA = filter_data(input, 'Site', keep_vals = 'DELA')
GRSM = filter_data(input, 'Site', keep_vals = 'GRSM')
HEAL = filter_data(input, 'Site', keep_vals = 'HEAL')
JERC = filter_data(input, 'Site', keep_vals = 'JERC')
JORN = filter_data(input, 'Site', keep_vals = 'JORN')
KONA = filter_data(input, 'Site', keep_vals = 'KONA')
KONZ = filter_data(input, 'Site', keep_vals = 'KONZ')
LENO = filter_data(input, 'Site', keep_vals = 'LENO')
MOAB = filter_data(input, 'Site', keep_vals = 'MOAB')
NIWO = filter_data(input, 'Site', keep_vals = 'NIWO')
OAES = filter_data(input, 'Site', keep_vals = 'OAES')
ONAQ = filter_data(input, 'Site', keep_vals = 'ONAQ')
ORNL = filter_data(input, 'Site', keep_vals = 'ORNL')
OSBS = filter_data(input, 'Site', keep_vals = 'OSBS')
PUUM = filter_data(input, 'Site', keep_vals = 'PUUM')
RMNP = filter_data(input, 'Site', keep_vals = 'RMNP')
SCBI = filter_data(input, 'Site', keep_vals = 'SCBI')
SERC = filter_data(input, 'Site', keep_vals = 'SERC')
SJER = filter_data(input, 'Site', keep_vals = 'SJER')
SOAP = filter_data(input, 'Site', keep_vals = 'SOAP')
STEI = filter_data(input, 'Site', keep_vals = 'STEI')
STER = filter_data(input, 'Site', keep_vals = 'STER')
TALL = filter_data(input, 'Site', keep_vals = 'TALL')
TOOL = filter_data(input, 'Site', keep_vals = 'TOOL')
TREE = filter_data(input, 'Site', keep_vals = 'TREE')
UKFS = filter_data(input, 'Site', keep_vals = 'UKFS')
WREF = filter_data(input, 'Site', keep_vals = 'WREF')
YELL = filter_data(input, 'Site', keep_vals = 'YELL')
```

```{r}
library(labdsv)
otuTAB<-ABBY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ABBY$data_loaded<-t(hell)

otuTAB<-BARR$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BARR$data_loaded<-t(hell)

otuTAB<-BART$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BART$data_loaded<-t(hell)

otuTAB<-BLAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BLAN$data_loaded<-t(hell)

otuTAB<-BONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BONA$data_loaded<-t(hell)

otuTAB<-CLBJ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CLBJ$data_loaded<-t(hell)

otuTAB<-CPER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CPER$data_loaded<-t(hell)

otuTAB<-DCFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DCFS$data_loaded<-t(hell)

otuTAB<-DELA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DELA$data_loaded<-t(hell)

otuTAB<-DSNY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DSNY$data_loaded<-t(hell)

otuTAB<-GRSM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GRSM$data_loaded<-t(hell)

otuTAB<-GUAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GUAN$data_loaded<-t(hell)

otuTAB<-HARV$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HARV$data_loaded<-t(hell)

otuTAB<-HEAL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HEAL$data_loaded<-t(hell)

otuTAB<-JERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JERC$data_loaded<-t(hell)

otuTAB<-JORN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JORN$data_loaded<-t(hell)

otuTAB<-KONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONA$data_loaded<-t(hell)

otuTAB<-KONZ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONZ$data_loaded<-t(hell)

otuTAB<-LAJA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LAJA$data_loaded<-t(hell)

otuTAB<-LENO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LENO$data_loaded<-t(hell)

otuTAB<-MLBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MLBS$data_loaded<-t(hell)

otuTAB<-MOAB$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MOAB$data_loaded<-t(hell)

otuTAB<-NIWO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
NIWO$data_loaded<-t(hell)

otuTAB<-OAES$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OAES$data_loaded<-t(hell)

otuTAB<-ONAQ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ONAQ$data_loaded<-t(hell)

otuTAB<-ORNL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ORNL$data_loaded<-t(hell)

otuTAB<-OSBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OSBS$data_loaded<-t(hell)

otuTAB<-PUUM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
PUUM$data_loaded<-t(hell)

otuTAB<-RMNP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
RMNP$data_loaded<-t(hell)

otuTAB<-SCBI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SCBI$data_loaded<-t(hell)

otuTAB<-SERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SERC$data_loaded<-t(hell)

otuTAB<-SJER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SJER$data_loaded<-t(hell)

otuTAB<-SOAP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SOAP$data_loaded<-t(hell)

otuTAB<-SRER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SRER$data_loaded<-t(hell)

otuTAB<-STEI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STEI$data_loaded<-t(hell)

otuTAB<-STER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STER$data_loaded<-t(hell)

otuTAB<-TALL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TALL$data_loaded<-t(hell)

otuTAB<-TOOL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TOOL$data_loaded<-t(hell)

otuTAB<-TREE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TREE$data_loaded<-t(hell)

otuTAB<-UKFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UKFS$data_loaded<-t(hell)

otuTAB<-UNDE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UNDE$data_loaded<-t(hell)

otuTAB<-WOOD$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WOOD$data_loaded<-t(hell)

otuTAB<-WREF$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WREF$data_loaded<-t(hell)

otuTAB<-YELL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
YELL$data_loaded<-t(hell)
```

```{r}
ABBYrich<-specnumber(t(ABBY$data_loaded))
BARRrich<-specnumber(t(BARR$data_loaded))
BARTrich<-specnumber(t(BART$data_loaded))
BLANrich<-specnumber(t(BLAN$data_loaded))
BONArich<-specnumber(t(BONA$data_loaded))
CLBJrich<-specnumber(t(CLBJ$data_loaded))
CPERrich<-specnumber(t(CPER$data_loaded))
DCFSrich<-specnumber(t(DCFS$data_loaded))
DEJUrich<-specnumber(t(DEJU$data_loaded))
DELArich<-specnumber(t(DELA$data_loaded))
DSNYrich<-specnumber(t(DSNY$data_loaded))
GRSMrich<-specnumber(t(GRSM$data_loaded))
GUANrich<-specnumber(t(GUAN$data_loaded))
HARVrich<-specnumber(t(HARV$data_loaded))
HEALrich<-specnumber(t(HEAL$data_loaded))
JERCrich<-specnumber(t(JERC$data_loaded))
JORNrich<-specnumber(t(JORN$data_loaded))
KONArich<-specnumber(t(KONA$data_loaded))
KONZrich<-specnumber(t(KONZ$data_loaded))
LAJArich<-specnumber(t(LAJA$data_loaded))
LENOrich<-specnumber(t(LENO$data_loaded))
MLBSrich<-specnumber(t(MLBS$data_loaded))
MOABrich<-specnumber(t(MOAB$data_loaded))
NIWOrich<-specnumber(t(NIWO$data_loaded))
OAESrich<-specnumber(t(OAES$data_loaded))
ONAQrich<-specnumber(t(ONAQ$data_loaded))
ORNLrich<-specnumber(t(ORNL$data_loaded))
OSBSrich<-specnumber(t(OSBS$data_loaded))
PUUMrich<-specnumber(t(PUUM$data_loaded))
RMNPrich<-specnumber(t(RMNP$data_loaded))
SCBIrich<-specnumber(t(SCBI$data_loaded))
SERCrich<-specnumber(t(SERC$data_loaded))
SJERrich<-specnumber(t(SJER$data_loaded))
SOAPrich<-specnumber(t(SOAP$data_loaded))
SRERrich<-specnumber(t(SRER$data_loaded))
STEIrich<-specnumber(t(STEI$data_loaded))
STERrich<-specnumber(t(STER$data_loaded))
TALLrich<-specnumber(t(TALL$data_loaded))
TOOLrich<-specnumber(t(TOOL$data_loaded))
TREErich<-specnumber(t(TREE$data_loaded))
UKFSrich<-specnumber(t(UKFS$data_loaded))
UNDErich<-specnumber(t(UNDE$data_loaded))
WOODrich<-specnumber(t(WOOD$data_loaded))
WREFrich<-specnumber(t(WREF$data_loaded))
YELLrich<-specnumber(t(YELL$data_loaded))
```

```{r}
ABBYrichdf<-as.data.frame(ABBYrich)
colnames(ABBYrichdf)[1] ="rich"
site<-rep("ABBY",times=nrow(ABBYrichdf))
ABBYrichdffinal<-cbind(ABBYrichdf,site)

BARRrichdf<-as.data.frame(BARRrich)
colnames(BARRrichdf)[1] ="rich"
site<-rep("BARR",times=nrow(BARRrichdf))
BARRrichdffinal<-cbind(BARRrichdf,site)

BARTrichdf<-as.data.frame(BARTrich)
colnames(BARTrichdf)[1] ="rich"
site<-rep("BART",times=nrow(BARTrichdf))
BARTrichdffinal<-cbind(BARTrichdf,site)

BLANrichdf<-as.data.frame(BLANrich)
colnames(BLANrichdf)[1] ="rich"
site<-rep("BLAN",times=nrow(BLANrichdf))
BLANrichdffinal<-cbind(BLANrichdf,site)

BONArichdf<-as.data.frame(BONArich)
colnames(BONArichdf)[1] ="rich"
site<-rep("BONA",times=nrow(BONArichdf))
BONArichdffinal<-cbind(BONArichdf,site)

CLBJrichdf<-as.data.frame(CLBJrich)
colnames(CLBJrichdf)[1] ="rich"
site<-rep("CLBJ",times=nrow(CLBJrichdf))
CLBJrichdffinal<-cbind(CLBJrichdf,site)

CPERrichdf<-as.data.frame(CPERrich)
colnames(CPERrichdf)[1] ="rich"
site<-rep("CPER",times=nrow(CPERrichdf))
CPERrichdffinal<-cbind(CPERrichdf,site)

DCFSrichdf<-as.data.frame(DCFSrich)
colnames(DCFSrichdf)[1] ="rich"
site<-rep("DCFS",times=nrow(DCFSrichdf))
DCFSrichdffinal<-cbind(DCFSrichdf,site)

DEJUrichdf<-as.data.frame(DEJUrich)
colnames(DEJUrichdf)[1] ="rich"
site<-rep("DEJU",times=nrow(DEJUrichdf))
DEJUrichdffinal<-cbind(DEJUrichdf,site)

DELArichdf<-as.data.frame(DELArich)
colnames(DELArichdf)[1] ="rich"
site<-rep("DELA",times=nrow(DELArichdf))
DELArichdffinal<-cbind(DELArichdf,site)

DSNYrichdf<-as.data.frame(DSNYrich)
colnames(DSNYrichdf)[1] ="rich"
site<-rep("DSNY",times=nrow(DSNYrichdf))
DSNYrichdffinal<-cbind(DSNYrichdf,site)

GRSMrichdf<-as.data.frame(GRSMrich)
colnames(GRSMrichdf)[1] ="rich"
site<-rep("GRSM",times=nrow(GRSMrichdf))
GRSMrichdffinal<-cbind(GRSMrichdf,site)

GUANrichdf<-as.data.frame(GUANrich)
colnames(GUANrichdf)[1] ="rich"
site<-rep("GUAN",times=nrow(GUANrichdf))
GUANrichdffinal<-cbind(GUANrichdf,site)

HARVrichdf<-as.data.frame(HARVrich)
colnames(HARVrichdf)[1] ="rich"
site<-rep("HARV",times=nrow(HARVrichdf))
HARVrichdffinal<-cbind(HARVrichdf,site)

HEALrichdf<-as.data.frame(HEALrich)
colnames(HEALrichdf)[1] ="rich"
site<-rep("HEAL",times=nrow(HEALrichdf))
HEALrichdffinal<-cbind(HEALrichdf,site)

JERCrichdf<-as.data.frame(JERCrich)
colnames(JERCrichdf)[1] ="rich"
site<-rep("JERC",times=nrow(JERCrichdf))
JERCrichdffinal<-cbind(JERCrichdf,site)

JORNrichdf<-as.data.frame(JORNrich)
colnames(JORNrichdf)[1] ="rich"
site<-rep("JORN",times=nrow(JORNrichdf))
JORNrichdffinal<-cbind(JORNrichdf,site)

KONArichdf<-as.data.frame(KONArich)
colnames(KONArichdf)[1] ="rich"
site<-rep("KONA",times=nrow(KONArichdf))
KONArichdffinal<-cbind(KONArichdf,site)

KONZrichdf<-as.data.frame(KONZrich)
colnames(KONZrichdf)[1] ="rich"
site<-rep("KONZ",times=nrow(KONZrichdf))
KONZrichdffinal<-cbind(KONZrichdf,site)

LAJArichdf<-as.data.frame(LAJArich)
colnames(LAJArichdf)[1] ="rich"
site<-rep("LAJA",times=nrow(LAJArichdf))
LAJArichdffinal<-cbind(LAJArichdf,site)

LENOrichdf<-as.data.frame(LENOrich)
colnames(LENOrichdf)[1] ="rich"
site<-rep("LENO",times=nrow(LENOrichdf))
LENOrichdffinal<-cbind(LENOrichdf,site)

MLBSrichdf<-as.data.frame(MLBSrich)
colnames(MLBSrichdf)[1] ="rich"
site<-rep("MLBS",times=nrow(MLBSrichdf))
MLBSrichdffinal<-cbind(MLBSrichdf,site)

MOABrichdf<-as.data.frame(MOABrich)
colnames(MOABrichdf)[1] ="rich"
site<-rep("MOAB",times=nrow(MOABrichdf))
MOABrichdffinal<-cbind(MOABrichdf,site)

NIWOrichdf<-as.data.frame(NIWOrich)
colnames(NIWOrichdf)[1] ="rich"
site<-rep("NIWO",times=nrow(NIWOrichdf))
NIWOrichdffinal<-cbind(NIWOrichdf,site)

OAESrichdf<-as.data.frame(OAESrich)
colnames(OAESrichdf)[1] ="rich"
site<-rep("OAES",times=nrow(OAESrichdf))
OAESrichdffinal<-cbind(OAESrichdf,site)

ONAQrichdf<-as.data.frame(ONAQrich)
colnames(ONAQrichdf)[1] ="rich"
site<-rep("ONAQ",times=nrow(ONAQrichdf))
ONAQrichdffinal<-cbind(ONAQrichdf,site)

ORNLrichdf<-as.data.frame(ORNLrich)
colnames(ORNLrichdf)[1] ="rich"
site<-rep("ORNL",times=nrow(ORNLrichdf))
ORNLrichdffinal<-cbind(ORNLrichdf,site)

OSBSrichdf<-as.data.frame(OSBSrich)
colnames(OSBSrichdf)[1] ="rich"
site<-rep("OSBS",times=nrow(OSBSrichdf))
OSBSrichdffinal<-cbind(OSBSrichdf,site)

PUUMrichdf<-as.data.frame(PUUMrich)
colnames(PUUMrichdf)[1] ="rich"
site<-rep("PUUM",times=nrow(PUUMrichdf))
PUUMrichdffinal<-cbind(PUUMrichdf,site)

RMNPrichdf<-as.data.frame(RMNPrich)
colnames(RMNPrichdf)[1] ="rich"
site<-rep("RMNP",times=nrow(RMNPrichdf))
RMNPrichdffinal<-cbind(RMNPrichdf,site)

SCBIrichdf<-as.data.frame(SCBIrich)
colnames(SCBIrichdf)[1] ="rich"
site<-rep("SCBI",times=nrow(SCBIrichdf))
SCBIrichdffinal<-cbind(SCBIrichdf,site)

SERCrichdf<-as.data.frame(SERCrich)
colnames(SERCrichdf)[1] ="rich"
site<-rep("SERC",times=nrow(SERCrichdf))
SERCrichdffinal<-cbind(SERCrichdf,site)

SJERrichdf<-as.data.frame(SJERrich)
colnames(SJERrichdf)[1] ="rich"
site<-rep("SJER",times=nrow(SJERrichdf))
SJERrichdffinal<-cbind(SJERrichdf,site)

SOAPrichdf<-as.data.frame(SOAPrich)
colnames(SOAPrichdf)[1] ="rich"
site<-rep("SOAP",times=nrow(SOAPrichdf))
SOAPrichdffinal<-cbind(SOAPrichdf,site)

SRERrichdf<-as.data.frame(SRERrich)
colnames(SRERrichdf)[1] ="rich"
site<-rep("SRER",times=nrow(SRERrichdf))
SRERrichdffinal<-cbind(SRERrichdf,site)

STEIrichdf<-as.data.frame(STEIrich)
colnames(STEIrichdf)[1] ="rich"
site<-rep("STEI",times=nrow(STEIrichdf))
STEIrichdffinal<-cbind(STEIrichdf,site)

STERrichdf<-as.data.frame(STERrich)
colnames(STERrichdf)[1] ="rich"
site<-rep("STER",times=nrow(STERrichdf))
STERrichdffinal<-cbind(STERrichdf,site)

TALLrichdf<-as.data.frame(TALLrich)
colnames(TALLrichdf)[1] ="rich"
site<-rep("TALL",times=nrow(TALLrichdf))
TALLrichdffinal<-cbind(TALLrichdf,site)

TOOLrichdf<-as.data.frame(TOOLrich)
colnames(TOOLrichdf)[1] ="rich"
site<-rep("TOOL",times=nrow(TOOLrichdf))
TOOLrichdffinal<-cbind(TOOLrichdf,site)

TREErichdf<-as.data.frame(TREErich)
colnames(TREErichdf)[1] ="rich"
site<-rep("TREE",times=nrow(TREErichdf))
TREErichdffinal<-cbind(TREErichdf,site)

UKFSrichdf<-as.data.frame(UKFSrich)
colnames(UKFSrichdf)[1] ="rich"
site<-rep("UKFS",times=nrow(UKFSrichdf))
UKFSrichdffinal<-cbind(UKFSrichdf,site)

UNDErichdf<-as.data.frame(UNDErich)
colnames(UNDErichdf)[1] ="rich"
site<-rep("UNDE",times=nrow(UNDErichdf))
UNDErichdffinal<-cbind(UNDErichdf,site)

WOODrichdf<-as.data.frame(WOODrich)
colnames(WOODrichdf)[1] ="rich"
site<-rep("WOOD",times=nrow(WOODrichdf))
WOODrichdffinal<-cbind(WOODrichdf,site)

WREFrichdf<-as.data.frame(WREFrich)
colnames(WREFrichdf)[1] ="rich"
site<-rep("WREF",times=nrow(WREFrichdf))
WREFrichdffinal<-cbind(WREFrichdf,site)

YELLrichdf<-as.data.frame(YELLrich)
colnames(YELLrichdf)[1] ="rich"
site<-rep("YELL",times=nrow(YELLrichdf))
YELLrichdffinal<-cbind(YELLrichdf,site)
```

```{r}
#now rbind them together
richfinalbind<-do.call("rbind",list(ABBYrichdffinal,BARRrichdffinal,BARTrichdffinal,BLANrichdffinal,BONArichdffinal,CLBJrichdffinal,CPERrichdffinal,DCFSrichdffinal,DEJUrichdffinal,DELArichdffinal,DSNYrichdffinal,GRSMrichdffinal,GUANrichdffinal,HARVrichdffinal,HEALrichdffinal,JERCrichdffinal,JORNrichdffinal,KONArichdffinal,KONZrichdffinal,LAJArichdffinal,LENOrichdffinal,MLBSrichdffinal,MOABrichdffinal,NIWOrichdffinal,OAESrichdffinal,ONAQrichdffinal,ORNLrichdffinal,OSBSrichdffinal,PUUMrichdffinal,RMNPrichdffinal,SCBIrichdffinal,SERCrichdffinal,SJERrichdffinal,SOAPrichdffinal,SRERrichdffinal,STEIrichdffinal,STERrichdffinal,TALLrichdffinal,TOOLrichdffinal,TREErichdffinal,UKFSrichdffinal,UNDErichdffinal,WOODrichdffinal,WREFrichdffinal,YELLrichdffinal))
```

positions <- c("Goalkeeper", "Defense", "Striker")
p <- ggplot(theTable, aes(x = Position)) + scale_x_discrete(limits = positions)

# By latitude

```{r}
library(ggplot2)
positions <- c("GUAN","LAJA","PUUM","DSNY","OSBS","JERC","LENO","SRER","DELA","JORN","TALL","CLBJ","OAES","GRSM","ORNL","SOAP","SJER","MLBS","MOAB","SERC","SCBI","BLAN","UKFS","KONZ","KONA","NIWO","ONAQ","RMNP","STER","CPER","HARV","BART","YELL","TREE","STEI","ABBY","WREF","UNDE","WOOD","DCFS","HEAL","DEJU","BONA","TOOL","BARR")
speciesrichness<-ggplot(data=richfinalbind,aes(x=site,y=rich))+
 geom_boxplot()+
  scale_x_discrete(limits = positions)+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Richness (# of species)")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))

ggsave(filename = "speciesrichness_NEON_latitude.pdf",
      plot = speciesrichness,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600) 
```

#Would be nice to do by biome. By increasing MAP
```{r}
positions <- c("BARR", "BONA", "JORN","ONAQ","DEJU","TOOL","MOAB","CPER","SRER","HEAL","STER","DCFS","YELL","WOOD","SJER","RMNP","OAES","STEI","TREE","UNDE","LAJA","GUAN","KONA","KONZ","SOAP","CLBJ","BLAN","UKFS","NIWO","SERC","SCBI","HARV","DSNY","MLBS","OSBS","JERC","BART","ORNL","DELA","GRSM","TALL","LENO","WREF","ABBY","PUUM")
speciesrichness<-ggplot(data=richfinalbind,aes(x=site,y=rich))+
 geom_boxplot()+
  scale_x_discrete(limits = positions)+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Richness (# of species)")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))

ggsave(filename = "speciesrichness_NEON_MAP.pdf",
      plot = speciesrichness,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600) 
```

By increasing MAT

```{r}
positions <- c("HEAL","BARR","BONA","DEJU","TOOL","NIWO","ABBY","MOAB","SCBI","BLAN","KONZ","KONA","UKFS","PUUM","GRSM","SOAP","SERC","ORNL","OAES","JORN","SJER","TALL","CLBJ","DELA","LENO","JERC","SRER","RMNP","OSBS","DSNY","GUAN","LAJA","YELL","UNDE","STEI","TREE","DCFS","WOOD","BART","HARV","CPER","MLBS","ONAQ","WREF","STER")
speciesrichness<-ggplot(data=richfinalbind,aes(x=site,y=rich))+
 geom_boxplot()+
  scale_x_discrete(limits = positions)+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Richness (# of species)")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))

ggsave(filename = "speciesrichness_NEON_MAT.pdf",
      plot = speciesrichness,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600) 
```

# Just with the managed sites next to eachother

```{r}
richfinalbind<-do.call("rbind",list(WREFrichdffinal,ABBYrichdffinal,TREErichdffinal,STEIrichdffinal,KONZrichdffinal,KONArichdffinal,CPERrichdffinal,STERrichdffinal,WOODrichdffinal,DCFSrichdffinal,GUANrichdffinal,LAJArichdffinal))
```

```{r}
positions <- c("CPER","STER","KONZ","KONA","WOOD","DCFS","GUAN","LAJA","WREF","ABBY","TREE","STEI")
speciesrichness<-ggplot(data=richfinalbind,aes(x=site,y=rich))+
 geom_boxplot()+
  scale_x_discrete(limits = positions)+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Richness (# of species)")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))

ggsave(filename = "speciesrichness_NEON_managed_by_MAP.pdf",
      plot = speciesrichness,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600) 
```

Fixing managed graph! 

```{r}
library(mctoolsr)
library(ggplot2)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input = filter_data(input, 'plotType', keep_vals = 'tower')
MLBS = filter_data(input, 'Site', keep_vals = 'MLBS')
HARV = filter_data(input, 'Site', keep_vals = 'HARV')
WOOD = filter_data(input, 'Site', keep_vals = 'WOOD')
LAJA = filter_data(input, 'Site', keep_vals = 'LAJA')
UNDE = filter_data(input, 'Site', keep_vals = 'UNDE')
SRER = filter_data(input, 'Site', keep_vals = 'SRER')
DSNY = filter_data(input, 'Site', keep_vals = 'DSNY')
BONA = filter_data(input, 'Site', keep_vals = 'BONA')
GUAN = filter_data(input, 'Site', keep_vals = 'GUAN')
ABBY = filter_data(input, 'Site', keep_vals = 'ABBY')
BARR = filter_data(input, 'Site', keep_vals = 'BARR')
BART = filter_data(input, 'Site', keep_vals = 'BART')
BLAN = filter_data(input, 'Site', keep_vals = 'BLAN')
CLBJ = filter_data(input, 'Site', keep_vals = 'CLBJ')
CPER = filter_data(input, 'Site', keep_vals = 'CPER')
DCFS = filter_data(input, 'Site', keep_vals = 'DCFS')
DEJU = filter_data(input, 'Site', keep_vals = 'DEJU')
DELA = filter_data(input, 'Site', keep_vals = 'DELA')
GRSM = filter_data(input, 'Site', keep_vals = 'GRSM')
HEAL = filter_data(input, 'Site', keep_vals = 'HEAL')
JERC = filter_data(input, 'Site', keep_vals = 'JERC')
JORN = filter_data(input, 'Site', keep_vals = 'JORN')
KONA = filter_data(input, 'Site', keep_vals = 'KONA')
KONZ = filter_data(input, 'Site', keep_vals = 'KONZ')
LENO = filter_data(input, 'Site', keep_vals = 'LENO')
MOAB = filter_data(input, 'Site', keep_vals = 'MOAB')
NIWO = filter_data(input, 'Site', keep_vals = 'NIWO')
OAES = filter_data(input, 'Site', keep_vals = 'OAES')
ONAQ = filter_data(input, 'Site', keep_vals = 'ONAQ')
ORNL = filter_data(input, 'Site', keep_vals = 'ORNL')
OSBS = filter_data(input, 'Site', keep_vals = 'OSBS')
PUUM = filter_data(input, 'Site', keep_vals = 'PUUM')
RMNP = filter_data(input, 'Site', keep_vals = 'RMNP')
SCBI = filter_data(input, 'Site', keep_vals = 'SCBI')
SERC = filter_data(input, 'Site', keep_vals = 'SERC')
SJER = filter_data(input, 'Site', keep_vals = 'SJER')
SOAP = filter_data(input, 'Site', keep_vals = 'SOAP')
STEI = filter_data(input, 'Site', keep_vals = 'STEI')
STER = filter_data(input, 'Site', keep_vals = 'STER')
TALL = filter_data(input, 'Site', keep_vals = 'TALL')
TOOL = filter_data(input, 'Site', keep_vals = 'TOOL')
#TREE = filter_data(input, 'Site', keep_vals = 'TREE')
UKFS = filter_data(input, 'Site', keep_vals = 'UKFS')
WREF = filter_data(input, 'Site', keep_vals = 'WREF')
YELL = filter_data(input, 'Site', keep_vals = 'YELL')
```

```{r}
library(labdsv)
otuTAB<-ABBY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ABBY$data_loaded<-t(hell)

otuTAB<-BARR$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BARR$data_loaded<-t(hell)

otuTAB<-BART$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BART$data_loaded<-t(hell)

otuTAB<-BLAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BLAN$data_loaded<-t(hell)

otuTAB<-BONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BONA$data_loaded<-t(hell)

otuTAB<-CLBJ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CLBJ$data_loaded<-t(hell)

otuTAB<-CPER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CPER$data_loaded<-t(hell)

otuTAB<-DCFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DCFS$data_loaded<-t(hell)

otuTAB<-DELA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DELA$data_loaded<-t(hell)

otuTAB<-DSNY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DSNY$data_loaded<-t(hell)

otuTAB<-GRSM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GRSM$data_loaded<-t(hell)

otuTAB<-GUAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GUAN$data_loaded<-t(hell)

otuTAB<-HARV$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HARV$data_loaded<-t(hell)

otuTAB<-HEAL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HEAL$data_loaded<-t(hell)

otuTAB<-JERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JERC$data_loaded<-t(hell)

otuTAB<-JORN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JORN$data_loaded<-t(hell)

otuTAB<-KONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONA$data_loaded<-t(hell)

otuTAB<-KONZ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONZ$data_loaded<-t(hell)

otuTAB<-LAJA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LAJA$data_loaded<-t(hell)

otuTAB<-LENO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LENO$data_loaded<-t(hell)

otuTAB<-MLBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MLBS$data_loaded<-t(hell)

otuTAB<-MOAB$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MOAB$data_loaded<-t(hell)

otuTAB<-NIWO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
NIWO$data_loaded<-t(hell)

otuTAB<-OAES$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OAES$data_loaded<-t(hell)

otuTAB<-ONAQ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ONAQ$data_loaded<-t(hell)

otuTAB<-ORNL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ORNL$data_loaded<-t(hell)

otuTAB<-OSBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OSBS$data_loaded<-t(hell)

otuTAB<-PUUM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
PUUM$data_loaded<-t(hell)

otuTAB<-RMNP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
RMNP$data_loaded<-t(hell)

otuTAB<-SCBI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SCBI$data_loaded<-t(hell)

otuTAB<-SERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SERC$data_loaded<-t(hell)

otuTAB<-SJER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SJER$data_loaded<-t(hell)

otuTAB<-SOAP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SOAP$data_loaded<-t(hell)

otuTAB<-SRER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SRER$data_loaded<-t(hell)

otuTAB<-STEI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STEI$data_loaded<-t(hell)

otuTAB<-STER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STER$data_loaded<-t(hell)

otuTAB<-TALL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TALL$data_loaded<-t(hell)

otuTAB<-TOOL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TOOL$data_loaded<-t(hell)

#otuTAB<-TREE$data_loaded
#otuTAB <- as.data.frame (otuTAB)
#otuTABtranspose<-t(otuTAB)
#hell<-hellinger(otuTABtranspose)
#TREE$data_loaded<-t(hell)

otuTAB<-UKFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UKFS$data_loaded<-t(hell)

otuTAB<-UNDE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UNDE$data_loaded<-t(hell)

otuTAB<-WOOD$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WOOD$data_loaded<-t(hell)

otuTAB<-WREF$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WREF$data_loaded<-t(hell)

otuTAB<-YELL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
YELL$data_loaded<-t(hell)
```

```{r}
library(vegan)
ABBYrich<-specnumber(t(ABBY$data_loaded))
BARRrich<-specnumber(t(BARR$data_loaded))
BARTrich<-specnumber(t(BART$data_loaded))
BLANrich<-specnumber(t(BLAN$data_loaded))
BONArich<-specnumber(t(BONA$data_loaded))
CLBJrich<-specnumber(t(CLBJ$data_loaded))
CPERrich<-specnumber(t(CPER$data_loaded))
DCFSrich<-specnumber(t(DCFS$data_loaded))
DEJUrich<-specnumber(t(DEJU$data_loaded))
DELArich<-specnumber(t(DELA$data_loaded))
DSNYrich<-specnumber(t(DSNY$data_loaded))
GRSMrich<-specnumber(t(GRSM$data_loaded))
GUANrich<-specnumber(t(GUAN$data_loaded))
HARVrich<-specnumber(t(HARV$data_loaded))
HEALrich<-specnumber(t(HEAL$data_loaded))
JERCrich<-specnumber(t(JERC$data_loaded))
JORNrich<-specnumber(t(JORN$data_loaded))
KONArich<-specnumber(t(KONA$data_loaded))
KONZrich<-specnumber(t(KONZ$data_loaded))
LAJArich<-specnumber(t(LAJA$data_loaded))
LENOrich<-specnumber(t(LENO$data_loaded))
MLBSrich<-specnumber(t(MLBS$data_loaded))
MOABrich<-specnumber(t(MOAB$data_loaded))
NIWOrich<-specnumber(t(NIWO$data_loaded))
OAESrich<-specnumber(t(OAES$data_loaded))
ONAQrich<-specnumber(t(ONAQ$data_loaded))
ORNLrich<-specnumber(t(ORNL$data_loaded))
OSBSrich<-specnumber(t(OSBS$data_loaded))
PUUMrich<-specnumber(t(PUUM$data_loaded))
RMNPrich<-specnumber(t(RMNP$data_loaded))
SCBIrich<-specnumber(t(SCBI$data_loaded))
SERCrich<-specnumber(t(SERC$data_loaded))
SJERrich<-specnumber(t(SJER$data_loaded))
SOAPrich<-specnumber(t(SOAP$data_loaded))
SRERrich<-specnumber(t(SRER$data_loaded))
STEIrich<-specnumber(t(STEI$data_loaded))
STERrich<-specnumber(t(STER$data_loaded))
TALLrich<-specnumber(t(TALL$data_loaded))
TOOLrich<-specnumber(t(TOOL$data_loaded))
#TREErich<-specnumber(t(TREE$data_loaded))
UKFSrich<-specnumber(t(UKFS$data_loaded))
UNDErich<-specnumber(t(UNDE$data_loaded))
WOODrich<-specnumber(t(WOOD$data_loaded))
WREFrich<-specnumber(t(WREF$data_loaded))
YELLrich<-specnumber(t(YELL$data_loaded))
```

```{r}
ABBYrichdf<-as.data.frame(ABBYrich)
colnames(ABBYrichdf)[1] ="rich"
site<-rep("ABBY",times=nrow(ABBYrichdf))
ABBYrichdffinal<-cbind(ABBYrichdf,site)

BARRrichdf<-as.data.frame(BARRrich)
colnames(BARRrichdf)[1] ="rich"
site<-rep("BARR",times=nrow(BARRrichdf))
BARRrichdffinal<-cbind(BARRrichdf,site)

BARTrichdf<-as.data.frame(BARTrich)
colnames(BARTrichdf)[1] ="rich"
site<-rep("BART",times=nrow(BARTrichdf))
BARTrichdffinal<-cbind(BARTrichdf,site)

BLANrichdf<-as.data.frame(BLANrich)
colnames(BLANrichdf)[1] ="rich"
site<-rep("BLAN",times=nrow(BLANrichdf))
BLANrichdffinal<-cbind(BLANrichdf,site)

BONArichdf<-as.data.frame(BONArich)
colnames(BONArichdf)[1] ="rich"
site<-rep("BONA",times=nrow(BONArichdf))
BONArichdffinal<-cbind(BONArichdf,site)

CLBJrichdf<-as.data.frame(CLBJrich)
colnames(CLBJrichdf)[1] ="rich"
site<-rep("CLBJ",times=nrow(CLBJrichdf))
CLBJrichdffinal<-cbind(CLBJrichdf,site)

CPERrichdf<-as.data.frame(CPERrich)
colnames(CPERrichdf)[1] ="rich"
site<-rep("CPER",times=nrow(CPERrichdf))
CPERrichdffinal<-cbind(CPERrichdf,site)

DCFSrichdf<-as.data.frame(DCFSrich)
colnames(DCFSrichdf)[1] ="rich"
site<-rep("DCFS",times=nrow(DCFSrichdf))
DCFSrichdffinal<-cbind(DCFSrichdf,site)

DEJUrichdf<-as.data.frame(DEJUrich)
colnames(DEJUrichdf)[1] ="rich"
site<-rep("DEJU",times=nrow(DEJUrichdf))
DEJUrichdffinal<-cbind(DEJUrichdf,site)

DELArichdf<-as.data.frame(DELArich)
colnames(DELArichdf)[1] ="rich"
site<-rep("DELA",times=nrow(DELArichdf))
DELArichdffinal<-cbind(DELArichdf,site)

DSNYrichdf<-as.data.frame(DSNYrich)
colnames(DSNYrichdf)[1] ="rich"
site<-rep("DSNY",times=nrow(DSNYrichdf))
DSNYrichdffinal<-cbind(DSNYrichdf,site)

GRSMrichdf<-as.data.frame(GRSMrich)
colnames(GRSMrichdf)[1] ="rich"
site<-rep("GRSM",times=nrow(GRSMrichdf))
GRSMrichdffinal<-cbind(GRSMrichdf,site)

GUANrichdf<-as.data.frame(GUANrich)
colnames(GUANrichdf)[1] ="rich"
site<-rep("GUAN",times=nrow(GUANrichdf))
GUANrichdffinal<-cbind(GUANrichdf,site)

HARVrichdf<-as.data.frame(HARVrich)
colnames(HARVrichdf)[1] ="rich"
site<-rep("HARV",times=nrow(HARVrichdf))
HARVrichdffinal<-cbind(HARVrichdf,site)

HEALrichdf<-as.data.frame(HEALrich)
colnames(HEALrichdf)[1] ="rich"
site<-rep("HEAL",times=nrow(HEALrichdf))
HEALrichdffinal<-cbind(HEALrichdf,site)

JERCrichdf<-as.data.frame(JERCrich)
colnames(JERCrichdf)[1] ="rich"
site<-rep("JERC",times=nrow(JERCrichdf))
JERCrichdffinal<-cbind(JERCrichdf,site)

JORNrichdf<-as.data.frame(JORNrich)
colnames(JORNrichdf)[1] ="rich"
site<-rep("JORN",times=nrow(JORNrichdf))
JORNrichdffinal<-cbind(JORNrichdf,site)

KONArichdf<-as.data.frame(KONArich)
colnames(KONArichdf)[1] ="rich"
site<-rep("KONA",times=nrow(KONArichdf))
KONArichdffinal<-cbind(KONArichdf,site)

KONZrichdf<-as.data.frame(KONZrich)
colnames(KONZrichdf)[1] ="rich"
site<-rep("KONZ",times=nrow(KONZrichdf))
KONZrichdffinal<-cbind(KONZrichdf,site)

LAJArichdf<-as.data.frame(LAJArich)
colnames(LAJArichdf)[1] ="rich"
site<-rep("LAJA",times=nrow(LAJArichdf))
LAJArichdffinal<-cbind(LAJArichdf,site)

LENOrichdf<-as.data.frame(LENOrich)
colnames(LENOrichdf)[1] ="rich"
site<-rep("LENO",times=nrow(LENOrichdf))
LENOrichdffinal<-cbind(LENOrichdf,site)

MLBSrichdf<-as.data.frame(MLBSrich)
colnames(MLBSrichdf)[1] ="rich"
site<-rep("MLBS",times=nrow(MLBSrichdf))
MLBSrichdffinal<-cbind(MLBSrichdf,site)

MOABrichdf<-as.data.frame(MOABrich)
colnames(MOABrichdf)[1] ="rich"
site<-rep("MOAB",times=nrow(MOABrichdf))
MOABrichdffinal<-cbind(MOABrichdf,site)

NIWOrichdf<-as.data.frame(NIWOrich)
colnames(NIWOrichdf)[1] ="rich"
site<-rep("NIWO",times=nrow(NIWOrichdf))
NIWOrichdffinal<-cbind(NIWOrichdf,site)

OAESrichdf<-as.data.frame(OAESrich)
colnames(OAESrichdf)[1] ="rich"
site<-rep("OAES",times=nrow(OAESrichdf))
OAESrichdffinal<-cbind(OAESrichdf,site)

ONAQrichdf<-as.data.frame(ONAQrich)
colnames(ONAQrichdf)[1] ="rich"
site<-rep("ONAQ",times=nrow(ONAQrichdf))
ONAQrichdffinal<-cbind(ONAQrichdf,site)

ORNLrichdf<-as.data.frame(ORNLrich)
colnames(ORNLrichdf)[1] ="rich"
site<-rep("ORNL",times=nrow(ORNLrichdf))
ORNLrichdffinal<-cbind(ORNLrichdf,site)

OSBSrichdf<-as.data.frame(OSBSrich)
colnames(OSBSrichdf)[1] ="rich"
site<-rep("OSBS",times=nrow(OSBSrichdf))
OSBSrichdffinal<-cbind(OSBSrichdf,site)

PUUMrichdf<-as.data.frame(PUUMrich)
colnames(PUUMrichdf)[1] ="rich"
site<-rep("PUUM",times=nrow(PUUMrichdf))
PUUMrichdffinal<-cbind(PUUMrichdf,site)

RMNPrichdf<-as.data.frame(RMNPrich)
colnames(RMNPrichdf)[1] ="rich"
site<-rep("RMNP",times=nrow(RMNPrichdf))
RMNPrichdffinal<-cbind(RMNPrichdf,site)

SCBIrichdf<-as.data.frame(SCBIrich)
colnames(SCBIrichdf)[1] ="rich"
site<-rep("SCBI",times=nrow(SCBIrichdf))
SCBIrichdffinal<-cbind(SCBIrichdf,site)

SERCrichdf<-as.data.frame(SERCrich)
colnames(SERCrichdf)[1] ="rich"
site<-rep("SERC",times=nrow(SERCrichdf))
SERCrichdffinal<-cbind(SERCrichdf,site)

SJERrichdf<-as.data.frame(SJERrich)
colnames(SJERrichdf)[1] ="rich"
site<-rep("SJER",times=nrow(SJERrichdf))
SJERrichdffinal<-cbind(SJERrichdf,site)

SOAPrichdf<-as.data.frame(SOAPrich)
colnames(SOAPrichdf)[1] ="rich"
site<-rep("SOAP",times=nrow(SOAPrichdf))
SOAPrichdffinal<-cbind(SOAPrichdf,site)

SRERrichdf<-as.data.frame(SRERrich)
colnames(SRERrichdf)[1] ="rich"
site<-rep("SRER",times=nrow(SRERrichdf))
SRERrichdffinal<-cbind(SRERrichdf,site)

STEIrichdf<-as.data.frame(STEIrich)
colnames(STEIrichdf)[1] ="rich"
site<-rep("STEI",times=nrow(STEIrichdf))
STEIrichdffinal<-cbind(STEIrichdf,site)

STERrichdf<-as.data.frame(STERrich)
colnames(STERrichdf)[1] ="rich"
site<-rep("STER",times=nrow(STERrichdf))
STERrichdffinal<-cbind(STERrichdf,site)

TALLrichdf<-as.data.frame(TALLrich)
colnames(TALLrichdf)[1] ="rich"
site<-rep("TALL",times=nrow(TALLrichdf))
TALLrichdffinal<-cbind(TALLrichdf,site)

TOOLrichdf<-as.data.frame(TOOLrich)
colnames(TOOLrichdf)[1] ="rich"
site<-rep("TOOL",times=nrow(TOOLrichdf))
TOOLrichdffinal<-cbind(TOOLrichdf,site)

#TREErichdf<-as.data.frame(TREErich)
#colnames(TREErichdf)[1] ="rich"
#site<-rep("TREE",times=nrow(TREErichdf))
#TREErichdffinal<-cbind(TREErichdf,site)

UKFSrichdf<-as.data.frame(UKFSrich)
colnames(UKFSrichdf)[1] ="rich"
site<-rep("UKFS",times=nrow(UKFSrichdf))
UKFSrichdffinal<-cbind(UKFSrichdf,site)

UNDErichdf<-as.data.frame(UNDErich)
colnames(UNDErichdf)[1] ="rich"
site<-rep("UNDE",times=nrow(UNDErichdf))
UNDErichdffinal<-cbind(UNDErichdf,site)

WOODrichdf<-as.data.frame(WOODrich)
colnames(WOODrichdf)[1] ="rich"
site<-rep("WOOD",times=nrow(WOODrichdf))
WOODrichdffinal<-cbind(WOODrichdf,site)

WREFrichdf<-as.data.frame(WREFrich)
colnames(WREFrichdf)[1] ="rich"
site<-rep("WREF",times=nrow(WREFrichdf))
WREFrichdffinal<-cbind(WREFrichdf,site)

YELLrichdf<-as.data.frame(YELLrich)
colnames(YELLrichdf)[1] ="rich"
site<-rep("YELL",times=nrow(YELLrichdf))
YELLrichdffinal<-cbind(YELLrichdf,site)
```

# Just with the managed sites next to eachother

```{r}
richfinalbind<-do.call("rbind",list(WREFrichdffinal,ABBYrichdffinal,STEIrichdffinal,KONZrichdffinal,KONArichdffinal,CPERrichdffinal,STERrichdffinal,WOODrichdffinal,DCFSrichdffinal,GUANrichdffinal,LAJArichdffinal,UNDErichdffinal))
```

```{r}
library(dplyr)
richfinalbind %>%
  group_by(site) %>%
  summarise(
    mean_rich = mean(rich),
    median_rich = median(rich),
    sd_rich = sd(rich)
  )
```

```{r}
positions <- c("CPER","STER","KONZ","KONA","WOOD","DCFS","GUAN","LAJA","WREF","ABBY","UNDE","STEI")
library(ggplot2)
speciesrichness<-ggplot(data=richfinalbind,aes(x=site,y=rich))+
 geom_boxplot()+
  scale_x_discrete(limits = positions)+
  geom_point(size=0.3)+
  coord_flip()+
  xlab("")+
  ylab("Richness (# of species)")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))

ggsave(filename = "speciesrichness_NEON_managed_by_MAP.pdf",
      plot = speciesrichness,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600) 
```

stats for this graph


```{r}
site <- c('STEI', 'UNDE', 'ABBY', 'WREF','LAJA','GUAN','DCFS','WOOD','KONA','KONZ','STER','CPER') 
management <- c('High', 'Low', 'High', 'Low','High', 'Low','High', 'Low','High', 'Low','High', 'Low')
site_type<- c('Forestry','Forestry','Forestry','Forestry','Cattle','Cattle','Cattle','Cattle','Cropland','Cropland','Cropland','Cropland')
stats<-data.frame(site,management,site_type)
merged_stats<-merge(stats,richfinalbind,by="site")
library(car)
model<-lm(data=merged_stats,rich~management)
model
anova(model)
lowman_mean<-subset(merged_stats,management=='Low')
mean(lowman_mean$rich)
highman_mean<-subset(merged_stats,management=='High')
mean(highman_mean$rich)
mean(lowman_mean$rich)-mean(highman_mean$rich)
100*(mean(lowman_mean$rich)-mean(highman_mean$rich))/mean(lowman_mean$rich)
```

```{r}
highman_mean_crop<-subset(highman_mean,site_type=='Cropland')
highman_mean_catt<-subset(highman_mean,site_type=='Cattle')
highman_mean_for<-subset(highman_mean,site_type=='Forestry')
mean(highman_mean_crop$rich)
mean(highman_mean_catt$rich)
mean(highman_mean_for$rich)
median(highman_mean_crop$rich)
median(highman_mean_catt$rich)
median(highman_mean_for$rich)
```


```{r}
library(reshape2)
ABBY_tax_sum_phylum = summarize_taxonomy(ABBY, level = 3, report_higher_tax = FALSE)
ABBY_tax_sum_phylum$Phylum <-row.names(ABBY_tax_sum_phylum)
ABBY_melt<- melt(ABBY_tax_sum_phylum, id.var="Phylum")
site<-rep("ABBY",times=nrow(ABBY_melt))
ABBY_melt<-cbind(ABBY_melt,site)

BARR_tax_sum_phylum = summarize_taxonomy(BARR, level = 3, report_higher_tax = FALSE)
BARR_tax_sum_phylum$Phylum <-row.names(BARR_tax_sum_phylum)
BARR_melt<- melt(BARR_tax_sum_phylum, id.var="Phylum")
nrow(BARR_melt)
site<-rep("BARR",times=nrow(BARR_melt))
BARR_melt<-cbind(BARR_melt,site)

BART_tax_sum_phylum = summarize_taxonomy(BART, level = 3, report_higher_tax = FALSE)
BART_tax_sum_phylum$Phylum <-row.names(BART_tax_sum_phylum)
BART_melt<- melt(BART_tax_sum_phylum, id.var="Phylum")
nrow(BART_melt)
site<-rep("BART",times=nrow(BART_melt))
BART_melt<-cbind(BART_melt,site)

BLAN_tax_sum_phylum = summarize_taxonomy(BLAN, level = 3, report_higher_tax = FALSE)
BLAN_tax_sum_phylum$Phylum <-row.names(BLAN_tax_sum_phylum)
BLAN_melt<- melt(BLAN_tax_sum_phylum, id.var="Phylum")
nrow(BLAN_melt)
site<-rep("BLAN",times=nrow(BLAN_melt))
BLAN_melt<-cbind(BLAN_melt,site)

BONA_tax_sum_phylum = summarize_taxonomy(BONA, level = 3, report_higher_tax = FALSE)
BONA_tax_sum_phylum$Phylum <-row.names(BONA_tax_sum_phylum)
BONA_melt<- melt(BONA_tax_sum_phylum, id.var="Phylum")
nrow(BONA_melt)
site<-rep("BONA",times=nrow(BONA_melt))
BONA_melt<-cbind(BONA_melt,site)

CLBJ_tax_sum_phylum = summarize_taxonomy(CLBJ, level = 3, report_higher_tax = FALSE)
CLBJ_tax_sum_phylum$Phylum <-row.names(CLBJ_tax_sum_phylum)
CLBJ_melt<- melt(CLBJ_tax_sum_phylum, id.var="Phylum")
nrow(CLBJ_melt)
site<-rep("CLBJ",times=nrow(CLBJ_melt))
CLBJ_melt<-cbind(CLBJ_melt,site)

CPER_tax_sum_phylum = summarize_taxonomy(CPER, level = 3, report_higher_tax = FALSE)
CPER_tax_sum_phylum$Phylum <-row.names(CPER_tax_sum_phylum)
CPER_melt<- melt(CPER_tax_sum_phylum, id.var="Phylum")
nrow(CPER_melt)
site<-rep("CPER",times=nrow(CPER_melt))
CPER_melt<-cbind(CPER_melt,site)

DCFS_tax_sum_phylum = summarize_taxonomy(DCFS, level = 3, report_higher_tax = FALSE)
DCFS_tax_sum_phylum$Phylum <-row.names(DCFS_tax_sum_phylum)
DCFS_melt<- melt(DCFS_tax_sum_phylum, id.var="Phylum")
nrow(DCFS_melt)
site<-rep("DCFS",times=nrow(DCFS_melt))
DCFS_melt<-cbind(DCFS_melt,site)

DEJU_tax_sum_phylum = summarize_taxonomy(DEJU, level = 3, report_higher_tax = FALSE)
DEJU_tax_sum_phylum$Phylum <-row.names(DEJU_tax_sum_phylum)
DEJU_melt<- melt(DEJU_tax_sum_phylum, id.var="Phylum")
nrow(DEJU_melt)
site<-rep("DEJU",times=nrow(DEJU_melt))
DEJU_melt<-cbind(DEJU_melt,site)

DELA_tax_sum_phylum = summarize_taxonomy(DELA, level = 3, report_higher_tax = FALSE)
DELA_tax_sum_phylum$Phylum <-row.names(DELA_tax_sum_phylum)
DELA_melt<- melt(DELA_tax_sum_phylum, id.var="Phylum")
nrow(DELA_melt)
site<-rep("DELA",times=nrow(DELA_melt))
DELA_melt<-cbind(DELA_melt,site)

DSNY_tax_sum_phylum = summarize_taxonomy(DSNY, level = 3, report_higher_tax = FALSE)
DSNY_tax_sum_phylum$Phylum <-row.names(DSNY_tax_sum_phylum)
DSNY_melt<- melt(DSNY_tax_sum_phylum, id.var="Phylum")
nrow(DSNY_melt)
site<-rep("DSNY",times=nrow(DSNY_melt))
DSNY_melt<-cbind(DSNY_melt,site)

GRSM_tax_sum_phylum = summarize_taxonomy(GRSM, level = 3, report_higher_tax = FALSE)
GRSM_tax_sum_phylum$Phylum <-row.names(GRSM_tax_sum_phylum)
GRSM_melt<- melt(GRSM_tax_sum_phylum, id.var="Phylum")
nrow(GRSM_melt)
site<-rep("GRSM",times=nrow(GRSM_melt))
GRSM_melt<-cbind(GRSM_melt,site)

GUAN_tax_sum_phylum = summarize_taxonomy(GUAN, level = 3, report_higher_tax = FALSE)
GUAN_tax_sum_phylum$Phylum <-row.names(GUAN_tax_sum_phylum)
GUAN_melt<- melt(GUAN_tax_sum_phylum, id.var="Phylum")
nrow(GUAN_melt)
site<-rep("GUAN",times=nrow(GUAN_melt))
GUAN_melt<-cbind(GUAN_melt,site)

HARV_tax_sum_phylum = summarize_taxonomy(HARV, level = 3, report_higher_tax = FALSE)
HARV_tax_sum_phylum$Phylum <-row.names(HARV_tax_sum_phylum)
HARV_melt<- melt(HARV_tax_sum_phylum, id.var="Phylum")
nrow(HARV_melt)
site<-rep("HARV",times=nrow(HARV_melt))
HARV_melt<-cbind(HARV_melt,site)

HEAL_tax_sum_phylum = summarize_taxonomy(HEAL, level = 3, report_higher_tax = FALSE)
HEAL_tax_sum_phylum$Phylum <-row.names(HEAL_tax_sum_phylum)
HEAL_melt<- melt(HEAL_tax_sum_phylum, id.var="Phylum")
nrow(HEAL_melt)
site<-rep("HEAL",times=nrow(HEAL_melt))
HEAL_melt<-cbind(HEAL_melt,site)

JERC_tax_sum_phylum = summarize_taxonomy(JERC, level = 3, report_higher_tax = FALSE)
JERC_tax_sum_phylum$Phylum <-row.names(JERC_tax_sum_phylum)
JERC_melt<- melt(JERC_tax_sum_phylum, id.var="Phylum")
nrow(JERC_melt)
site<-rep("JERC",times=nrow(JERC_melt))
JERC_melt<-cbind(JERC_melt,site)

JORN_tax_sum_phylum = summarize_taxonomy(JORN, level = 3, report_higher_tax = FALSE)
JORN_tax_sum_phylum$Phylum <-row.names(JORN_tax_sum_phylum)
JORN_melt<- melt(JORN_tax_sum_phylum, id.var="Phylum")
nrow(JORN_melt)
site<-rep("JORN",times=nrow(JORN_melt))
JORN_melt<-cbind(JORN_melt,site)

KONA_tax_sum_phylum = summarize_taxonomy(KONA, level = 3, report_higher_tax = FALSE)
KONA_tax_sum_phylum$Phylum <-row.names(KONA_tax_sum_phylum)
KONA_melt<- melt(KONA_tax_sum_phylum, id.var="Phylum")
nrow(KONA_melt)
site<-rep("KONA",times=nrow(KONA_melt))
KONA_melt<-cbind(KONA_melt,site)

KONZ_tax_sum_phylum = summarize_taxonomy(KONZ, level = 3, report_higher_tax = FALSE)
KONZ_tax_sum_phylum$Phylum <-row.names(KONZ_tax_sum_phylum)
KONZ_melt<- melt(KONZ_tax_sum_phylum, id.var="Phylum")
nrow(KONZ_melt)
site<-rep("KONZ",times=nrow(KONZ_melt))
KONZ_melt<-cbind(KONZ_melt,site)

LAJA_tax_sum_phylum = summarize_taxonomy(LAJA, level = 3, report_higher_tax = FALSE)
LAJA_tax_sum_phylum$Phylum <-row.names(LAJA_tax_sum_phylum)
LAJA_melt<- melt(LAJA_tax_sum_phylum, id.var="Phylum")
nrow(LAJA_melt)
site<-rep("LAJA",times=nrow(LAJA_melt))
LAJA_melt<-cbind(LAJA_melt,site)

LENO_tax_sum_phylum = summarize_taxonomy(LENO, level = 3, report_higher_tax = FALSE)
LENO_tax_sum_phylum$Phylum <-row.names(LENO_tax_sum_phylum)
LENO_melt<- melt(LENO_tax_sum_phylum, id.var="Phylum")
nrow(LENO_melt)
site<-rep("LENO",times=nrow(LENO_melt))
LENO_melt<-cbind(LENO_melt,site)

MLBS_tax_sum_phylum = summarize_taxonomy(MLBS, level = 3, report_higher_tax = FALSE)
MLBS_tax_sum_phylum$Phylum <-row.names(MLBS_tax_sum_phylum)
MLBS_melt<- melt(MLBS_tax_sum_phylum, id.var="Phylum")
nrow(MLBS_melt)
site<-rep("MLBS",times=nrow(MLBS_melt))
MLBS_melt<-cbind(MLBS_melt,site)

MOAB_tax_sum_phylum = summarize_taxonomy(MOAB, level = 3, report_higher_tax = FALSE)
MOAB_tax_sum_phylum$Phylum <-row.names(MOAB_tax_sum_phylum)
MOAB_melt<- melt(MOAB_tax_sum_phylum, id.var="Phylum")
nrow(MOAB_melt)
site<-rep("MOAB",times=nrow(MOAB_melt))
MOAB_melt<-cbind(MOAB_melt,site)

NIWO_tax_sum_phylum = summarize_taxonomy(NIWO, level = 3, report_higher_tax = FALSE)
NIWO_tax_sum_phylum$Phylum <-row.names(NIWO_tax_sum_phylum)
NIWO_melt<- melt(NIWO_tax_sum_phylum, id.var="Phylum")
nrow(NIWO_melt)
site<-rep("NIWO",times=nrow(NIWO_melt))
NIWO_melt<-cbind(NIWO_melt,site)

OAES_tax_sum_phylum = summarize_taxonomy(OAES, level = 3, report_higher_tax = FALSE)
OAES_tax_sum_phylum$Phylum <-row.names(OAES_tax_sum_phylum)
OAES_melt<- melt(OAES_tax_sum_phylum, id.var="Phylum")
nrow(OAES_melt)
site<-rep("OAES",times=nrow(OAES_melt))
OAES_melt<-cbind(OAES_melt,site)

ONAQ_tax_sum_phylum = summarize_taxonomy(ONAQ, level = 3, report_higher_tax = FALSE)
ONAQ_tax_sum_phylum$Phylum <-row.names(ONAQ_tax_sum_phylum)
ONAQ_melt<- melt(ONAQ_tax_sum_phylum, id.var="Phylum")
nrow(ONAQ_melt)
site<-rep("ONAQ",times=nrow(ONAQ_melt))
ONAQ_melt<-cbind(ONAQ_melt,site)

ORNL_tax_sum_phylum = summarize_taxonomy(ORNL, level = 3, report_higher_tax = FALSE)
ORNL_tax_sum_phylum$Phylum <-row.names(ORNL_tax_sum_phylum)
ORNL_melt<- melt(ORNL_tax_sum_phylum, id.var="Phylum")
nrow(ORNL_melt)
site<-rep("ORNL",times=nrow(ORNL_melt))
ORNL_melt<-cbind(ORNL_melt,site)

OSBS_tax_sum_phylum = summarize_taxonomy(OSBS, level = 3, report_higher_tax = FALSE)
OSBS_tax_sum_phylum$Phylum <-row.names(OSBS_tax_sum_phylum)
OSBS_melt<- melt(OSBS_tax_sum_phylum, id.var="Phylum")
nrow(OSBS_melt)
site<-rep("OSBS",times=nrow(OSBS_melt))
OSBS_melt<-cbind(OSBS_melt,site)

PUUM_tax_sum_phylum = summarize_taxonomy(PUUM, level = 3, report_higher_tax = FALSE)
PUUM_tax_sum_phylum$Phylum <-row.names(PUUM_tax_sum_phylum)
PUUM_melt<- melt(PUUM_tax_sum_phylum, id.var="Phylum")
nrow(PUUM_melt)
site<-rep("PUUM",times=nrow(PUUM_melt))
PUUM_melt<-cbind(PUUM_melt,site)

RMNP_tax_sum_phylum = summarize_taxonomy(RMNP, level = 3, report_higher_tax = FALSE)
RMNP_tax_sum_phylum$Phylum <-row.names(RMNP_tax_sum_phylum)
RMNP_melt<- melt(RMNP_tax_sum_phylum, id.var="Phylum")
nrow(RMNP_melt)
site<-rep("RMNP",times=nrow(RMNP_melt))
RMNP_melt<-cbind(RMNP_melt,site)

SCBI_tax_sum_phylum = summarize_taxonomy(SCBI, level = 3, report_higher_tax = FALSE)
SCBI_tax_sum_phylum$Phylum <-row.names(SCBI_tax_sum_phylum)
SCBI_melt<- melt(SCBI_tax_sum_phylum, id.var="Phylum")
nrow(SCBI_melt)
site<-rep("SCBI",times=nrow(SCBI_melt))
SCBI_melt<-cbind(SCBI_melt,site)

SERC_tax_sum_phylum = summarize_taxonomy(SERC, level = 3, report_higher_tax = FALSE)
SERC_tax_sum_phylum$Phylum <-row.names(SERC_tax_sum_phylum)
SERC_melt<- melt(SERC_tax_sum_phylum, id.var="Phylum")
nrow(SERC_melt)
site<-rep("SERC",times=nrow(SERC_melt))
SERC_melt<-cbind(SERC_melt,site)

SJER_tax_sum_phylum = summarize_taxonomy(SJER, level = 3, report_higher_tax = FALSE)
SJER_tax_sum_phylum$Phylum <-row.names(SJER_tax_sum_phylum)
SJER_melt<- melt(SJER_tax_sum_phylum, id.var="Phylum")
nrow(SJER_melt)
site<-rep("SJER",times=nrow(SJER_melt))
SJER_melt<-cbind(SJER_melt,site)

SOAP_tax_sum_phylum = summarize_taxonomy(SOAP, level = 3, report_higher_tax = FALSE)
SOAP_tax_sum_phylum$Phylum <-row.names(SOAP_tax_sum_phylum)
SOAP_melt<- melt(SOAP_tax_sum_phylum, id.var="Phylum")
nrow(SOAP_melt)
site<-rep("SOAP",times=nrow(SOAP_melt))
SOAP_melt<-cbind(SOAP_melt,site)

SRER_tax_sum_phylum = summarize_taxonomy(SRER, level = 3, report_higher_tax = FALSE)
SRER_tax_sum_phylum$Phylum <-row.names(SRER_tax_sum_phylum)
SRER_melt<- melt(SRER_tax_sum_phylum, id.var="Phylum")
nrow(SRER_melt)
site<-rep("SRER",times=nrow(SRER_melt))
SRER_melt<-cbind(SRER_melt,site)

STEI_tax_sum_phylum = summarize_taxonomy(STEI, level = 3, report_higher_tax = FALSE)
STEI_tax_sum_phylum$Phylum <-row.names(STEI_tax_sum_phylum)
STEI_melt<- melt(STEI_tax_sum_phylum, id.var="Phylum")
nrow(STEI_melt)
site<-rep("STEI",times=nrow(STEI_melt))
STEI_melt<-cbind(STEI_melt,site)

STER_tax_sum_phylum = summarize_taxonomy(STER, level = 3, report_higher_tax = FALSE)
STER_tax_sum_phylum$Phylum <-row.names(STER_tax_sum_phylum)
STER_melt<- melt(STER_tax_sum_phylum, id.var="Phylum")
nrow(STER_melt)
site<-rep("STER",times=nrow(STER_melt))
STER_melt<-cbind(STER_melt,site)

TALL_tax_sum_phylum = summarize_taxonomy(TALL, level = 3, report_higher_tax = FALSE)
TALL_tax_sum_phylum$Phylum <-row.names(TALL_tax_sum_phylum)
TALL_melt<- melt(TALL_tax_sum_phylum, id.var="Phylum")
nrow(TALL_melt)
site<-rep("TALL",times=nrow(TALL_melt))
TALL_melt<-cbind(TALL_melt,site)

TOOL_tax_sum_phylum = summarize_taxonomy(TOOL, level = 3, report_higher_tax = FALSE)
TOOL_tax_sum_phylum$Phylum <-row.names(TOOL_tax_sum_phylum)
TOOL_melt<- melt(TOOL_tax_sum_phylum, id.var="Phylum")
nrow(TOOL_melt)
site<-rep("TOOL",times=nrow(TOOL_melt))
TOOL_melt<-cbind(TOOL_melt,site)

TREE_tax_sum_phylum = summarize_taxonomy(TREE, level = 3, report_higher_tax = FALSE)
TREE_tax_sum_phylum$Phylum <-row.names(TREE_tax_sum_phylum)
TREE_melt<- melt(TREE_tax_sum_phylum, id.var="Phylum")
nrow(TREE_melt)
site<-rep("TREE",times=nrow(TREE_melt))
TREE_melt<-cbind(TREE_melt,site)

UKFS_tax_sum_phylum = summarize_taxonomy(UKFS, level = 3, report_higher_tax = FALSE)
UKFS_tax_sum_phylum$Phylum <-row.names(UKFS_tax_sum_phylum)
UKFS_melt<- melt(UKFS_tax_sum_phylum, id.var="Phylum")
nrow(UKFS_melt)
site<-rep("UKFS",times=nrow(UKFS_melt))
UKFS_melt<-cbind(UKFS_melt,site)

UNDE_tax_sum_phylum = summarize_taxonomy(UNDE, level = 3, report_higher_tax = FALSE)
UNDE_tax_sum_phylum$Phylum <-row.names(UNDE_tax_sum_phylum)
UNDE_melt<- melt(UNDE_tax_sum_phylum, id.var="Phylum")
nrow(UNDE_melt)
site<-rep("UNDE",times=nrow(UNDE_melt))
UNDE_melt<-cbind(UNDE_melt,site)

WOOD_tax_sum_phylum = summarize_taxonomy(WOOD, level = 3, report_higher_tax = FALSE)
WOOD_tax_sum_phylum$Phylum <-row.names(WOOD_tax_sum_phylum)
WOOD_melt<- melt(WOOD_tax_sum_phylum, id.var="Phylum")
nrow(WOOD_melt)
site<-rep("WOOD",times=nrow(WOOD_melt))
WOOD_melt<-cbind(WOOD_melt,site)

WREF_tax_sum_phylum = summarize_taxonomy(WREF, level = 3, report_higher_tax = FALSE)
WREF_tax_sum_phylum$Phylum <-row.names(WREF_tax_sum_phylum)
WREF_melt<- melt(WREF_tax_sum_phylum, id.var="Phylum")
nrow(WREF_melt)
site<-rep("WREF",times=nrow(WREF_melt))
WREF_melt<-cbind(WREF_melt,site)

YELL_tax_sum_phylum = summarize_taxonomy(YELL, level = 3, report_higher_tax = FALSE)
YELL_tax_sum_phylum$Phylum <-row.names(YELL_tax_sum_phylum)
YELL_melt<- melt(YELL_tax_sum_phylum, id.var="Phylum")
nrow(YELL_melt)
site<-rep("YELL",times=nrow(YELL_melt))
YELL_melt<-cbind(YELL_melt,site)
```

```{r}
phylumfinalbind<-do.call("rbind",list(ABBY_melt, BARR_melt, BART_melt, BLAN_melt, BONA_melt, CLBJ_melt, CPER_melt, DCFS_melt, DEJU_melt, DELA_melt, DSNY_melt, GRSM_melt, GUAN_melt, HARV_melt, HEAL_melt, JERC_melt, JORN_melt, KONA_melt, KONZ_melt, LAJA_melt, LENO_melt, MLBS_melt, MOAB_melt, NIWO_melt, OAES_melt, ONAQ_melt, ORNL_melt, OSBS_melt, PUUM_melt, RMNP_melt, SCBI_melt, SERC_melt, SJER_melt, SOAP_melt, SRER_melt, STEI_melt, STER_melt, TALL_melt, TOOL_melt, TREE_melt, UKFS_melt, UNDE_melt, WOOD_melt, WREF_melt, YELL_melt))
```

phylum graph
```{r}

b<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Phylum Graphs")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

b
```

```{r}

ggsave(filename = "phylum_bysite.pdf",
      plot = b,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```



```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input = filter_data(input, 'nlcdClass', filter_vals = '0')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
cultivatedCrops = filter_data(input, 'nlcdClass', keep_vals = 'cultivatedCrops')
deciduousForest = filter_data(input, 'nlcdClass', keep_vals = 'deciduousForest')
dwarfScrub = filter_data(input, 'nlcdClass', keep_vals = 'dwarfScrub')
emergentHerbaceousWetlands = filter_data(input, 'nlcdClass', keep_vals = 'emergentHerbaceousWetlands')
evergreenForest = filter_data(input, 'nlcdClass', keep_vals = 'evergreenForest')
grasslandHerbaceous = filter_data(input, 'nlcdClass', keep_vals = 'grasslandHerbaceous')
mixedForest = filter_data(input, 'nlcdClass', keep_vals = 'mixedForest')
pastureHay = filter_data(input, 'nlcdClass', keep_vals = 'pastureHay')
sedgeHerbaceous = filter_data(input, 'nlcdClass', keep_vals = 'sedgeHerbaceous')
shrubScrub = filter_data(input, 'nlcdClass', keep_vals = 'shrubScrub')
woodyWetlands = filter_data(input, 'nlcdClass', keep_vals = 'woodyWetlands')
```

```{r}
otuTAB<-cultivatedCrops$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
cultivatedCrops$data_loaded<-t(hell)

otuTAB<-deciduousForest$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
deciduousForest$data_loaded<-t(hell)

otuTAB<-dwarfScrub$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
dwarfScrub$data_loaded<-t(hell)

otuTAB<-emergentHerbaceousWetlands$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
emergentHerbaceousWetlands$data_loaded<-t(hell)

otuTAB<-evergreenForest$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
evergreenForest$data_loaded<-t(hell)

otuTAB<-grasslandHerbaceous$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
grasslandHerbaceous$data_loaded<-t(hell)

otuTAB<-mixedForest$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
mixedForest$data_loaded<-t(hell)

otuTAB<-pastureHay$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
pastureHay$data_loaded<-t(hell)

otuTAB<-sedgeHerbaceous$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
sedgeHerbaceous$data_loaded<-t(hell)

otuTAB<-shrubScrub$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
shrubScrub$data_loaded<-t(hell)

otuTAB<-woodyWetlands$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
woodyWetlands$data_loaded<-t(hell)
```

```{r}
cultivatedCrops_tax_sum_phylum = summarize_taxonomy(cultivatedCrops, level = 3, report_higher_tax = FALSE)
cultivatedCrops_tax_sum_phylum$Phylum <-row.names(cultivatedCrops_tax_sum_phylum)
cultivatedCrops_melt<- melt(cultivatedCrops_tax_sum_phylum, id.var="Phylum")
nrow(cultivatedCrops_melt)
site<-rep("cultivatedCrops",times=nrow(cultivatedCrops_melt))
cultivatedCrops_melt<-cbind(cultivatedCrops_melt,site)

deciduousForest_tax_sum_phylum = summarize_taxonomy(deciduousForest, level = 3, report_higher_tax = FALSE)
deciduousForest_tax_sum_phylum$Phylum <-row.names(deciduousForest_tax_sum_phylum)
deciduousForest_melt<- melt(deciduousForest_tax_sum_phylum, id.var="Phylum")
nrow(deciduousForest_melt)
site<-rep("deciduousForest",times=nrow(deciduousForest_melt))
deciduousForest_melt<-cbind(deciduousForest_melt,site)

dwarfScrub_tax_sum_phylum = summarize_taxonomy(dwarfScrub, level = 3, report_higher_tax = FALSE)
dwarfScrub_tax_sum_phylum$Phylum <-row.names(dwarfScrub_tax_sum_phylum)
dwarfScrub_melt<- melt(dwarfScrub_tax_sum_phylum, id.var="Phylum")
nrow(dwarfScrub_melt)
site<-rep("dwarfScrub",times=nrow(dwarfScrub_melt))
dwarfScrub_melt<-cbind(dwarfScrub_melt,site)

emergentHerbaceousWetlands_tax_sum_phylum = summarize_taxonomy(emergentHerbaceousWetlands, level = 3, report_higher_tax = FALSE)
emergentHerbaceousWetlands_tax_sum_phylum$Phylum <-row.names(emergentHerbaceousWetlands_tax_sum_phylum)
emergentHerbaceousWetlands_melt<- melt(emergentHerbaceousWetlands_tax_sum_phylum, id.var="Phylum")
nrow(emergentHerbaceousWetlands_melt)
site<-rep("emergentHerbaceousWetlands",times=nrow(emergentHerbaceousWetlands_melt))
emergentHerbaceousWetlands_melt<-cbind(emergentHerbaceousWetlands_melt,site)

evergreenForest_tax_sum_phylum = summarize_taxonomy(evergreenForest, level = 3, report_higher_tax = FALSE)
evergreenForest_tax_sum_phylum$Phylum <-row.names(evergreenForest_tax_sum_phylum)
evergreenForest_melt<- melt(evergreenForest_tax_sum_phylum, id.var="Phylum")
nrow(evergreenForest_melt)
site<-rep("evergreenForest",times=nrow(evergreenForest_melt))
evergreenForest_melt<-cbind(evergreenForest_melt,site)

grasslandHerbaceous_tax_sum_phylum = summarize_taxonomy(grasslandHerbaceous, level = 3, report_higher_tax = FALSE)
grasslandHerbaceous_tax_sum_phylum$Phylum <-row.names(grasslandHerbaceous_tax_sum_phylum)
grasslandHerbaceous_melt<- melt(grasslandHerbaceous_tax_sum_phylum, id.var="Phylum")
nrow(grasslandHerbaceous_melt)
site<-rep("grasslandHerbaceous",times=nrow(grasslandHerbaceous_melt))
grasslandHerbaceous_melt<-cbind(grasslandHerbaceous_melt,site)

mixedForest_tax_sum_phylum = summarize_taxonomy(mixedForest, level = 3, report_higher_tax = FALSE)
mixedForest_tax_sum_phylum$Phylum <-row.names(mixedForest_tax_sum_phylum)
mixedForest_melt<- melt(mixedForest_tax_sum_phylum, id.var="Phylum")
nrow(mixedForest_melt)
site<-rep("mixedForest",times=nrow(mixedForest_melt))
mixedForest_melt<-cbind(mixedForest_melt,site)

pastureHay_tax_sum_phylum = summarize_taxonomy(pastureHay, level = 3, report_higher_tax = FALSE)
pastureHay_tax_sum_phylum$Phylum <-row.names(pastureHay_tax_sum_phylum)
pastureHay_melt<- melt(pastureHay_tax_sum_phylum, id.var="Phylum")
nrow(pastureHay_melt)
site<-rep("pastureHay",times=nrow(pastureHay_melt))
pastureHay_melt<-cbind(pastureHay_melt,site)

sedgeHerbaceous_tax_sum_phylum = summarize_taxonomy(sedgeHerbaceous, level = 3, report_higher_tax = FALSE)
sedgeHerbaceous_tax_sum_phylum$Phylum <-row.names(sedgeHerbaceous_tax_sum_phylum)
sedgeHerbaceous_melt<- melt(sedgeHerbaceous_tax_sum_phylum, id.var="Phylum")
nrow(sedgeHerbaceous_melt)
site<-rep("sedgeHerbaceous",times=nrow(sedgeHerbaceous_melt))
sedgeHerbaceous_melt<-cbind(sedgeHerbaceous_melt,site)

shrubScrub_tax_sum_phylum = summarize_taxonomy(shrubScrub, level = 3, report_higher_tax = FALSE)
shrubScrub_tax_sum_phylum$Phylum <-row.names(shrubScrub_tax_sum_phylum)
shrubScrub_melt<- melt(shrubScrub_tax_sum_phylum, id.var="Phylum")
nrow(shrubScrub_melt)
site<-rep("shrubScrub",times=nrow(shrubScrub_melt))
shrubScrub_melt<-cbind(shrubScrub_melt,site)

woodyWetlands_tax_sum_phylum = summarize_taxonomy(woodyWetlands, level = 3, report_higher_tax = FALSE)
woodyWetlands_tax_sum_phylum$Phylum <-row.names(woodyWetlands_tax_sum_phylum)
woodyWetlands_melt<- melt(woodyWetlands_tax_sum_phylum, id.var="Phylum")
nrow(woodyWetlands_melt)
site<-rep("woodyWetlands",times=nrow(woodyWetlands_melt))
woodyWetlands_melt<-cbind(woodyWetlands_melt,site)
```

```{r}
phylumbiomebind<-do.call("rbind",list(cultivatedCrops_melt,deciduousForest_melt,dwarfScrub_melt,emergentHerbaceousWetlands_melt,evergreenForest_melt,grasslandHerbaceous_melt,mixedForest_melt,pastureHay_melt,sedgeHerbaceous_melt,shrubScrub_melt,woodyWetlands_melt))
```

phylum graph
```{r}

c<-ggplot(phylumbiomebind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Phylum Graphs by Biome")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "phylum_bybiome.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```


```{r}
cultivatedCrops_tax_sum_phylum = summarize_taxonomy(cultivatedCrops, level = 2, report_higher_tax = FALSE)
cultivatedCrops_tax_sum_phylum$Phylum <-row.names(cultivatedCrops_tax_sum_phylum)
cultivatedCrops_melt<- melt(cultivatedCrops_tax_sum_phylum, id.var="Phylum")
nrow(cultivatedCrops_melt)
site<-rep("cultivatedCrops",times=nrow(cultivatedCrops_melt))
cultivatedCrops_melt<-cbind(cultivatedCrops_melt,site)

deciduousForest_tax_sum_phylum = summarize_taxonomy(deciduousForest, level = 2, report_higher_tax = FALSE)
deciduousForest_tax_sum_phylum$Phylum <-row.names(deciduousForest_tax_sum_phylum)
deciduousForest_melt<- melt(deciduousForest_tax_sum_phylum, id.var="Phylum")
nrow(deciduousForest_melt)
site<-rep("deciduousForest",times=nrow(deciduousForest_melt))
deciduousForest_melt<-cbind(deciduousForest_melt,site)

dwarfScrub_tax_sum_phylum = summarize_taxonomy(dwarfScrub, level = 2, report_higher_tax = FALSE)
dwarfScrub_tax_sum_phylum$Phylum <-row.names(dwarfScrub_tax_sum_phylum)
dwarfScrub_melt<- melt(dwarfScrub_tax_sum_phylum, id.var="Phylum")
nrow(dwarfScrub_melt)
site<-rep("dwarfScrub",times=nrow(dwarfScrub_melt))
dwarfScrub_melt<-cbind(dwarfScrub_melt,site)

emergentHerbaceousWetlands_tax_sum_phylum = summarize_taxonomy(emergentHerbaceousWetlands, level = 2, report_higher_tax = FALSE)
emergentHerbaceousWetlands_tax_sum_phylum$Phylum <-row.names(emergentHerbaceousWetlands_tax_sum_phylum)
emergentHerbaceousWetlands_melt<- melt(emergentHerbaceousWetlands_tax_sum_phylum, id.var="Phylum")
nrow(emergentHerbaceousWetlands_melt)
site<-rep("emergentHerbaceousWetlands",times=nrow(emergentHerbaceousWetlands_melt))
emergentHerbaceousWetlands_melt<-cbind(emergentHerbaceousWetlands_melt,site)

evergreenForest_tax_sum_phylum = summarize_taxonomy(evergreenForest, level = 2, report_higher_tax = FALSE)
evergreenForest_tax_sum_phylum$Phylum <-row.names(evergreenForest_tax_sum_phylum)
evergreenForest_melt<- melt(evergreenForest_tax_sum_phylum, id.var="Phylum")
nrow(evergreenForest_melt)
site<-rep("evergreenForest",times=nrow(evergreenForest_melt))
evergreenForest_melt<-cbind(evergreenForest_melt,site)

grasslandHerbaceous_tax_sum_phylum = summarize_taxonomy(grasslandHerbaceous, level = 2, report_higher_tax = FALSE)
grasslandHerbaceous_tax_sum_phylum$Phylum <-row.names(grasslandHerbaceous_tax_sum_phylum)
grasslandHerbaceous_melt<- melt(grasslandHerbaceous_tax_sum_phylum, id.var="Phylum")
nrow(grasslandHerbaceous_melt)
site<-rep("grasslandHerbaceous",times=nrow(grasslandHerbaceous_melt))
grasslandHerbaceous_melt<-cbind(grasslandHerbaceous_melt,site)

mixedForest_tax_sum_phylum = summarize_taxonomy(mixedForest, level = 2, report_higher_tax = FALSE)
mixedForest_tax_sum_phylum$Phylum <-row.names(mixedForest_tax_sum_phylum)
mixedForest_melt<- melt(mixedForest_tax_sum_phylum, id.var="Phylum")
nrow(mixedForest_melt)
site<-rep("mixedForest",times=nrow(mixedForest_melt))
mixedForest_melt<-cbind(mixedForest_melt,site)

pastureHay_tax_sum_phylum = summarize_taxonomy(pastureHay, level = 2, report_higher_tax = FALSE)
pastureHay_tax_sum_phylum$Phylum <-row.names(pastureHay_tax_sum_phylum)
pastureHay_melt<- melt(pastureHay_tax_sum_phylum, id.var="Phylum")
nrow(pastureHay_melt)
site<-rep("pastureHay",times=nrow(pastureHay_melt))
pastureHay_melt<-cbind(pastureHay_melt,site)

sedgeHerbaceous_tax_sum_phylum = summarize_taxonomy(sedgeHerbaceous, level = 2, report_higher_tax = FALSE)
sedgeHerbaceous_tax_sum_phylum$Phylum <-row.names(sedgeHerbaceous_tax_sum_phylum)
sedgeHerbaceous_melt<- melt(sedgeHerbaceous_tax_sum_phylum, id.var="Phylum")
nrow(sedgeHerbaceous_melt)
site<-rep("sedgeHerbaceous",times=nrow(sedgeHerbaceous_melt))
sedgeHerbaceous_melt<-cbind(sedgeHerbaceous_melt,site)

shrubScrub_tax_sum_phylum = summarize_taxonomy(shrubScrub, level = 2, report_higher_tax = FALSE)
shrubScrub_tax_sum_phylum$Phylum <-row.names(shrubScrub_tax_sum_phylum)
shrubScrub_melt<- melt(shrubScrub_tax_sum_phylum, id.var="Phylum")
nrow(shrubScrub_melt)
site<-rep("shrubScrub",times=nrow(shrubScrub_melt))
shrubScrub_melt<-cbind(shrubScrub_melt,site)

woodyWetlands_tax_sum_phylum = summarize_taxonomy(woodyWetlands, level = 2, report_higher_tax = FALSE)
woodyWetlands_tax_sum_phylum$Phylum <-row.names(woodyWetlands_tax_sum_phylum)
woodyWetlands_melt<- melt(woodyWetlands_tax_sum_phylum, id.var="Phylum")
nrow(woodyWetlands_melt)
site<-rep("woodyWetlands",times=nrow(woodyWetlands_melt))
woodyWetlands_melt<-cbind(woodyWetlands_melt,site)
```

```{r}
supergroupbiomebind<-do.call("rbind",list(cultivatedCrops_melt,deciduousForest_melt,dwarfScrub_melt,emergentHerbaceousWetlands_melt,evergreenForest_melt,grasslandHerbaceous_melt,mixedForest_melt,pastureHay_melt,sedgeHerbaceous_melt,shrubScrub_melt,woodyWetlands_melt))
```

phylum graph
```{r}

p<-ggplot(supergroupbiomebind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("SuperGroup Graphs by Biome")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

p
```

```{r}

ggsave(filename = "supergroup_bybiome_.pdf",
      plot = p,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

#3. Basic Stats
  #3A. Count sampling events per site 
  #3B. Count Unique OTUs and Reads per Group and Site 
#4. Sampling Depth Estimation
  #4A. Rarefaction Curves - done
  #4B. Species Accumulation Curves

```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
```

Reads per sample
```{r}
library(dplyr)
reads_sample<-rowSums(input$data_loaded)
reads_sample<-as.data.frame(reads_sample)
reads_sample<-reads_sample %>% rename(value=reads_sample)
mean(reads_sample$value)
reads_sample <-reads_sample[order(reads_sample$value),]
reads_sample<-as.data.frame(reads_sample)
reads_sample<-reads_sample %>% rename(value=reads_sample)
mean(reads_sample$value)

```
```{r}
library(janitor)
reads_sample <- tibble::rownames_to_column(reads_sample, "order")
g<-ggplot(data=reads_sample,aes(x=order,y=value)) + 
  geom_point()+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2)) 
ggsave(filename = "Reads per sample.pdf",
      plot = g,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

Reads per sample (1 removed)

```{r}
inputminusone<-as.data.frame(input$data_loaded)
inputminusone_filtered <- inputminusone[which(rowSums(inputminusone)>=2), ]
reads_sample_minusone<-rowSums(inputminusone_filtered)
reads_sample_minusone<-as.data.frame(reads_sample_minusone)
reads_sample_minusone<-reads_sample_minusone %>% rename(value=reads_sample_minusone)
reads_sample_minusone <-reads_sample_minusone[order(reads_sample_minusone$value),]
reads_sample_minusone<-as.data.frame(reads_sample_minusone)
reads_sample_minusone<-reads_sample_minusone %>% rename(value=reads_sample_minusone)
mean(reads_sample_minusone$value)
```

```{r}
library(janitor)
reads_sample_minusone <- tibble::rownames_to_column(reads_sample_minusone, "order")
h<-ggplot(data=reads_sample_minusone,aes(x=order,y=value)) + 
  geom_point()+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2)) 
ggsave(filename = "Reads per sample_oneremoved.pdf",
      plot = h,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

Reads per Site + OTU by Site
Filter by site
```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
MLBS = filter_data(input, 'Site', keep_vals = 'MLBS')
HARV = filter_data(input, 'Site', keep_vals = 'HARV')
WOOD = filter_data(input, 'Site', keep_vals = 'WOOD')
LAJA = filter_data(input, 'Site', keep_vals = 'LAJA')
UNDE = filter_data(input, 'Site', keep_vals = 'UNDE')
SRER = filter_data(input, 'Site', keep_vals = 'SRER')
DSNY = filter_data(input, 'Site', keep_vals = 'DSNY')
BONA = filter_data(input, 'Site', keep_vals = 'BONA')
GUAN = filter_data(input, 'Site', keep_vals = 'GUAN')
ABBY = filter_data(input, 'Site', keep_vals = 'ABBY')
BARR = filter_data(input, 'Site', keep_vals = 'BARR')
BART = filter_data(input, 'Site', keep_vals = 'BART')
BLAN = filter_data(input, 'Site', keep_vals = 'BLAN')
CLBJ = filter_data(input, 'Site', keep_vals = 'CLBJ')
CPER = filter_data(input, 'Site', keep_vals = 'CPER')
DCFS = filter_data(input, 'Site', keep_vals = 'DCFS')
DEJU = filter_data(input, 'Site', keep_vals = 'DEJU')
DELA = filter_data(input, 'Site', keep_vals = 'DELA')
GRSM = filter_data(input, 'Site', keep_vals = 'GRSM')
HEAL = filter_data(input, 'Site', keep_vals = 'HEAL')
JERC = filter_data(input, 'Site', keep_vals = 'JERC')
JORN = filter_data(input, 'Site', keep_vals = 'JORN')
KONA = filter_data(input, 'Site', keep_vals = 'KONA')
KONZ = filter_data(input, 'Site', keep_vals = 'KONZ')
LENO = filter_data(input, 'Site', keep_vals = 'LENO')
MOAB = filter_data(input, 'Site', keep_vals = 'MOAB')
NIWO = filter_data(input, 'Site', keep_vals = 'NIWO')
OAES = filter_data(input, 'Site', keep_vals = 'OAES')
ONAQ = filter_data(input, 'Site', keep_vals = 'ONAQ')
ORNL = filter_data(input, 'Site', keep_vals = 'ORNL')
OSBS = filter_data(input, 'Site', keep_vals = 'OSBS')
PUUM = filter_data(input, 'Site', keep_vals = 'PUUM')
RMNP = filter_data(input, 'Site', keep_vals = 'RMNP')
SCBI = filter_data(input, 'Site', keep_vals = 'SCBI')
SERC = filter_data(input, 'Site', keep_vals = 'SERC')
SJER = filter_data(input, 'Site', keep_vals = 'SJER')
SOAP = filter_data(input, 'Site', keep_vals = 'SOAP')
STEI = filter_data(input, 'Site', keep_vals = 'STEI')
STER = filter_data(input, 'Site', keep_vals = 'STER')
TALL = filter_data(input, 'Site', keep_vals = 'TALL')
TOOL = filter_data(input, 'Site', keep_vals = 'TOOL')
TREE = filter_data(input, 'Site', keep_vals = 'TREE')
UKFS = filter_data(input, 'Site', keep_vals = 'UKFS')
WREF = filter_data(input, 'Site', keep_vals = 'WREF')
YELL = filter_data(input, 'Site', keep_vals = 'YELL')
```

```{r}
site<-c("ABBY","BARR","BART","BLAN","BONA","CLBJ","CPER","DCFS","DEJU","DELA","DSNY","GRSM","GUAN","HARV","HEAL","JERC","JORN","KONA","KONZ","LAJA","LENO","MLBS","MOAB","NIWO","OAES","ONAQ","ORNL","OSBS","PUUM","RMNP","SCBI","SERC","SJER","SOAP","SRER","STEI","STER","TALL","TOOL","TREE","UKFS","UNDE","WOOD","WREF","YELL")
OTUnumber<-c(nrow(ABBY$data_loaded),nrow(BARR$data_loaded),nrow(BART$data_loaded),nrow(BLAN$data_loaded),nrow(BONA$data_loaded),nrow(CLBJ$data_loaded),nrow(CPER$data_loaded),nrow(DCFS$data_loaded),nrow(DEJU$data_loaded),nrow(DELA$data_loaded),nrow(DSNY$data_loaded),nrow(GRSM$data_loaded),nrow(GUAN$data_loaded),nrow(HARV$data_loaded),nrow(HEAL$data_loaded),nrow(JERC$data_loaded),nrow(JORN$data_loaded),nrow(KONA$data_loaded),nrow(KONZ$data_loaded),nrow(LAJA$data_loaded),nrow(LENO$data_loaded),nrow(MLBS$data_loaded),nrow(MOAB$data_loaded),nrow(NIWO$data_loaded),nrow(OAES$data_loaded),nrow(ONAQ$data_loaded),nrow(ORNL$data_loaded),nrow(OSBS$data_loaded),nrow(PUUM$data_loaded),nrow(RMNP$data_loaded),nrow(SCBI$data_loaded),nrow(SERC$data_loaded),nrow(SJER$data_loaded),nrow(SOAP$data_loaded),nrow(SRER$data_loaded),nrow(STEI$data_loaded),nrow(STER$data_loaded),nrow(TALL$data_loaded),nrow(TOOL$data_loaded),nrow(TREE$data_loaded),nrow(UKFS$data_loaded),nrow(UNDE$data_loaded),nrow(WOOD$data_loaded),nrow(WREF$data_loaded),nrow(YELL$data_loaded))
site<-as.data.frame(site)
OTU_numbers_by_site<-cbind(site,OTUnumber)
```

```{r}
OTU_numbers_by_site<-OTU_numbers_by_site[order(OTU_numbers_by_site$OTUnumber),]
i<-ggplot(data=OTU_numbers_by_site,aes(x=site,y=OTUnumber)) + 
  geom_bar(stat="identity")+
  coord_flip()+
  xlab("Site")+
  ylab("# of OTUs")+
  theme(
    axis.text = element_text(color='black',size=10),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2)) 
ggsave(filename = "OTU Numbers by site.pdf",
      plot = i,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
ABBY_rowsums<-rowSums(ABBY$data_loaded)
ABBY_rowsums<-as.data.frame(ABBY_rowsums)
site<-rep("ABBY",times=nrow(ABBY_rowsums))
ABBY_rowsums_f<-cbind(ABBY_rowsums,site)
ABBY_rowsums_f<-ABBY_rowsums_f %>% rename(value=ABBY_rowsums)

BARR_rowsums<-rowSums(BARR$data_loaded)
BARR_rowsums<-as.data.frame(BARR_rowsums)
site<-rep("BARR",times=nrow(BARR_rowsums))
BARR_rowsums_f<-cbind(BARR_rowsums,site)
BARR_rowsums_f<-BARR_rowsums_f %>% rename(value=BARR_rowsums)

BART_rowsums<-rowSums(BART$data_loaded)
BART_rowsums<-as.data.frame(BART_rowsums)
site<-rep("BART",times=nrow(BART_rowsums))
BART_rowsums_f<-cbind(BART_rowsums,site)
BART_rowsums_f<-BART_rowsums_f %>% rename(value=BART_rowsums)

BLAN_rowsums<-rowSums(BLAN$data_loaded)
BLAN_rowsums<-as.data.frame(BLAN_rowsums)
site<-rep("BLAN",times=nrow(BLAN_rowsums))
BLAN_rowsums_f<-cbind(BLAN_rowsums,site)
BLAN_rowsums_f<-BLAN_rowsums_f %>% rename(value=BLAN_rowsums)

BONA_rowsums<-rowSums(BONA$data_loaded)
BONA_rowsums<-as.data.frame(BONA_rowsums)
site<-rep("BONA",times=nrow(BONA_rowsums))
BONA_rowsums_f<-cbind(BONA_rowsums,site)
BONA_rowsums_f<-BONA_rowsums_f %>% rename(value=BONA_rowsums)

CLBJ_rowsums<-rowSums(CLBJ$data_loaded)
CLBJ_rowsums<-as.data.frame(CLBJ_rowsums)
site<-rep("CLBJ",times=nrow(CLBJ_rowsums))
CLBJ_rowsums_f<-cbind(CLBJ_rowsums,site)
CLBJ_rowsums_f<-CLBJ_rowsums_f %>% rename(value=CLBJ_rowsums)

CPER_rowsums<-rowSums(CPER$data_loaded)
CPER_rowsums<-as.data.frame(CPER_rowsums)
site<-rep("CPER",times=nrow(CPER_rowsums))
CPER_rowsums_f<-cbind(CPER_rowsums,site)
CPER_rowsums_f<-CPER_rowsums_f %>% rename(value=CPER_rowsums)

DCFS_rowsums<-rowSums(DCFS$data_loaded)
DCFS_rowsums<-as.data.frame(DCFS_rowsums)
site<-rep("DCFS",times=nrow(DCFS_rowsums))
DCFS_rowsums_f<-cbind(DCFS_rowsums,site)
DCFS_rowsums_f<-DCFS_rowsums_f %>% rename(value=DCFS_rowsums)

DEJU_rowsums<-rowSums(DEJU$data_loaded)
DEJU_rowsums<-as.data.frame(DEJU_rowsums)
site<-rep("DEJU",times=nrow(DEJU_rowsums))
DEJU_rowsums_f<-cbind(DEJU_rowsums,site)
DEJU_rowsums_f<-DEJU_rowsums_f %>% rename(value=DEJU_rowsums)

DELA_rowsums<-rowSums(DELA$data_loaded)
DELA_rowsums<-as.data.frame(DELA_rowsums)
site<-rep("DELA",times=nrow(DELA_rowsums))
DELA_rowsums_f<-cbind(DELA_rowsums,site)
DELA_rowsums_f<-DELA_rowsums_f %>% rename(value=DELA_rowsums)

DSNY_rowsums<-rowSums(DSNY$data_loaded)
DSNY_rowsums<-as.data.frame(DSNY_rowsums)
site<-rep("DSNY",times=nrow(DSNY_rowsums))
DSNY_rowsums_f<-cbind(DSNY_rowsums,site)
DSNY_rowsums_f<-DSNY_rowsums_f %>% rename(value=DSNY_rowsums)

GRSM_rowsums<-rowSums(GRSM$data_loaded)
GRSM_rowsums<-as.data.frame(GRSM_rowsums)
site<-rep("GRSM",times=nrow(GRSM_rowsums))
GRSM_rowsums_f<-cbind(GRSM_rowsums,site)
GRSM_rowsums_f<-GRSM_rowsums_f %>% rename(value=GRSM_rowsums)

GUAN_rowsums<-rowSums(GUAN$data_loaded)
GUAN_rowsums<-as.data.frame(GUAN_rowsums)
site<-rep("GUAN",times=nrow(GUAN_rowsums))
GUAN_rowsums_f<-cbind(GUAN_rowsums,site)
GUAN_rowsums_f<-GUAN_rowsums_f %>% rename(value=GUAN_rowsums)

HARV_rowsums<-rowSums(HARV$data_loaded)
HARV_rowsums<-as.data.frame(HARV_rowsums)
site<-rep("HARV",times=nrow(HARV_rowsums))
HARV_rowsums_f<-cbind(HARV_rowsums,site)
HARV_rowsums_f<-HARV_rowsums_f %>% rename(value=HARV_rowsums)

HEAL_rowsums<-rowSums(HEAL$data_loaded)
HEAL_rowsums<-as.data.frame(HEAL_rowsums)
site<-rep("HEAL",times=nrow(HEAL_rowsums))
HEAL_rowsums_f<-cbind(HEAL_rowsums,site)
HEAL_rowsums_f<-HEAL_rowsums_f %>% rename(value=HEAL_rowsums)

JERC_rowsums<-rowSums(JERC$data_loaded)
JERC_rowsums<-as.data.frame(JERC_rowsums)
site<-rep("JERC",times=nrow(JERC_rowsums))
JERC_rowsums_f<-cbind(JERC_rowsums,site)
JERC_rowsums_f<-JERC_rowsums_f %>% rename(value=JERC_rowsums)

JORN_rowsums<-rowSums(JORN$data_loaded)
JORN_rowsums<-as.data.frame(JORN_rowsums)
site<-rep("JORN",times=nrow(JORN_rowsums))
JORN_rowsums_f<-cbind(JORN_rowsums,site)
JORN_rowsums_f<-JORN_rowsums_f %>% rename(value=JORN_rowsums)

KONA_rowsums<-rowSums(KONA$data_loaded)
KONA_rowsums<-as.data.frame(KONA_rowsums)
site<-rep("KONA",times=nrow(KONA_rowsums))
KONA_rowsums_f<-cbind(KONA_rowsums,site)
KONA_rowsums_f<-KONA_rowsums_f %>% rename(value=KONA_rowsums)

KONZ_rowsums<-rowSums(KONZ$data_loaded)
KONZ_rowsums<-as.data.frame(KONZ_rowsums)
site<-rep("KONZ",times=nrow(KONZ_rowsums))
KONZ_rowsums_f<-cbind(KONZ_rowsums,site)
KONZ_rowsums_f<-KONZ_rowsums_f %>% rename(value=KONZ_rowsums)

LAJA_rowsums<-rowSums(LAJA$data_loaded)
LAJA_rowsums<-as.data.frame(LAJA_rowsums)
site<-rep("LAJA",times=nrow(LAJA_rowsums))
LAJA_rowsums_f<-cbind(LAJA_rowsums,site)
LAJA_rowsums_f<-LAJA_rowsums_f %>% rename(value=LAJA_rowsums)

LENO_rowsums<-rowSums(LENO$data_loaded)
LENO_rowsums<-as.data.frame(LENO_rowsums)
site<-rep("LENO",times=nrow(LENO_rowsums))
LENO_rowsums_f<-cbind(LENO_rowsums,site)
LENO_rowsums_f<-LENO_rowsums_f %>% rename(value=LENO_rowsums)

MLBS_rowsums<-rowSums(MLBS$data_loaded)
MLBS_rowsums<-as.data.frame(MLBS_rowsums)
site<-rep("MLBS",times=nrow(MLBS_rowsums))
MLBS_rowsums_f<-cbind(MLBS_rowsums,site)
MLBS_rowsums_f<-MLBS_rowsums_f %>% rename(value=MLBS_rowsums)

MOAB_rowsums<-rowSums(MOAB$data_loaded)
MOAB_rowsums<-as.data.frame(MOAB_rowsums)
site<-rep("MOAB",times=nrow(MOAB_rowsums))
MOAB_rowsums_f<-cbind(MOAB_rowsums,site)
MOAB_rowsums_f<-MOAB_rowsums_f %>% rename(value=MOAB_rowsums)

NIWO_rowsums<-rowSums(NIWO$data_loaded)
NIWO_rowsums<-as.data.frame(NIWO_rowsums)
site<-rep("NIWO",times=nrow(NIWO_rowsums))
NIWO_rowsums_f<-cbind(NIWO_rowsums,site)
NIWO_rowsums_f<-NIWO_rowsums_f %>% rename(value=NIWO_rowsums)

OAES_rowsums<-rowSums(OAES$data_loaded)
OAES_rowsums<-as.data.frame(OAES_rowsums)
site<-rep("OAES",times=nrow(OAES_rowsums))
OAES_rowsums_f<-cbind(OAES_rowsums,site)
OAES_rowsums_f<-OAES_rowsums_f %>% rename(value=OAES_rowsums)

ONAQ_rowsums<-rowSums(ONAQ$data_loaded)
ONAQ_rowsums<-as.data.frame(ONAQ_rowsums)
site<-rep("ONAQ",times=nrow(ONAQ_rowsums))
ONAQ_rowsums_f<-cbind(ONAQ_rowsums,site)
ONAQ_rowsums_f<-ONAQ_rowsums_f %>% rename(value=ONAQ_rowsums)

ORNL_rowsums<-rowSums(ORNL$data_loaded)
ORNL_rowsums<-as.data.frame(ORNL_rowsums)
site<-rep("ORNL",times=nrow(ORNL_rowsums))
ORNL_rowsums_f<-cbind(ORNL_rowsums,site)
ORNL_rowsums_f<-ORNL_rowsums_f %>% rename(value=ORNL_rowsums)

OSBS_rowsums<-rowSums(OSBS$data_loaded)
OSBS_rowsums<-as.data.frame(OSBS_rowsums)
site<-rep("OSBS",times=nrow(OSBS_rowsums))
OSBS_rowsums_f<-cbind(OSBS_rowsums,site)
OSBS_rowsums_f<-OSBS_rowsums_f %>% rename(value=OSBS_rowsums)

PUUM_rowsums<-rowSums(PUUM$data_loaded)
PUUM_rowsums<-as.data.frame(PUUM_rowsums)
site<-rep("PUUM",times=nrow(PUUM_rowsums))
PUUM_rowsums_f<-cbind(PUUM_rowsums,site)
PUUM_rowsums_f<-PUUM_rowsums_f %>% rename(value=PUUM_rowsums)

RMNP_rowsums<-rowSums(RMNP$data_loaded)
RMNP_rowsums<-as.data.frame(RMNP_rowsums)
site<-rep("RMNP",times=nrow(RMNP_rowsums))
RMNP_rowsums_f<-cbind(RMNP_rowsums,site)
RMNP_rowsums_f<-RMNP_rowsums_f %>% rename(value=RMNP_rowsums)

SCBI_rowsums<-rowSums(SCBI$data_loaded)
SCBI_rowsums<-as.data.frame(SCBI_rowsums)
site<-rep("SCBI",times=nrow(SCBI_rowsums))
SCBI_rowsums_f<-cbind(SCBI_rowsums,site)
SCBI_rowsums_f<-SCBI_rowsums_f %>% rename(value=SCBI_rowsums)

SERC_rowsums<-rowSums(SERC$data_loaded)
SERC_rowsums<-as.data.frame(SERC_rowsums)
site<-rep("SERC",times=nrow(SERC_rowsums))
SERC_rowsums_f<-cbind(SERC_rowsums,site)
SERC_rowsums_f<-SERC_rowsums_f %>% rename(value=SERC_rowsums)

SJER_rowsums<-rowSums(SJER$data_loaded)
SJER_rowsums<-as.data.frame(SJER_rowsums)
site<-rep("SJER",times=nrow(SJER_rowsums))
SJER_rowsums_f<-cbind(SJER_rowsums,site)
SJER_rowsums_f<-SJER_rowsums_f %>% rename(value=SJER_rowsums)

SOAP_rowsums<-rowSums(SOAP$data_loaded)
SOAP_rowsums<-as.data.frame(SOAP_rowsums)
site<-rep("SOAP",times=nrow(SOAP_rowsums))
SOAP_rowsums_f<-cbind(SOAP_rowsums,site)
SOAP_rowsums_f<-SOAP_rowsums_f %>% rename(value=SOAP_rowsums)

SRER_rowsums<-rowSums(SRER$data_loaded)
SRER_rowsums<-as.data.frame(SRER_rowsums)
site<-rep("SRER",times=nrow(SRER_rowsums))
SRER_rowsums_f<-cbind(SRER_rowsums,site)
SRER_rowsums_f<-SRER_rowsums_f %>% rename(value=SRER_rowsums)

STEI_rowsums<-rowSums(STEI$data_loaded)
STEI_rowsums<-as.data.frame(STEI_rowsums)
site<-rep("STEI",times=nrow(STEI_rowsums))
STEI_rowsums_f<-cbind(STEI_rowsums,site)
STEI_rowsums_f<-STEI_rowsums_f %>% rename(value=STEI_rowsums)

STER_rowsums<-rowSums(STER$data_loaded)
STER_rowsums<-as.data.frame(STER_rowsums)
site<-rep("STER",times=nrow(STER_rowsums))
STER_rowsums_f<-cbind(STER_rowsums,site)
STER_rowsums_f<-STER_rowsums_f %>% rename(value=STER_rowsums)

TALL_rowsums<-rowSums(TALL$data_loaded)
TALL_rowsums<-as.data.frame(TALL_rowsums)
site<-rep("TALL",times=nrow(TALL_rowsums))
TALL_rowsums_f<-cbind(TALL_rowsums,site)
TALL_rowsums_f<-TALL_rowsums_f %>% rename(value=TALL_rowsums)

TOOL_rowsums<-rowSums(TOOL$data_loaded)
TOOL_rowsums<-as.data.frame(TOOL_rowsums)
site<-rep("TOOL",times=nrow(TOOL_rowsums))
TOOL_rowsums_f<-cbind(TOOL_rowsums,site)
TOOL_rowsums_f<-TOOL_rowsums_f %>% rename(value=TOOL_rowsums)

TREE_rowsums<-rowSums(TREE$data_loaded)
TREE_rowsums<-as.data.frame(TREE_rowsums)
site<-rep("TREE",times=nrow(TREE_rowsums))
TREE_rowsums_f<-cbind(TREE_rowsums,site)
TREE_rowsums_f<-TREE_rowsums_f %>% rename(value=TREE_rowsums)

UKFS_rowsums<-rowSums(UKFS$data_loaded)
UKFS_rowsums<-as.data.frame(UKFS_rowsums)
site<-rep("UKFS",times=nrow(UKFS_rowsums))
UKFS_rowsums_f<-cbind(UKFS_rowsums,site)
UKFS_rowsums_f<-UKFS_rowsums_f %>% rename(value=UKFS_rowsums)

UNDE_rowsums<-rowSums(UNDE$data_loaded)
UNDE_rowsums<-as.data.frame(UNDE_rowsums)
site<-rep("UNDE",times=nrow(UNDE_rowsums))
UNDE_rowsums_f<-cbind(UNDE_rowsums,site)
UNDE_rowsums_f<-UNDE_rowsums_f %>% rename(value=UNDE_rowsums)

WOOD_rowsums<-rowSums(WOOD$data_loaded)
WOOD_rowsums<-as.data.frame(WOOD_rowsums)
site<-rep("WOOD",times=nrow(WOOD_rowsums))
WOOD_rowsums_f<-cbind(WOOD_rowsums,site)
WOOD_rowsums_f<-WOOD_rowsums_f %>% rename(value=WOOD_rowsums)

WREF_rowsums<-rowSums(WREF$data_loaded)
WREF_rowsums<-as.data.frame(WREF_rowsums)
site<-rep("WREF",times=nrow(WREF_rowsums))
WREF_rowsums_f<-cbind(WREF_rowsums,site)
WREF_rowsums_f<-WREF_rowsums_f %>% rename(value=WREF_rowsums)

YELL_rowsums<-rowSums(YELL$data_loaded)
YELL_rowsums<-as.data.frame(YELL_rowsums)
site<-rep("YELL",times=nrow(YELL_rowsums))
YELL_rowsums_f<-cbind(YELL_rowsums,site)
YELL_rowsums_f<-YELL_rowsums_f %>% rename(value=YELL_rowsums)
```


```{r}
reads_site<-do.call("rbind",list(ABBY_rowsums_f, BARR_rowsums_f, BART_rowsums_f, BLAN_rowsums_f, BONA_rowsums_f, CLBJ_rowsums_f, CPER_rowsums_f, DCFS_rowsums_f, DEJU_rowsums_f, DELA_rowsums_f, DSNY_rowsums_f, GRSM_rowsums_f, GUAN_rowsums_f, HARV_rowsums_f, HEAL_rowsums_f, JERC_rowsums_f, JORN_rowsums_f, KONA_rowsums_f, KONZ_rowsums_f, LAJA_rowsums_f, LENO_rowsums_f, MLBS_rowsums_f, MOAB_rowsums_f, NIWO_rowsums_f, OAES_rowsums_f, ONAQ_rowsums_f, ORNL_rowsums_f, OSBS_rowsums_f, PUUM_rowsums_f, RMNP_rowsums_f, SCBI_rowsums_f, SERC_rowsums_f, SJER_rowsums_f, SOAP_rowsums_f, SRER_rowsums_f, STEI_rowsums_f, STER_rowsums_f, TALL_rowsums_f, TOOL_rowsums_f, TREE_rowsums_f, UKFS_rowsums_f, UNDE_rowsums_f, WOOD_rowsums_f, WREF_rowsums_f, YELL_rowsums_f))
```

```{r}
j<-ggplot(data=reads_site,aes(x=site,y=value,color=factor(site))) + 
  geom_point()+
  coord_flip()+
  xlab("Site")+
  ylab("# of reads")+
  theme(
    axis.text = element_text(color='black',size=10),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2)) 
j
```

```{r}
ggsave(filename = "Reads by site.pdf",
      plot = j,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
pdf(file = "rarecurve.pdf",   
    width = 8, 
    height = 8) 
rarecurve(t(input$data_loaded), step=100, label=FALSE, cex=0.5, main="Whole Eukaryote Community") 
dev.off()
```

```{r}
 sp1 <- specaccum(t(input$data_loaded), method="exact")
pdf(file = "specaccum.pdf",   
    width = 8, 
    height = 8) 
 plot(sp1, ci.type="poly", col="blue", lwd=2, ci.lty=0, ci.col="lightblue" )
 dev.off()
```

## BETA DIVERSITY ##
#Q4 How do communities (site compositions) differ by biome assignment?
  #PCoA biome (Bates 2013 Fig 3)
#Q5 How homogenous are communities over large spatial scales?
  #Community Dissimilarity - Phyla (Bates 2013 Fig 2)
#Q6 Is everything everywhere? Prediction: dispersal is constrained by body size patterns (papers?) This one would have to be genus or at least family.
  #PCoA is community dissimilarity driven by body size?  Or another test?

```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
```

```{r}
library(labdsv)
biome = filter_data(input, 'nlcdClass', filter_vals = '0')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'SRER_003-M-20180801-COMP')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'ONAQ_010-M-20200710-COMP')
otuTAB<-biome$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
biome$data_loaded<-t(hell)
```

```{r}
biome.mds<-metaMDS(t(biome$data_loaded))
biome.scores<-vegan::scores(biome.mds)
data.scoresbiome <- as.data.frame(biome.scores$sites) 
data.scoresbiome$biomes <- biome$map_loaded$nlcdClass
```

```{r}
library(ggplot2)
a<-ggplot(data=data.scoresbiome,aes(x=NMDS1,y=NMDS2,color=factor(biomes))) + 
  
  
  geom_point(data=data.scoresbiome,aes(x=NMDS1,y=NMDS2,color=factor(biomes))) + # add the point markers
  xlim(-1,1.5)+
  ylim(-1.5,1.5)+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
a

```

```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary <- summarySE(data.scoresbiome, measurevar=c("NMDS1"), groupvars=("biomes"))
nmds1_summary
nmds2_summary <- summarySE(data.scoresbiome, measurevar=c("NMDS2"), groupvars=("biomes"))
nmds2_summary
library(dplyr)
nmds1_summary<-select(nmds1_summary,-c(N,sd,ci))
nmds1_summary<-nmds1_summary %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary<-select(nmds2_summary,-c(N,sd,ci))
nmds2_summary<-nmds2_summary %>% dplyr::rename("se_NMDS2"= se)
nmds_summary<-merge(nmds1_summary,nmds2_summary,by="biomes")
```

```{r}
library(ggplot2)
q<-ggplot(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(biomes))) + 
  geom_point(data=data.scoresbiome,aes(x=NMDS1,y=NMDS2,color=factor(biomes)),alpha = 0.2) +
  
  geom_point(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(biomes)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.1) +
  geom_errorbar(aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.1) +
  ylim(-0.75,0.75) +
  xlim(-1,1) +
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
q

```

```{r}
ggsave(filename = "Biome_average_se_withpoints.pdf",
      plot = q,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
model1<-adonis2(t(biome$data_loaded)~nlcdClass,data=biome$map_loaded,permutations=999,method="bray")
model1
```



# Let's try presence/absence, because let's say an earthworm is in there, it could skew the # of reads

```{r}
biome.mds<-metaMDS(comm = t(biome$data_loaded),distance = "jaccard", k=2)
biome.scores<-vegan::scores(biome.mds)
data.scoresbiome <- as.data.frame(biome.scores$sites) 
data.scoresbiome$biomes <- biome$map_loaded$nlcdClass
```

```{r}
library(ggplot2)
a<-ggplot(data=data.scoresbiome,aes(x=NMDS1,y=NMDS2,color=factor(biomes))) + 
  
  
  geom_point(data=data.scoresbiome,aes(x=NMDS1,y=NMDS2,color=factor(biomes))) + # add the point markers
  xlim(-0.75,0.75)+
  ylim(-1,1)+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
a

```
```{r}
ggsave(filename = "Biome_nmds_presenceandabsence.pdf",
      plot = a,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```


```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary <- summarySE(data.scoresbiome, measurevar=c("NMDS1"), groupvars=("biomes"))
nmds1_summary
nmds2_summary <- summarySE(data.scoresbiome, measurevar=c("NMDS2"), groupvars=("biomes"))
nmds2_summary
library(dplyr)
nmds1_summary<-select(nmds1_summary,-c(N,sd,ci))
nmds1_summary<-nmds1_summary %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary<-select(nmds2_summary,-c(N,sd,ci))
nmds2_summary<-nmds2_summary %>% dplyr::rename("se_NMDS2"= se)
nmds_summary<-merge(nmds1_summary,nmds2_summary,by="biomes")
```

```{r}
library(ggplot2)
q<-ggplot(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(biomes))) + 
  geom_point(data=data.scoresbiome,aes(x=NMDS1,y=NMDS2,color=factor(biomes)),alpha = 0.2) +
  
  geom_point(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(biomes)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.025) +
  geom_errorbar(aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.025) +
  ylim(-0.5,0.5) +
  xlim(-0.5,0.5) +
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
q

```

```{r}
ggsave(filename = "Biome_average_se_withpoints_presenceandabsence.pdf",
      plot = q,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```


```{r}
model1<-adonis2(t(biome$data_loaded)~nlcdClass,data=biome$map_loaded,permutations=999,method="jaccard")
model1
```

```{r}
library(ecole)
a<-permanova_pairwise(
  t(biome$data_loaded),
  biome$map_loaded$nlcdClass,
  permutations = 999,
  method = "jaccard",
  padj = "bonferroni"
)
write.csv(a,file='perm_all.csv', row.names=FALSE)
```


## Dive into Metazoa

# First, we will make an NMDS plot by biome with metazoa only

We will need to first filter out everything, but metazoa

```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input_metazoa = filter_taxa_from_input(input, taxa_to_keep = 'Metazoa')
```

Now that we have metazoa removed, we can make the NMDS by biome and Phylum barchart by biome and site

```{r}
input_metazoa = filter_data(input_metazoa, 'nlcdClass', filter_vals = '0')
cultivatedCrops = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'cultivatedCrops')
deciduousForest = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'deciduousForest')
dwarfScrub = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'dwarfScrub')
emergentHerbaceousWetlands = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'emergentHerbaceousWetlands')
evergreenForest = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'evergreenForest')
grasslandHerbaceous = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'grasslandHerbaceous')
mixedForest = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'mixedForest')
pastureHay = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'pastureHay')
sedgeHerbaceous = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'sedgeHerbaceous')
shrubScrub = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'shrubScrub')
woodyWetlands = filter_data(input_metazoa, 'nlcdClass', keep_vals = 'woodyWetlands')
```

```{r}
otuTAB<-cultivatedCrops$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
cultivatedCrops$data_loaded<-t(hell)

otuTAB<-deciduousForest$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
deciduousForest$data_loaded<-t(hell)

otuTAB<-dwarfScrub$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
dwarfScrub$data_loaded<-t(hell)

otuTAB<-emergentHerbaceousWetlands$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
emergentHerbaceousWetlands$data_loaded<-t(hell)

otuTAB<-evergreenForest$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
evergreenForest$data_loaded<-t(hell)

otuTAB<-grasslandHerbaceous$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
grasslandHerbaceous$data_loaded<-t(hell)

otuTAB<-mixedForest$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
mixedForest$data_loaded<-t(hell)

otuTAB<-pastureHay$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
pastureHay$data_loaded<-t(hell)

otuTAB<-sedgeHerbaceous$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
sedgeHerbaceous$data_loaded<-t(hell)

otuTAB<-shrubScrub$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
shrubScrub$data_loaded<-t(hell)

otuTAB<-woodyWetlands$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
woodyWetlands$data_loaded<-t(hell)
```

```{r}
library(reshape2)
cultivatedCrops_tax_sum_phylum = summarize_taxonomy(cultivatedCrops, level = 4, report_higher_tax = FALSE)
cultivatedCrops_tax_sum_phylum$Phylum <-row.names(cultivatedCrops_tax_sum_phylum)
cultivatedCrops_melt<- melt(cultivatedCrops_tax_sum_phylum, id.var="Phylum")
nrow(cultivatedCrops_melt)
site<-rep("cultivatedCrops",times=nrow(cultivatedCrops_melt))
cultivatedCrops_melt<-cbind(cultivatedCrops_melt,site)

deciduousForest_tax_sum_phylum = summarize_taxonomy(deciduousForest, level = 4, report_higher_tax = FALSE)
deciduousForest_tax_sum_phylum$Phylum <-row.names(deciduousForest_tax_sum_phylum)
deciduousForest_melt<- melt(deciduousForest_tax_sum_phylum, id.var="Phylum")
nrow(deciduousForest_melt)
site<-rep("deciduousForest",times=nrow(deciduousForest_melt))
deciduousForest_melt<-cbind(deciduousForest_melt,site)

dwarfScrub_tax_sum_phylum = summarize_taxonomy(dwarfScrub, level = 4, report_higher_tax = FALSE)
dwarfScrub_tax_sum_phylum$Phylum <-row.names(dwarfScrub_tax_sum_phylum)
dwarfScrub_melt<- melt(dwarfScrub_tax_sum_phylum, id.var="Phylum")
nrow(dwarfScrub_melt)
site<-rep("dwarfScrub",times=nrow(dwarfScrub_melt))
dwarfScrub_melt<-cbind(dwarfScrub_melt,site)

emergentHerbaceousWetlands_tax_sum_phylum = summarize_taxonomy(emergentHerbaceousWetlands, level = 4, report_higher_tax = FALSE)
emergentHerbaceousWetlands_tax_sum_phylum$Phylum <-row.names(emergentHerbaceousWetlands_tax_sum_phylum)
emergentHerbaceousWetlands_melt<- melt(emergentHerbaceousWetlands_tax_sum_phylum, id.var="Phylum")
nrow(emergentHerbaceousWetlands_melt)
site<-rep("emergentHerbaceousWetlands",times=nrow(emergentHerbaceousWetlands_melt))
emergentHerbaceousWetlands_melt<-cbind(emergentHerbaceousWetlands_melt,site)

evergreenForest_tax_sum_phylum = summarize_taxonomy(evergreenForest, level = 4, report_higher_tax = FALSE)
evergreenForest_tax_sum_phylum$Phylum <-row.names(evergreenForest_tax_sum_phylum)
evergreenForest_melt<- melt(evergreenForest_tax_sum_phylum, id.var="Phylum")
nrow(evergreenForest_melt)
site<-rep("evergreenForest",times=nrow(evergreenForest_melt))
evergreenForest_melt<-cbind(evergreenForest_melt,site)

grasslandHerbaceous_tax_sum_phylum = summarize_taxonomy(grasslandHerbaceous, level = 4, report_higher_tax = FALSE)
grasslandHerbaceous_tax_sum_phylum$Phylum <-row.names(grasslandHerbaceous_tax_sum_phylum)
grasslandHerbaceous_melt<- melt(grasslandHerbaceous_tax_sum_phylum, id.var="Phylum")
nrow(grasslandHerbaceous_melt)
site<-rep("grasslandHerbaceous",times=nrow(grasslandHerbaceous_melt))
grasslandHerbaceous_melt<-cbind(grasslandHerbaceous_melt,site)

mixedForest_tax_sum_phylum = summarize_taxonomy(mixedForest, level = 4, report_higher_tax = FALSE)
mixedForest_tax_sum_phylum$Phylum <-row.names(mixedForest_tax_sum_phylum)
mixedForest_melt<- melt(mixedForest_tax_sum_phylum, id.var="Phylum")
nrow(mixedForest_melt)
site<-rep("mixedForest",times=nrow(mixedForest_melt))
mixedForest_melt<-cbind(mixedForest_melt,site)

pastureHay_tax_sum_phylum = summarize_taxonomy(pastureHay, level = 4, report_higher_tax = FALSE)
pastureHay_tax_sum_phylum$Phylum <-row.names(pastureHay_tax_sum_phylum)
pastureHay_melt<- melt(pastureHay_tax_sum_phylum, id.var="Phylum")
nrow(pastureHay_melt)
site<-rep("pastureHay",times=nrow(pastureHay_melt))
pastureHay_melt<-cbind(pastureHay_melt,site)

sedgeHerbaceous_tax_sum_phylum = summarize_taxonomy(sedgeHerbaceous, level = 4, report_higher_tax = FALSE)
sedgeHerbaceous_tax_sum_phylum$Phylum <-row.names(sedgeHerbaceous_tax_sum_phylum)
sedgeHerbaceous_melt<- melt(sedgeHerbaceous_tax_sum_phylum, id.var="Phylum")
nrow(sedgeHerbaceous_melt)
site<-rep("sedgeHerbaceous",times=nrow(sedgeHerbaceous_melt))
sedgeHerbaceous_melt<-cbind(sedgeHerbaceous_melt,site)

shrubScrub_tax_sum_phylum = summarize_taxonomy(shrubScrub, level = 4, report_higher_tax = FALSE)
shrubScrub_tax_sum_phylum$Phylum <-row.names(shrubScrub_tax_sum_phylum)
shrubScrub_melt<- melt(shrubScrub_tax_sum_phylum, id.var="Phylum")
nrow(shrubScrub_melt)
site<-rep("shrubScrub",times=nrow(shrubScrub_melt))
shrubScrub_melt<-cbind(shrubScrub_melt,site)

woodyWetlands_tax_sum_phylum = summarize_taxonomy(woodyWetlands, level = 4, report_higher_tax = FALSE)
woodyWetlands_tax_sum_phylum$Phylum <-row.names(woodyWetlands_tax_sum_phylum)
woodyWetlands_melt<- melt(woodyWetlands_tax_sum_phylum, id.var="Phylum")
nrow(woodyWetlands_melt)
site<-rep("woodyWetlands",times=nrow(woodyWetlands_melt))
woodyWetlands_melt<-cbind(woodyWetlands_melt,site)
```

```{r}
phylumbiomebind<-do.call("rbind",list(cultivatedCrops_melt,deciduousForest_melt,dwarfScrub_melt,emergentHerbaceousWetlands_melt,evergreenForest_melt,grasslandHerbaceous_melt,mixedForest_melt,pastureHay_melt,sedgeHerbaceous_melt,shrubScrub_melt,woodyWetlands_melt))
```

phylum graph
```{r}
library(ggthemes)
z<-ggplot(phylumbiomebind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(6))(23)) + 
  ggtitle("Metazoa Only by Biome")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

z
```

```{r}

ggsave(filename = "metazoaonly_bybiome.pdf",
      plot = z,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

Now we can make biome with metazoa
```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input_metazoa = filter_taxa_from_input(input, taxa_to_keep = 'Metazoa')
#write.csv(input_metazoa$taxonomy_loaded,'input_metazoa_taxonomy.csv')
write.csv(input_metazoa$data_loaded,'input_metazoa_otu.csv')
write.csv(input_metazoa$map_loaded,'input_metazoa_map.csv')
```

```{r}
library(labdsv)
biome = filter_data(input_metazoa, 'nlcdClass', filter_vals = '0')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'SRER_003-M-20180801-COMP')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'ONAQ_010-M-20200710-COMP')
otuTAB<-biome$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
biome$data_loaded<-t(hell)
biome$data_loaded<-biome$data_loaded[ , apply(biome$data_loaded, 2, function(x) !any(is.na(x)))]

```

```{r}
biome.mds<-metaMDS(t(biome$data_loaded))
biome.scores<-vegan::scores(biome.mds)
data.scoresbiome <- as.data.frame(biome.scores$sites) 
library(janitor)
data.scoresbiome.merge<-merge(data.scoresbiome,biome$map_loaded,by = 'row.names')
```

Get rid of outliers
```{r}
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<0.4)
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1>(-0.5))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2>(-1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2<0.05)
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(-0.14))
```

```{r}
library(ggplot2)
a<-ggplot(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
a

```

```{r}
ggsave(filename = "Biome_nmds_metazoa.pdf",
      plot = a,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds1_summary
nmds2_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
nmds2_summary
library(dplyr)
nmds1_summary<-select(nmds1_summary,-c(N,sd,ci))
nmds1_summary<-nmds1_summary %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary<-select(nmds2_summary,-c(N,sd,ci))
nmds2_summary<-nmds2_summary %>% dplyr::rename("se_NMDS2"= se)
nmds_summary<-merge(nmds1_summary,nmds2_summary,by="nlcdClass")
```

```{r}
library(ggplot2)
q<-ggplot(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.001) +
  geom_errorbar(data=nmds_summary,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.001) +
  xlim(-0.21,-0.19)+
 
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
q

```

```{r}
ggsave(filename = "Biome_average_se_withpoints_nmds_metazoa.pdf",
      plot = q,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```



# Let's do this with jaccard (presence/absence) -> explaiined why above

```{r}
biome.mds<-metaMDS(comm = t(biome$data_loaded),distance = "jaccard", k=2)
biome.scores<-vegan::scores(biome.mds)
data.scoresbiome <- as.data.frame(biome.scores$sites) 
library(janitor)
data.scoresbiome.merge<-merge(data.scoresbiome,biome$map_loaded,by = 'row.names')
```

Get rid of outliers
```{r}
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(-0.1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1>(-1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2>(-1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2<(0.1))
#data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(-0.14))
```

```{r}
library(ggplot2)
a<-ggplot(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
a

```

```{r}
ggsave(filename = "Biome_nmds_metazoa_presenceandabsence.pdf",
      plot = a,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds1_summary
nmds2_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
nmds2_summary
library(dplyr)
nmds1_summary<-select(nmds1_summary,-c(N,sd,ci))
nmds1_summary<-nmds1_summary %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary<-select(nmds2_summary,-c(N,sd,ci))
nmds2_summary<-nmds2_summary %>% dplyr::rename("se_NMDS2"= se)
nmds_summary<-merge(nmds1_summary,nmds2_summary,by="nlcdClass")
```

```{r}
library(ggplot2)
q<-ggplot(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.00025) +
  geom_errorbar(data=nmds_summary,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.00025) +
  ylim(-0.012,-0.003)+
  xlim(-0.265,-0.252)+

 
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
q

```

```{r}
ggsave(filename = "Biome_average_se_withpoints_nmds_metazoa_presenceabsence.pdf",
      plot = q,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```



Fungi
```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input_metazoa = filter_taxa_from_input(input, taxa_to_keep = 'Fungi')
```

```{r}
library(labdsv)
biome = filter_data(input_metazoa, 'nlcdClass', filter_vals = '0')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'SRER_003-M-20180801-COMP')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'ONAQ_010-M-20200710-COMP')
otuTAB<-biome$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
biome$data_loaded<-t(hell)
biome$data_loaded<-biome$data_loaded[ , apply(biome$data_loaded, 2, function(x) !any(is.na(x)))]

```


# Let's do this with jaccard (presence/absence) -> explaiined why above

```{r}
biome.mds<-metaMDS(comm = t(biome$data_loaded),distance = "jaccard", k=2)
biome.scores<-vegan::scores(biome.mds)
data.scoresbiome <- as.data.frame(biome.scores$sites) 
library(janitor)
data.scoresbiome.merge<-merge(data.scoresbiome,biome$map_loaded,by = 'row.names')
```

Get rid of outliers
```{r}
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1>(-1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2>(-1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2<(1))
#data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(-0.14))
```

```{r}
library(ggplot2)
a<-ggplot(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
a

```

```{r}
ggsave(filename = "Biome_nmds_fungi_presenceandabsence.pdf",
      plot = a,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds1_summary
nmds2_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
nmds2_summary
library(dplyr)
nmds1_summary<-select(nmds1_summary,-c(N,sd,ci))
nmds1_summary<-nmds1_summary %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary<-select(nmds2_summary,-c(N,sd,ci))
nmds2_summary<-nmds2_summary %>% dplyr::rename("se_NMDS2"= se)
nmds_summary<-merge(nmds1_summary,nmds2_summary,by="nlcdClass")
```

```{r}
library(ggplot2)
q<-ggplot(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.00025) +
  geom_errorbar(data=nmds_summary,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.00025) +
  ylim(-0.25,0.25)+
  xlim(-0.25,0.5)+

 
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
q

```

```{r}
ggsave(filename = "Biome_average_se_withpoints_nmds_fungi_presenceabsence.pdf",
      plot = q,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

Nematoda
```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input_metazoa = filter_taxa_from_input(input, taxa_to_keep = 'Nematoda')
```

```{r}
library(labdsv)
biome = filter_data(input_metazoa, 'nlcdClass', filter_vals = '0')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'SRER_003-M-20180801-COMP')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'ONAQ_010-M-20200710-COMP')
otuTAB<-biome$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
biome$data_loaded<-t(hell)
biome$data_loaded<-biome$data_loaded[ , apply(biome$data_loaded, 2, function(x) !any(is.na(x)))]

```


# Let's do this with jaccard (presence/absence) -> explaiined why above

```{r}
biome.mds<-metaMDS(comm = t(biome$data_loaded),distance = "jaccard", k=2)
biome.scores<-vegan::scores(biome.mds)
data.scoresbiome <- as.data.frame(biome.scores$sites) 
library(janitor)
data.scoresbiome.merge<-merge(data.scoresbiome,biome$map_loaded,by = 'row.names')
```

Get rid of outliers
```{r}
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1>(-1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2>(-1))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2<(1))
#data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(-0.14))
```

```{r}
library(ggplot2)
a<-ggplot(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
a

```

```{r}
ggsave(filename = "Biome_nmds_nematodes_presenceandabsence.pdf",
      plot = a,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds1_summary
nmds2_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
nmds2_summary
library(dplyr)
nmds1_summary<-select(nmds1_summary,-c(N,sd,ci))
nmds1_summary<-nmds1_summary %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary<-select(nmds2_summary,-c(N,sd,ci))
nmds2_summary<-nmds2_summary %>% dplyr::rename("se_NMDS2"= se)
nmds_summary<-merge(nmds1_summary,nmds2_summary,by="nlcdClass")
```

```{r}
library(ggplot2)
q<-ggplot(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.00025) +
  geom_errorbar(data=nmds_summary,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.00025) +
  ylim(-0.35,0)+
  xlim(-0.6,-0.35)+

 
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
q

```

```{r}
ggsave(filename = "Biome_average_se_withpoints_nmds_nematodes_presenceabsence.pdf",
      plot = q,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

Arthopoda
```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input_metazoa = filter_taxa_from_input(input, taxa_to_keep = 'Arthropoda')
```

```{r}
library(labdsv)
biome = filter_data(input_metazoa, 'nlcdClass', filter_vals = '0')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'SRER_003-M-20180801-COMP')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'ONAQ_010-M-20200710-COMP')
otuTAB<-biome$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
biome$data_loaded<-t(hell)
biome$data_loaded<-biome$data_loaded[ , apply(biome$data_loaded, 2, function(x) !any(is.na(x)))]

```


# Let's do this with jaccard (presence/absence) -> explaiined why above

```{r}
biome.mds<-metaMDS(comm = t(biome$data_loaded),distance = "jaccard", k=2)
biome.scores<-vegan::scores(biome.mds)
data.scoresbiome <- as.data.frame(biome.scores$sites) 
library(janitor)
data.scoresbiome.merge<-merge(data.scoresbiome,biome$map_loaded,by = 'row.names')
```

Get rid of outliers
```{r}
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(0))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1>(-0.3))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2>(-0.5))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2<(0))
#data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(-0.14))
```

```{r}
library(ggplot2)
a<-ggplot(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
a

```

```{r}
ggsave(filename = "Biome_nmds_arthropoda_presenceandabsence.pdf",
      plot = a,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds1_summary
nmds2_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
nmds2_summary
library(dplyr)
nmds1_summary<-select(nmds1_summary,-c(N,sd,ci))
nmds1_summary<-nmds1_summary %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary<-select(nmds2_summary,-c(N,sd,ci))
nmds2_summary<-nmds2_summary %>% dplyr::rename("se_NMDS2"= se)
nmds_summary<-merge(nmds1_summary,nmds2_summary,by="nlcdClass")
```

```{r}
library(ggplot2)
q<-ggplot(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.00025) +
  geom_errorbar(data=nmds_summary,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.00025) +
  ylim(-0.15,-0.03)+
  xlim(-0.235,-0.2)+

 
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
q

```

```{r}
ggsave(filename = "Biome_average_se_withpoints_nmds_arthropods_presenceabsence.pdf",
      plot = q,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

Annelids
```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input_metazoa = filter_taxa_from_input(input, taxa_to_keep = 'Annelida')
```

```{r}
library(labdsv)
biome = filter_data(input_metazoa, 'nlcdClass', filter_vals = '0')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'SRER_003-M-20180801-COMP')
biome = filter_data(biome,'genomicsSampleID',filter_vals = 'ONAQ_010-M-20200710-COMP')
otuTAB<-biome$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
biome$data_loaded<-t(hell)
biome$data_loaded<-biome$data_loaded[ , apply(biome$data_loaded, 2, function(x) !any(is.na(x)))]

```


# Let's do this with jaccard (presence/absence) -> explaiined why above

```{r}
biome.mds<-metaMDS(comm = t(biome$data_loaded),distance = "jaccard", k=2)
biome.scores<-vegan::scores(biome.mds)
data.scoresbiome <- as.data.frame(biome.scores$sites) 
library(janitor)
data.scoresbiome.merge<-merge(data.scoresbiome,biome$map_loaded,by = 'row.names')
```

Get rid of outliers
```{r}
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(-0.3))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1>(-0.5))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2<(0))
data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS2>(-0.13))
#data.scoresbiome.merge<-subset(data.scoresbiome.merge,NMDS1<(-0.14))
```

```{r}
library(ggplot2)
a<-ggplot(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
a

```

```{r}
ggsave(filename = "Biome_nmds_annelida_presenceandabsence.pdf",
      plot = a,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds1_summary
nmds2_summary <- summarySE(data.scoresbiome.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
nmds2_summary
library(dplyr)
nmds1_summary<-select(nmds1_summary,-c(N,sd,ci))
nmds1_summary<-nmds1_summary %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary<-select(nmds2_summary,-c(N,sd,ci))
nmds2_summary<-nmds2_summary %>% dplyr::rename("se_NMDS2"= se)
nmds_summary<-merge(nmds1_summary,nmds2_summary,by="nlcdClass")
```

```{r}
library(ggplot2)
q<-ggplot(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresbiome.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.00025) +
  geom_errorbar(data=nmds_summary,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.00025) +
  ylim(-0.11,-0.09)+
  xlim(-0.37,-0.34)+

 
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
q

```

```{r}
ggsave(filename = "Biome_average_se_withpoints_nmds_annelids_presenceabsence.pdf",
      plot = q,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

Filter by site
```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input_metazoa = filter_taxa_from_input(input, taxa_to_keep = 'Metazoa')
MLBS = filter_data(input_metazoa, 'Site', keep_vals = 'MLBS')
HARV = filter_data(input_metazoa, 'Site', keep_vals = 'HARV')
WOOD = filter_data(input_metazoa, 'Site', keep_vals = 'WOOD')
LAJA = filter_data(input_metazoa, 'Site', keep_vals = 'LAJA')
UNDE = filter_data(input_metazoa, 'Site', keep_vals = 'UNDE')
SRER = filter_data(input_metazoa, 'Site', keep_vals = 'SRER')
DSNY = filter_data(input_metazoa, 'Site', keep_vals = 'DSNY')
BONA = filter_data(input_metazoa, 'Site', keep_vals = 'BONA')
GUAN = filter_data(input_metazoa, 'Site', keep_vals = 'GUAN')
ABBY = filter_data(input_metazoa, 'Site', keep_vals = 'ABBY')
BARR = filter_data(input_metazoa, 'Site', keep_vals = 'BARR')
BART = filter_data(input_metazoa, 'Site', keep_vals = 'BART')
BLAN = filter_data(input_metazoa, 'Site', keep_vals = 'BLAN')
CLBJ = filter_data(input_metazoa, 'Site', keep_vals = 'CLBJ')
CPER = filter_data(input_metazoa, 'Site', keep_vals = 'CPER')
DCFS = filter_data(input_metazoa, 'Site', keep_vals = 'DCFS')
DEJU = filter_data(input_metazoa, 'Site', keep_vals = 'DEJU')
DELA = filter_data(input_metazoa, 'Site', keep_vals = 'DELA')
GRSM = filter_data(input_metazoa, 'Site', keep_vals = 'GRSM')
HEAL = filter_data(input_metazoa, 'Site', keep_vals = 'HEAL')
JERC = filter_data(input_metazoa, 'Site', keep_vals = 'JERC')
JORN = filter_data(input_metazoa, 'Site', keep_vals = 'JORN')
KONA = filter_data(input_metazoa, 'Site', keep_vals = 'KONA')
KONZ = filter_data(input_metazoa, 'Site', keep_vals = 'KONZ')
LENO = filter_data(input_metazoa, 'Site', keep_vals = 'LENO')
MOAB = filter_data(input_metazoa, 'Site', keep_vals = 'MOAB')
NIWO = filter_data(input_metazoa, 'Site', keep_vals = 'NIWO')
OAES = filter_data(input_metazoa, 'Site', keep_vals = 'OAES')
ONAQ = filter_data(input_metazoa, 'Site', keep_vals = 'ONAQ')
ORNL = filter_data(input_metazoa, 'Site', keep_vals = 'ORNL')
OSBS = filter_data(input_metazoa, 'Site', keep_vals = 'OSBS')
PUUM = filter_data(input_metazoa, 'Site', keep_vals = 'PUUM')
RMNP = filter_data(input_metazoa, 'Site', keep_vals = 'RMNP')
SCBI = filter_data(input_metazoa, 'Site', keep_vals = 'SCBI')
SERC = filter_data(input_metazoa, 'Site', keep_vals = 'SERC')
SJER = filter_data(input_metazoa, 'Site', keep_vals = 'SJER')
SOAP = filter_data(input_metazoa, 'Site', keep_vals = 'SOAP')
STEI = filter_data(input_metazoa, 'Site', keep_vals = 'STEI')
STER = filter_data(input_metazoa, 'Site', keep_vals = 'STER')
TALL = filter_data(input_metazoa, 'Site', keep_vals = 'TALL')
TOOL = filter_data(input_metazoa, 'Site', keep_vals = 'TOOL')
TREE = filter_data(input_metazoa, 'Site', keep_vals = 'TREE')
UKFS = filter_data(input_metazoa, 'Site', keep_vals = 'UKFS')
WREF = filter_data(input_metazoa, 'Site', keep_vals = 'WREF')
YELL = filter_data(input_metazoa, 'Site', keep_vals = 'YELL')
```


```{r}
otuTAB<-ABBY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ABBY$data_loaded<-t(hell)

otuTAB<-BARR$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BARR$data_loaded<-t(hell)

otuTAB<-BART$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BART$data_loaded<-t(hell)

otuTAB<-BLAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BLAN$data_loaded<-t(hell)

otuTAB<-BONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BONA$data_loaded<-t(hell)

otuTAB<-CLBJ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CLBJ$data_loaded<-t(hell)

otuTAB<-CPER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CPER$data_loaded<-t(hell)

otuTAB<-DCFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DCFS$data_loaded<-t(hell)

otuTAB<-DELA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DELA$data_loaded<-t(hell)

otuTAB<-DSNY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DSNY$data_loaded<-t(hell)

otuTAB<-GRSM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GRSM$data_loaded<-t(hell)

otuTAB<-GUAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GUAN$data_loaded<-t(hell)

otuTAB<-HARV$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HARV$data_loaded<-t(hell)

otuTAB<-HEAL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HEAL$data_loaded<-t(hell)

otuTAB<-JERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JERC$data_loaded<-t(hell)

otuTAB<-JORN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JORN$data_loaded<-t(hell)

otuTAB<-KONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONA$data_loaded<-t(hell)

otuTAB<-KONZ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONZ$data_loaded<-t(hell)

otuTAB<-LAJA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LAJA$data_loaded<-t(hell)

otuTAB<-LENO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LENO$data_loaded<-t(hell)

otuTAB<-MLBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MLBS$data_loaded<-t(hell)

otuTAB<-MOAB$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MOAB$data_loaded<-t(hell)

otuTAB<-NIWO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
NIWO$data_loaded<-t(hell)

otuTAB<-OAES$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OAES$data_loaded<-t(hell)

otuTAB<-ONAQ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ONAQ$data_loaded<-t(hell)

otuTAB<-ORNL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ORNL$data_loaded<-t(hell)

otuTAB<-OSBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OSBS$data_loaded<-t(hell)

otuTAB<-PUUM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
PUUM$data_loaded<-t(hell)

otuTAB<-RMNP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
RMNP$data_loaded<-t(hell)

otuTAB<-SCBI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SCBI$data_loaded<-t(hell)

otuTAB<-SERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SERC$data_loaded<-t(hell)

otuTAB<-SJER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SJER$data_loaded<-t(hell)

otuTAB<-SOAP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SOAP$data_loaded<-t(hell)

otuTAB<-SRER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SRER$data_loaded<-t(hell)

otuTAB<-STEI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STEI$data_loaded<-t(hell)

otuTAB<-STER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STER$data_loaded<-t(hell)

otuTAB<-TALL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TALL$data_loaded<-t(hell)

otuTAB<-TOOL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TOOL$data_loaded<-t(hell)

otuTAB<-TREE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TREE$data_loaded<-t(hell)

otuTAB<-UKFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UKFS$data_loaded<-t(hell)

otuTAB<-UNDE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UNDE$data_loaded<-t(hell)

otuTAB<-WOOD$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WOOD$data_loaded<-t(hell)

otuTAB<-WREF$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WREF$data_loaded<-t(hell)

otuTAB<-YELL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
YELL$data_loaded<-t(hell)
```

```{r}
library(reshape2)
ABBY_tax_sum_phylum = summarize_taxonomy(ABBY, level = 4, report_higher_tax = FALSE)
ABBY_tax_sum_phylum$Phylum <-row.names(ABBY_tax_sum_phylum)
ABBY_melt<- melt(ABBY_tax_sum_phylum, id.var="Phylum")
site<-rep("ABBY",times=nrow(ABBY_melt))
ABBY_melt<-cbind(ABBY_melt,site)

BARR_tax_sum_phylum = summarize_taxonomy(BARR, level = 4, report_higher_tax = FALSE)
BARR_tax_sum_phylum$Phylum <-row.names(BARR_tax_sum_phylum)
BARR_melt<- melt(BARR_tax_sum_phylum, id.var="Phylum")
nrow(BARR_melt)
site<-rep("BARR",times=nrow(BARR_melt))
BARR_melt<-cbind(BARR_melt,site)

BART_tax_sum_phylum = summarize_taxonomy(BART, level = 4, report_higher_tax = FALSE)
BART_tax_sum_phylum$Phylum <-row.names(BART_tax_sum_phylum)
BART_melt<- melt(BART_tax_sum_phylum, id.var="Phylum")
nrow(BART_melt)
site<-rep("BART",times=nrow(BART_melt))
BART_melt<-cbind(BART_melt,site)

BLAN_tax_sum_phylum = summarize_taxonomy(BLAN, level = 4, report_higher_tax = FALSE)
BLAN_tax_sum_phylum$Phylum <-row.names(BLAN_tax_sum_phylum)
BLAN_melt<- melt(BLAN_tax_sum_phylum, id.var="Phylum")
nrow(BLAN_melt)
site<-rep("BLAN",times=nrow(BLAN_melt))
BLAN_melt<-cbind(BLAN_melt,site)

BONA_tax_sum_phylum = summarize_taxonomy(BONA, level = 4, report_higher_tax = FALSE)
BONA_tax_sum_phylum$Phylum <-row.names(BONA_tax_sum_phylum)
BONA_melt<- melt(BONA_tax_sum_phylum, id.var="Phylum")
nrow(BONA_melt)
site<-rep("BONA",times=nrow(BONA_melt))
BONA_melt<-cbind(BONA_melt,site)

CLBJ_tax_sum_phylum = summarize_taxonomy(CLBJ, level = 4, report_higher_tax = FALSE)
CLBJ_tax_sum_phylum$Phylum <-row.names(CLBJ_tax_sum_phylum)
CLBJ_melt<- melt(CLBJ_tax_sum_phylum, id.var="Phylum")
nrow(CLBJ_melt)
site<-rep("CLBJ",times=nrow(CLBJ_melt))
CLBJ_melt<-cbind(CLBJ_melt,site)

CPER_tax_sum_phylum = summarize_taxonomy(CPER, level = 4, report_higher_tax = FALSE)
CPER_tax_sum_phylum$Phylum <-row.names(CPER_tax_sum_phylum)
CPER_melt<- melt(CPER_tax_sum_phylum, id.var="Phylum")
nrow(CPER_melt)
site<-rep("CPER",times=nrow(CPER_melt))
CPER_melt<-cbind(CPER_melt,site)

DCFS_tax_sum_phylum = summarize_taxonomy(DCFS, level = 4, report_higher_tax = FALSE)
DCFS_tax_sum_phylum$Phylum <-row.names(DCFS_tax_sum_phylum)
DCFS_melt<- melt(DCFS_tax_sum_phylum, id.var="Phylum")
nrow(DCFS_melt)
site<-rep("DCFS",times=nrow(DCFS_melt))
DCFS_melt<-cbind(DCFS_melt,site)

DEJU_tax_sum_phylum = summarize_taxonomy(DEJU, level = 4, report_higher_tax = FALSE)
DEJU_tax_sum_phylum$Phylum <-row.names(DEJU_tax_sum_phylum)
DEJU_melt<- melt(DEJU_tax_sum_phylum, id.var="Phylum")
nrow(DEJU_melt)
site<-rep("DEJU",times=nrow(DEJU_melt))
DEJU_melt<-cbind(DEJU_melt,site)

DELA_tax_sum_phylum = summarize_taxonomy(DELA, level = 4, report_higher_tax = FALSE)
DELA_tax_sum_phylum$Phylum <-row.names(DELA_tax_sum_phylum)
DELA_melt<- melt(DELA_tax_sum_phylum, id.var="Phylum")
nrow(DELA_melt)
site<-rep("DELA",times=nrow(DELA_melt))
DELA_melt<-cbind(DELA_melt,site)

DSNY_tax_sum_phylum = summarize_taxonomy(DSNY, level = 4, report_higher_tax = FALSE)
DSNY_tax_sum_phylum$Phylum <-row.names(DSNY_tax_sum_phylum)
DSNY_melt<- melt(DSNY_tax_sum_phylum, id.var="Phylum")
nrow(DSNY_melt)
site<-rep("DSNY",times=nrow(DSNY_melt))
DSNY_melt<-cbind(DSNY_melt,site)

GRSM_tax_sum_phylum = summarize_taxonomy(GRSM, level = 4, report_higher_tax = FALSE)
GRSM_tax_sum_phylum$Phylum <-row.names(GRSM_tax_sum_phylum)
GRSM_melt<- melt(GRSM_tax_sum_phylum, id.var="Phylum")
nrow(GRSM_melt)
site<-rep("GRSM",times=nrow(GRSM_melt))
GRSM_melt<-cbind(GRSM_melt,site)

GUAN_tax_sum_phylum = summarize_taxonomy(GUAN, level = 4, report_higher_tax = FALSE)
GUAN_tax_sum_phylum$Phylum <-row.names(GUAN_tax_sum_phylum)
GUAN_melt<- melt(GUAN_tax_sum_phylum, id.var="Phylum")
nrow(GUAN_melt)
site<-rep("GUAN",times=nrow(GUAN_melt))
GUAN_melt<-cbind(GUAN_melt,site)

HARV_tax_sum_phylum = summarize_taxonomy(HARV, level = 4, report_higher_tax = FALSE)
HARV_tax_sum_phylum$Phylum <-row.names(HARV_tax_sum_phylum)
HARV_melt<- melt(HARV_tax_sum_phylum, id.var="Phylum")
nrow(HARV_melt)
site<-rep("HARV",times=nrow(HARV_melt))
HARV_melt<-cbind(HARV_melt,site)

HEAL_tax_sum_phylum = summarize_taxonomy(HEAL, level = 4, report_higher_tax = FALSE)
HEAL_tax_sum_phylum$Phylum <-row.names(HEAL_tax_sum_phylum)
HEAL_melt<- melt(HEAL_tax_sum_phylum, id.var="Phylum")
nrow(HEAL_melt)
site<-rep("HEAL",times=nrow(HEAL_melt))
HEAL_melt<-cbind(HEAL_melt,site)

JERC_tax_sum_phylum = summarize_taxonomy(JERC, level = 4, report_higher_tax = FALSE)
JERC_tax_sum_phylum$Phylum <-row.names(JERC_tax_sum_phylum)
JERC_melt<- melt(JERC_tax_sum_phylum, id.var="Phylum")
nrow(JERC_melt)
site<-rep("JERC",times=nrow(JERC_melt))
JERC_melt<-cbind(JERC_melt,site)

JORN_tax_sum_phylum = summarize_taxonomy(JORN, level = 4, report_higher_tax = FALSE)
JORN_tax_sum_phylum$Phylum <-row.names(JORN_tax_sum_phylum)
JORN_melt<- melt(JORN_tax_sum_phylum, id.var="Phylum")
nrow(JORN_melt)
site<-rep("JORN",times=nrow(JORN_melt))
JORN_melt<-cbind(JORN_melt,site)

KONA_tax_sum_phylum = summarize_taxonomy(KONA, level = 4, report_higher_tax = FALSE)
KONA_tax_sum_phylum$Phylum <-row.names(KONA_tax_sum_phylum)
KONA_melt<- melt(KONA_tax_sum_phylum, id.var="Phylum")
nrow(KONA_melt)
site<-rep("KONA",times=nrow(KONA_melt))
KONA_melt<-cbind(KONA_melt,site)

KONZ_tax_sum_phylum = summarize_taxonomy(KONZ, level = 4, report_higher_tax = FALSE)
KONZ_tax_sum_phylum$Phylum <-row.names(KONZ_tax_sum_phylum)
KONZ_melt<- melt(KONZ_tax_sum_phylum, id.var="Phylum")
nrow(KONZ_melt)
site<-rep("KONZ",times=nrow(KONZ_melt))
KONZ_melt<-cbind(KONZ_melt,site)

LAJA_tax_sum_phylum = summarize_taxonomy(LAJA, level = 4, report_higher_tax = FALSE)
LAJA_tax_sum_phylum$Phylum <-row.names(LAJA_tax_sum_phylum)
LAJA_melt<- melt(LAJA_tax_sum_phylum, id.var="Phylum")
nrow(LAJA_melt)
site<-rep("LAJA",times=nrow(LAJA_melt))
LAJA_melt<-cbind(LAJA_melt,site)

LENO_tax_sum_phylum = summarize_taxonomy(LENO, level = 4, report_higher_tax = FALSE)
LENO_tax_sum_phylum$Phylum <-row.names(LENO_tax_sum_phylum)
LENO_melt<- melt(LENO_tax_sum_phylum, id.var="Phylum")
nrow(LENO_melt)
site<-rep("LENO",times=nrow(LENO_melt))
LENO_melt<-cbind(LENO_melt,site)

MLBS_tax_sum_phylum = summarize_taxonomy(MLBS, level = 4, report_higher_tax = FALSE)
MLBS_tax_sum_phylum$Phylum <-row.names(MLBS_tax_sum_phylum)
MLBS_melt<- melt(MLBS_tax_sum_phylum, id.var="Phylum")
nrow(MLBS_melt)
site<-rep("MLBS",times=nrow(MLBS_melt))
MLBS_melt<-cbind(MLBS_melt,site)

MOAB_tax_sum_phylum = summarize_taxonomy(MOAB, level = 4, report_higher_tax = FALSE)
MOAB_tax_sum_phylum$Phylum <-row.names(MOAB_tax_sum_phylum)
MOAB_melt<- melt(MOAB_tax_sum_phylum, id.var="Phylum")
nrow(MOAB_melt)
site<-rep("MOAB",times=nrow(MOAB_melt))
MOAB_melt<-cbind(MOAB_melt,site)

NIWO_tax_sum_phylum = summarize_taxonomy(NIWO, level = 4, report_higher_tax = FALSE)
NIWO_tax_sum_phylum$Phylum <-row.names(NIWO_tax_sum_phylum)
NIWO_melt<- melt(NIWO_tax_sum_phylum, id.var="Phylum")
nrow(NIWO_melt)
site<-rep("NIWO",times=nrow(NIWO_melt))
NIWO_melt<-cbind(NIWO_melt,site)

OAES_tax_sum_phylum = summarize_taxonomy(OAES, level = 4, report_higher_tax = FALSE)
OAES_tax_sum_phylum$Phylum <-row.names(OAES_tax_sum_phylum)
OAES_melt<- melt(OAES_tax_sum_phylum, id.var="Phylum")
nrow(OAES_melt)
site<-rep("OAES",times=nrow(OAES_melt))
OAES_melt<-cbind(OAES_melt,site)

ONAQ_tax_sum_phylum = summarize_taxonomy(ONAQ, level = 4, report_higher_tax = FALSE)
ONAQ_tax_sum_phylum$Phylum <-row.names(ONAQ_tax_sum_phylum)
ONAQ_melt<- melt(ONAQ_tax_sum_phylum, id.var="Phylum")
nrow(ONAQ_melt)
site<-rep("ONAQ",times=nrow(ONAQ_melt))
ONAQ_melt<-cbind(ONAQ_melt,site)

ORNL_tax_sum_phylum = summarize_taxonomy(ORNL, level = 4, report_higher_tax = FALSE)
ORNL_tax_sum_phylum$Phylum <-row.names(ORNL_tax_sum_phylum)
ORNL_melt<- melt(ORNL_tax_sum_phylum, id.var="Phylum")
nrow(ORNL_melt)
site<-rep("ORNL",times=nrow(ORNL_melt))
ORNL_melt<-cbind(ORNL_melt,site)

OSBS_tax_sum_phylum = summarize_taxonomy(OSBS, level = 4, report_higher_tax = FALSE)
OSBS_tax_sum_phylum$Phylum <-row.names(OSBS_tax_sum_phylum)
OSBS_melt<- melt(OSBS_tax_sum_phylum, id.var="Phylum")
nrow(OSBS_melt)
site<-rep("OSBS",times=nrow(OSBS_melt))
OSBS_melt<-cbind(OSBS_melt,site)

PUUM_tax_sum_phylum = summarize_taxonomy(PUUM, level = 4, report_higher_tax = FALSE)
PUUM_tax_sum_phylum$Phylum <-row.names(PUUM_tax_sum_phylum)
PUUM_melt<- melt(PUUM_tax_sum_phylum, id.var="Phylum")
nrow(PUUM_melt)
site<-rep("PUUM",times=nrow(PUUM_melt))
PUUM_melt<-cbind(PUUM_melt,site)

RMNP_tax_sum_phylum = summarize_taxonomy(RMNP, level = 4, report_higher_tax = FALSE)
RMNP_tax_sum_phylum$Phylum <-row.names(RMNP_tax_sum_phylum)
RMNP_melt<- melt(RMNP_tax_sum_phylum, id.var="Phylum")
nrow(RMNP_melt)
site<-rep("RMNP",times=nrow(RMNP_melt))
RMNP_melt<-cbind(RMNP_melt,site)

SCBI_tax_sum_phylum = summarize_taxonomy(SCBI, level = 4, report_higher_tax = FALSE)
SCBI_tax_sum_phylum$Phylum <-row.names(SCBI_tax_sum_phylum)
SCBI_melt<- melt(SCBI_tax_sum_phylum, id.var="Phylum")
nrow(SCBI_melt)
site<-rep("SCBI",times=nrow(SCBI_melt))
SCBI_melt<-cbind(SCBI_melt,site)

SERC_tax_sum_phylum = summarize_taxonomy(SERC, level = 4, report_higher_tax = FALSE)
SERC_tax_sum_phylum$Phylum <-row.names(SERC_tax_sum_phylum)
SERC_melt<- melt(SERC_tax_sum_phylum, id.var="Phylum")
nrow(SERC_melt)
site<-rep("SERC",times=nrow(SERC_melt))
SERC_melt<-cbind(SERC_melt,site)

SJER_tax_sum_phylum = summarize_taxonomy(SJER, level = 4, report_higher_tax = FALSE)
SJER_tax_sum_phylum$Phylum <-row.names(SJER_tax_sum_phylum)
SJER_melt<- melt(SJER_tax_sum_phylum, id.var="Phylum")
nrow(SJER_melt)
site<-rep("SJER",times=nrow(SJER_melt))
SJER_melt<-cbind(SJER_melt,site)

SOAP_tax_sum_phylum = summarize_taxonomy(SOAP, level = 4, report_higher_tax = FALSE)
SOAP_tax_sum_phylum$Phylum <-row.names(SOAP_tax_sum_phylum)
SOAP_melt<- melt(SOAP_tax_sum_phylum, id.var="Phylum")
nrow(SOAP_melt)
site<-rep("SOAP",times=nrow(SOAP_melt))
SOAP_melt<-cbind(SOAP_melt,site)

SRER_tax_sum_phylum = summarize_taxonomy(SRER, level = 4, report_higher_tax = FALSE)
SRER_tax_sum_phylum$Phylum <-row.names(SRER_tax_sum_phylum)
SRER_melt<- melt(SRER_tax_sum_phylum, id.var="Phylum")
nrow(SRER_melt)
site<-rep("SRER",times=nrow(SRER_melt))
SRER_melt<-cbind(SRER_melt,site)

STEI_tax_sum_phylum = summarize_taxonomy(STEI, level = 4, report_higher_tax = FALSE)
STEI_tax_sum_phylum$Phylum <-row.names(STEI_tax_sum_phylum)
STEI_melt<- melt(STEI_tax_sum_phylum, id.var="Phylum")
nrow(STEI_melt)
site<-rep("STEI",times=nrow(STEI_melt))
STEI_melt<-cbind(STEI_melt,site)

STER_tax_sum_phylum = summarize_taxonomy(STER, level = 4, report_higher_tax = FALSE)
STER_tax_sum_phylum$Phylum <-row.names(STER_tax_sum_phylum)
STER_melt<- melt(STER_tax_sum_phylum, id.var="Phylum")
nrow(STER_melt)
site<-rep("STER",times=nrow(STER_melt))
STER_melt<-cbind(STER_melt,site)

TALL_tax_sum_phylum = summarize_taxonomy(TALL, level = 4, report_higher_tax = FALSE)
TALL_tax_sum_phylum$Phylum <-row.names(TALL_tax_sum_phylum)
TALL_melt<- melt(TALL_tax_sum_phylum, id.var="Phylum")
nrow(TALL_melt)
site<-rep("TALL",times=nrow(TALL_melt))
TALL_melt<-cbind(TALL_melt,site)

TOOL_tax_sum_phylum = summarize_taxonomy(TOOL, level = 4, report_higher_tax = FALSE)
TOOL_tax_sum_phylum$Phylum <-row.names(TOOL_tax_sum_phylum)
TOOL_melt<- melt(TOOL_tax_sum_phylum, id.var="Phylum")
nrow(TOOL_melt)
site<-rep("TOOL",times=nrow(TOOL_melt))
TOOL_melt<-cbind(TOOL_melt,site)

TREE_tax_sum_phylum = summarize_taxonomy(TREE, level = 4, report_higher_tax = FALSE)
TREE_tax_sum_phylum$Phylum <-row.names(TREE_tax_sum_phylum)
TREE_melt<- melt(TREE_tax_sum_phylum, id.var="Phylum")
nrow(TREE_melt)
site<-rep("TREE",times=nrow(TREE_melt))
TREE_melt<-cbind(TREE_melt,site)

UKFS_tax_sum_phylum = summarize_taxonomy(UKFS, level = 4, report_higher_tax = FALSE)
UKFS_tax_sum_phylum$Phylum <-row.names(UKFS_tax_sum_phylum)
UKFS_melt<- melt(UKFS_tax_sum_phylum, id.var="Phylum")
nrow(UKFS_melt)
site<-rep("UKFS",times=nrow(UKFS_melt))
UKFS_melt<-cbind(UKFS_melt,site)

UNDE_tax_sum_phylum = summarize_taxonomy(UNDE, level = 4, report_higher_tax = FALSE)
UNDE_tax_sum_phylum$Phylum <-row.names(UNDE_tax_sum_phylum)
UNDE_melt<- melt(UNDE_tax_sum_phylum, id.var="Phylum")
nrow(UNDE_melt)
site<-rep("UNDE",times=nrow(UNDE_melt))
UNDE_melt<-cbind(UNDE_melt,site)

WOOD_tax_sum_phylum = summarize_taxonomy(WOOD, level = 4, report_higher_tax = FALSE)
WOOD_tax_sum_phylum$Phylum <-row.names(WOOD_tax_sum_phylum)
WOOD_melt<- melt(WOOD_tax_sum_phylum, id.var="Phylum")
nrow(WOOD_melt)
site<-rep("WOOD",times=nrow(WOOD_melt))
WOOD_melt<-cbind(WOOD_melt,site)

WREF_tax_sum_phylum = summarize_taxonomy(WREF, level = 4, report_higher_tax = FALSE)
WREF_tax_sum_phylum$Phylum <-row.names(WREF_tax_sum_phylum)
WREF_melt<- melt(WREF_tax_sum_phylum, id.var="Phylum")
nrow(WREF_melt)
site<-rep("WREF",times=nrow(WREF_melt))
WREF_melt<-cbind(WREF_melt,site)

YELL_tax_sum_phylum = summarize_taxonomy(YELL, level = 4, report_higher_tax = FALSE)
YELL_tax_sum_phylum$Phylum <-row.names(YELL_tax_sum_phylum)
YELL_melt<- melt(YELL_tax_sum_phylum, id.var="Phylum")
nrow(YELL_melt)
site<-rep("YELL",times=nrow(YELL_melt))
YELL_melt<-cbind(YELL_melt,site)
```

```{r}
phylumfinalbind<-do.call("rbind",list(ABBY_melt, BARR_melt, BART_melt, BLAN_melt, BONA_melt, CLBJ_melt, CPER_melt, DCFS_melt, DEJU_melt, DELA_melt, DSNY_melt, GRSM_melt, GUAN_melt, HARV_melt, HEAL_melt, JERC_melt, JORN_melt, KONA_melt, KONZ_melt, LAJA_melt, LENO_melt, MLBS_melt, MOAB_melt, NIWO_melt, OAES_melt, ONAQ_melt, ORNL_melt, OSBS_melt, PUUM_melt, RMNP_melt, SCBI_melt, SERC_melt, SJER_melt, SOAP_melt, SRER_melt, STEI_melt, STER_melt, TALL_melt, TOOL_melt, TREE_melt, UKFS_melt, UNDE_melt, WOOD_melt, WREF_melt, YELL_melt))
```

phylum graph
```{r}

b<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(23)) + 
  ggtitle("Phylum Graphs")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

b
```

```{r}

ggsave(filename = "phylum_bysite_metazoa_only.pdf",
      plot = b,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```


Filter by site
```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input_metazoa = filter_taxa_from_input(input, taxa_to_keep = 'Fungi')
MLBS = filter_data(input_metazoa, 'Site', keep_vals = 'MLBS')
HARV = filter_data(input_metazoa, 'Site', keep_vals = 'HARV')
WOOD = filter_data(input_metazoa, 'Site', keep_vals = 'WOOD')
LAJA = filter_data(input_metazoa, 'Site', keep_vals = 'LAJA')
UNDE = filter_data(input_metazoa, 'Site', keep_vals = 'UNDE')
SRER = filter_data(input_metazoa, 'Site', keep_vals = 'SRER')
DSNY = filter_data(input_metazoa, 'Site', keep_vals = 'DSNY')
BONA = filter_data(input_metazoa, 'Site', keep_vals = 'BONA')
GUAN = filter_data(input_metazoa, 'Site', keep_vals = 'GUAN')
ABBY = filter_data(input_metazoa, 'Site', keep_vals = 'ABBY')
BARR = filter_data(input_metazoa, 'Site', keep_vals = 'BARR')
BART = filter_data(input_metazoa, 'Site', keep_vals = 'BART')
BLAN = filter_data(input_metazoa, 'Site', keep_vals = 'BLAN')
CLBJ = filter_data(input_metazoa, 'Site', keep_vals = 'CLBJ')
CPER = filter_data(input_metazoa, 'Site', keep_vals = 'CPER')
DCFS = filter_data(input_metazoa, 'Site', keep_vals = 'DCFS')
DEJU = filter_data(input_metazoa, 'Site', keep_vals = 'DEJU')
DELA = filter_data(input_metazoa, 'Site', keep_vals = 'DELA')
GRSM = filter_data(input_metazoa, 'Site', keep_vals = 'GRSM')
HEAL = filter_data(input_metazoa, 'Site', keep_vals = 'HEAL')
JERC = filter_data(input_metazoa, 'Site', keep_vals = 'JERC')
JORN = filter_data(input_metazoa, 'Site', keep_vals = 'JORN')
KONA = filter_data(input_metazoa, 'Site', keep_vals = 'KONA')
KONZ = filter_data(input_metazoa, 'Site', keep_vals = 'KONZ')
LENO = filter_data(input_metazoa, 'Site', keep_vals = 'LENO')
MOAB = filter_data(input_metazoa, 'Site', keep_vals = 'MOAB')
NIWO = filter_data(input_metazoa, 'Site', keep_vals = 'NIWO')
OAES = filter_data(input_metazoa, 'Site', keep_vals = 'OAES')
ONAQ = filter_data(input_metazoa, 'Site', keep_vals = 'ONAQ')
ORNL = filter_data(input_metazoa, 'Site', keep_vals = 'ORNL')
OSBS = filter_data(input_metazoa, 'Site', keep_vals = 'OSBS')
PUUM = filter_data(input_metazoa, 'Site', keep_vals = 'PUUM')
RMNP = filter_data(input_metazoa, 'Site', keep_vals = 'RMNP')
SCBI = filter_data(input_metazoa, 'Site', keep_vals = 'SCBI')
SERC = filter_data(input_metazoa, 'Site', keep_vals = 'SERC')
SJER = filter_data(input_metazoa, 'Site', keep_vals = 'SJER')
SOAP = filter_data(input_metazoa, 'Site', keep_vals = 'SOAP')
STEI = filter_data(input_metazoa, 'Site', keep_vals = 'STEI')
STER = filter_data(input_metazoa, 'Site', keep_vals = 'STER')
TALL = filter_data(input_metazoa, 'Site', keep_vals = 'TALL')
TOOL = filter_data(input_metazoa, 'Site', keep_vals = 'TOOL')
TREE = filter_data(input_metazoa, 'Site', keep_vals = 'TREE')
UKFS = filter_data(input_metazoa, 'Site', keep_vals = 'UKFS')
WREF = filter_data(input_metazoa, 'Site', keep_vals = 'WREF')
YELL = filter_data(input_metazoa, 'Site', keep_vals = 'YELL')
```


```{r}
otuTAB<-ABBY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ABBY$data_loaded<-t(hell)

otuTAB<-BARR$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BARR$data_loaded<-t(hell)

otuTAB<-BART$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BART$data_loaded<-t(hell)

otuTAB<-BLAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BLAN$data_loaded<-t(hell)

otuTAB<-BONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BONA$data_loaded<-t(hell)

otuTAB<-CLBJ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CLBJ$data_loaded<-t(hell)

otuTAB<-CPER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CPER$data_loaded<-t(hell)

otuTAB<-DCFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DCFS$data_loaded<-t(hell)

otuTAB<-DELA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DELA$data_loaded<-t(hell)

otuTAB<-DSNY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DSNY$data_loaded<-t(hell)

otuTAB<-GRSM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GRSM$data_loaded<-t(hell)

otuTAB<-GUAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GUAN$data_loaded<-t(hell)

otuTAB<-HARV$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HARV$data_loaded<-t(hell)

otuTAB<-HEAL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HEAL$data_loaded<-t(hell)

otuTAB<-JERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JERC$data_loaded<-t(hell)

otuTAB<-JORN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JORN$data_loaded<-t(hell)

otuTAB<-KONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONA$data_loaded<-t(hell)

otuTAB<-KONZ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONZ$data_loaded<-t(hell)

otuTAB<-LAJA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LAJA$data_loaded<-t(hell)

otuTAB<-LENO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LENO$data_loaded<-t(hell)

otuTAB<-MLBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MLBS$data_loaded<-t(hell)

otuTAB<-MOAB$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MOAB$data_loaded<-t(hell)

otuTAB<-NIWO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
NIWO$data_loaded<-t(hell)

otuTAB<-OAES$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OAES$data_loaded<-t(hell)

otuTAB<-ONAQ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ONAQ$data_loaded<-t(hell)

otuTAB<-ORNL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ORNL$data_loaded<-t(hell)

otuTAB<-OSBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OSBS$data_loaded<-t(hell)

otuTAB<-PUUM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
PUUM$data_loaded<-t(hell)

otuTAB<-RMNP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
RMNP$data_loaded<-t(hell)

otuTAB<-SCBI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SCBI$data_loaded<-t(hell)

otuTAB<-SERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SERC$data_loaded<-t(hell)

otuTAB<-SJER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SJER$data_loaded<-t(hell)

otuTAB<-SOAP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SOAP$data_loaded<-t(hell)

otuTAB<-SRER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SRER$data_loaded<-t(hell)

otuTAB<-STEI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STEI$data_loaded<-t(hell)

otuTAB<-STER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STER$data_loaded<-t(hell)

otuTAB<-TALL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TALL$data_loaded<-t(hell)

otuTAB<-TOOL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TOOL$data_loaded<-t(hell)

otuTAB<-TREE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TREE$data_loaded<-t(hell)

otuTAB<-UKFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UKFS$data_loaded<-t(hell)

otuTAB<-UNDE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UNDE$data_loaded<-t(hell)

otuTAB<-WOOD$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WOOD$data_loaded<-t(hell)

otuTAB<-WREF$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WREF$data_loaded<-t(hell)

otuTAB<-YELL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
YELL$data_loaded<-t(hell)
```

```{r}
library(reshape2)
ABBY_tax_sum_phylum = summarize_taxonomy(ABBY, level = 4, report_higher_tax = FALSE)
ABBY_tax_sum_phylum$Phylum <-row.names(ABBY_tax_sum_phylum)
ABBY_melt<- melt(ABBY_tax_sum_phylum, id.var="Phylum")
site<-rep("ABBY",times=nrow(ABBY_melt))
ABBY_melt<-cbind(ABBY_melt,site)

BARR_tax_sum_phylum = summarize_taxonomy(BARR, level = 4, report_higher_tax = FALSE)
BARR_tax_sum_phylum$Phylum <-row.names(BARR_tax_sum_phylum)
BARR_melt<- melt(BARR_tax_sum_phylum, id.var="Phylum")
nrow(BARR_melt)
site<-rep("BARR",times=nrow(BARR_melt))
BARR_melt<-cbind(BARR_melt,site)

BART_tax_sum_phylum = summarize_taxonomy(BART, level = 4, report_higher_tax = FALSE)
BART_tax_sum_phylum$Phylum <-row.names(BART_tax_sum_phylum)
BART_melt<- melt(BART_tax_sum_phylum, id.var="Phylum")
nrow(BART_melt)
site<-rep("BART",times=nrow(BART_melt))
BART_melt<-cbind(BART_melt,site)

BLAN_tax_sum_phylum = summarize_taxonomy(BLAN, level = 4, report_higher_tax = FALSE)
BLAN_tax_sum_phylum$Phylum <-row.names(BLAN_tax_sum_phylum)
BLAN_melt<- melt(BLAN_tax_sum_phylum, id.var="Phylum")
nrow(BLAN_melt)
site<-rep("BLAN",times=nrow(BLAN_melt))
BLAN_melt<-cbind(BLAN_melt,site)

BONA_tax_sum_phylum = summarize_taxonomy(BONA, level = 4, report_higher_tax = FALSE)
BONA_tax_sum_phylum$Phylum <-row.names(BONA_tax_sum_phylum)
BONA_melt<- melt(BONA_tax_sum_phylum, id.var="Phylum")
nrow(BONA_melt)
site<-rep("BONA",times=nrow(BONA_melt))
BONA_melt<-cbind(BONA_melt,site)

CLBJ_tax_sum_phylum = summarize_taxonomy(CLBJ, level = 4, report_higher_tax = FALSE)
CLBJ_tax_sum_phylum$Phylum <-row.names(CLBJ_tax_sum_phylum)
CLBJ_melt<- melt(CLBJ_tax_sum_phylum, id.var="Phylum")
nrow(CLBJ_melt)
site<-rep("CLBJ",times=nrow(CLBJ_melt))
CLBJ_melt<-cbind(CLBJ_melt,site)

CPER_tax_sum_phylum = summarize_taxonomy(CPER, level = 4, report_higher_tax = FALSE)
CPER_tax_sum_phylum$Phylum <-row.names(CPER_tax_sum_phylum)
CPER_melt<- melt(CPER_tax_sum_phylum, id.var="Phylum")
nrow(CPER_melt)
site<-rep("CPER",times=nrow(CPER_melt))
CPER_melt<-cbind(CPER_melt,site)

DCFS_tax_sum_phylum = summarize_taxonomy(DCFS, level = 4, report_higher_tax = FALSE)
DCFS_tax_sum_phylum$Phylum <-row.names(DCFS_tax_sum_phylum)
DCFS_melt<- melt(DCFS_tax_sum_phylum, id.var="Phylum")
nrow(DCFS_melt)
site<-rep("DCFS",times=nrow(DCFS_melt))
DCFS_melt<-cbind(DCFS_melt,site)

DEJU_tax_sum_phylum = summarize_taxonomy(DEJU, level = 4, report_higher_tax = FALSE)
DEJU_tax_sum_phylum$Phylum <-row.names(DEJU_tax_sum_phylum)
DEJU_melt<- melt(DEJU_tax_sum_phylum, id.var="Phylum")
nrow(DEJU_melt)
site<-rep("DEJU",times=nrow(DEJU_melt))
DEJU_melt<-cbind(DEJU_melt,site)

DELA_tax_sum_phylum = summarize_taxonomy(DELA, level = 4, report_higher_tax = FALSE)
DELA_tax_sum_phylum$Phylum <-row.names(DELA_tax_sum_phylum)
DELA_melt<- melt(DELA_tax_sum_phylum, id.var="Phylum")
nrow(DELA_melt)
site<-rep("DELA",times=nrow(DELA_melt))
DELA_melt<-cbind(DELA_melt,site)

DSNY_tax_sum_phylum = summarize_taxonomy(DSNY, level = 4, report_higher_tax = FALSE)
DSNY_tax_sum_phylum$Phylum <-row.names(DSNY_tax_sum_phylum)
DSNY_melt<- melt(DSNY_tax_sum_phylum, id.var="Phylum")
nrow(DSNY_melt)
site<-rep("DSNY",times=nrow(DSNY_melt))
DSNY_melt<-cbind(DSNY_melt,site)

GRSM_tax_sum_phylum = summarize_taxonomy(GRSM, level = 4, report_higher_tax = FALSE)
GRSM_tax_sum_phylum$Phylum <-row.names(GRSM_tax_sum_phylum)
GRSM_melt<- melt(GRSM_tax_sum_phylum, id.var="Phylum")
nrow(GRSM_melt)
site<-rep("GRSM",times=nrow(GRSM_melt))
GRSM_melt<-cbind(GRSM_melt,site)

GUAN_tax_sum_phylum = summarize_taxonomy(GUAN, level = 4, report_higher_tax = FALSE)
GUAN_tax_sum_phylum$Phylum <-row.names(GUAN_tax_sum_phylum)
GUAN_melt<- melt(GUAN_tax_sum_phylum, id.var="Phylum")
nrow(GUAN_melt)
site<-rep("GUAN",times=nrow(GUAN_melt))
GUAN_melt<-cbind(GUAN_melt,site)

HARV_tax_sum_phylum = summarize_taxonomy(HARV, level = 4, report_higher_tax = FALSE)
HARV_tax_sum_phylum$Phylum <-row.names(HARV_tax_sum_phylum)
HARV_melt<- melt(HARV_tax_sum_phylum, id.var="Phylum")
nrow(HARV_melt)
site<-rep("HARV",times=nrow(HARV_melt))
HARV_melt<-cbind(HARV_melt,site)

HEAL_tax_sum_phylum = summarize_taxonomy(HEAL, level = 4, report_higher_tax = FALSE)
HEAL_tax_sum_phylum$Phylum <-row.names(HEAL_tax_sum_phylum)
HEAL_melt<- melt(HEAL_tax_sum_phylum, id.var="Phylum")
nrow(HEAL_melt)
site<-rep("HEAL",times=nrow(HEAL_melt))
HEAL_melt<-cbind(HEAL_melt,site)

JERC_tax_sum_phylum = summarize_taxonomy(JERC, level = 4, report_higher_tax = FALSE)
JERC_tax_sum_phylum$Phylum <-row.names(JERC_tax_sum_phylum)
JERC_melt<- melt(JERC_tax_sum_phylum, id.var="Phylum")
nrow(JERC_melt)
site<-rep("JERC",times=nrow(JERC_melt))
JERC_melt<-cbind(JERC_melt,site)

JORN_tax_sum_phylum = summarize_taxonomy(JORN, level = 4, report_higher_tax = FALSE)
JORN_tax_sum_phylum$Phylum <-row.names(JORN_tax_sum_phylum)
JORN_melt<- melt(JORN_tax_sum_phylum, id.var="Phylum")
nrow(JORN_melt)
site<-rep("JORN",times=nrow(JORN_melt))
JORN_melt<-cbind(JORN_melt,site)

KONA_tax_sum_phylum = summarize_taxonomy(KONA, level = 4, report_higher_tax = FALSE)
KONA_tax_sum_phylum$Phylum <-row.names(KONA_tax_sum_phylum)
KONA_melt<- melt(KONA_tax_sum_phylum, id.var="Phylum")
nrow(KONA_melt)
site<-rep("KONA",times=nrow(KONA_melt))
KONA_melt<-cbind(KONA_melt,site)

KONZ_tax_sum_phylum = summarize_taxonomy(KONZ, level = 4, report_higher_tax = FALSE)
KONZ_tax_sum_phylum$Phylum <-row.names(KONZ_tax_sum_phylum)
KONZ_melt<- melt(KONZ_tax_sum_phylum, id.var="Phylum")
nrow(KONZ_melt)
site<-rep("KONZ",times=nrow(KONZ_melt))
KONZ_melt<-cbind(KONZ_melt,site)

LAJA_tax_sum_phylum = summarize_taxonomy(LAJA, level = 4, report_higher_tax = FALSE)
LAJA_tax_sum_phylum$Phylum <-row.names(LAJA_tax_sum_phylum)
LAJA_melt<- melt(LAJA_tax_sum_phylum, id.var="Phylum")
nrow(LAJA_melt)
site<-rep("LAJA",times=nrow(LAJA_melt))
LAJA_melt<-cbind(LAJA_melt,site)

LENO_tax_sum_phylum = summarize_taxonomy(LENO, level = 4, report_higher_tax = FALSE)
LENO_tax_sum_phylum$Phylum <-row.names(LENO_tax_sum_phylum)
LENO_melt<- melt(LENO_tax_sum_phylum, id.var="Phylum")
nrow(LENO_melt)
site<-rep("LENO",times=nrow(LENO_melt))
LENO_melt<-cbind(LENO_melt,site)

MLBS_tax_sum_phylum = summarize_taxonomy(MLBS, level = 4, report_higher_tax = FALSE)
MLBS_tax_sum_phylum$Phylum <-row.names(MLBS_tax_sum_phylum)
MLBS_melt<- melt(MLBS_tax_sum_phylum, id.var="Phylum")
nrow(MLBS_melt)
site<-rep("MLBS",times=nrow(MLBS_melt))
MLBS_melt<-cbind(MLBS_melt,site)

MOAB_tax_sum_phylum = summarize_taxonomy(MOAB, level = 4, report_higher_tax = FALSE)
MOAB_tax_sum_phylum$Phylum <-row.names(MOAB_tax_sum_phylum)
MOAB_melt<- melt(MOAB_tax_sum_phylum, id.var="Phylum")
nrow(MOAB_melt)
site<-rep("MOAB",times=nrow(MOAB_melt))
MOAB_melt<-cbind(MOAB_melt,site)

NIWO_tax_sum_phylum = summarize_taxonomy(NIWO, level = 4, report_higher_tax = FALSE)
NIWO_tax_sum_phylum$Phylum <-row.names(NIWO_tax_sum_phylum)
NIWO_melt<- melt(NIWO_tax_sum_phylum, id.var="Phylum")
nrow(NIWO_melt)
site<-rep("NIWO",times=nrow(NIWO_melt))
NIWO_melt<-cbind(NIWO_melt,site)

OAES_tax_sum_phylum = summarize_taxonomy(OAES, level = 4, report_higher_tax = FALSE)
OAES_tax_sum_phylum$Phylum <-row.names(OAES_tax_sum_phylum)
OAES_melt<- melt(OAES_tax_sum_phylum, id.var="Phylum")
nrow(OAES_melt)
site<-rep("OAES",times=nrow(OAES_melt))
OAES_melt<-cbind(OAES_melt,site)

ONAQ_tax_sum_phylum = summarize_taxonomy(ONAQ, level = 4, report_higher_tax = FALSE)
ONAQ_tax_sum_phylum$Phylum <-row.names(ONAQ_tax_sum_phylum)
ONAQ_melt<- melt(ONAQ_tax_sum_phylum, id.var="Phylum")
nrow(ONAQ_melt)
site<-rep("ONAQ",times=nrow(ONAQ_melt))
ONAQ_melt<-cbind(ONAQ_melt,site)

ORNL_tax_sum_phylum = summarize_taxonomy(ORNL, level = 4, report_higher_tax = FALSE)
ORNL_tax_sum_phylum$Phylum <-row.names(ORNL_tax_sum_phylum)
ORNL_melt<- melt(ORNL_tax_sum_phylum, id.var="Phylum")
nrow(ORNL_melt)
site<-rep("ORNL",times=nrow(ORNL_melt))
ORNL_melt<-cbind(ORNL_melt,site)

OSBS_tax_sum_phylum = summarize_taxonomy(OSBS, level = 4, report_higher_tax = FALSE)
OSBS_tax_sum_phylum$Phylum <-row.names(OSBS_tax_sum_phylum)
OSBS_melt<- melt(OSBS_tax_sum_phylum, id.var="Phylum")
nrow(OSBS_melt)
site<-rep("OSBS",times=nrow(OSBS_melt))
OSBS_melt<-cbind(OSBS_melt,site)

PUUM_tax_sum_phylum = summarize_taxonomy(PUUM, level = 4, report_higher_tax = FALSE)
PUUM_tax_sum_phylum$Phylum <-row.names(PUUM_tax_sum_phylum)
PUUM_melt<- melt(PUUM_tax_sum_phylum, id.var="Phylum")
nrow(PUUM_melt)
site<-rep("PUUM",times=nrow(PUUM_melt))
PUUM_melt<-cbind(PUUM_melt,site)

RMNP_tax_sum_phylum = summarize_taxonomy(RMNP, level = 4, report_higher_tax = FALSE)
RMNP_tax_sum_phylum$Phylum <-row.names(RMNP_tax_sum_phylum)
RMNP_melt<- melt(RMNP_tax_sum_phylum, id.var="Phylum")
nrow(RMNP_melt)
site<-rep("RMNP",times=nrow(RMNP_melt))
RMNP_melt<-cbind(RMNP_melt,site)

SCBI_tax_sum_phylum = summarize_taxonomy(SCBI, level = 4, report_higher_tax = FALSE)
SCBI_tax_sum_phylum$Phylum <-row.names(SCBI_tax_sum_phylum)
SCBI_melt<- melt(SCBI_tax_sum_phylum, id.var="Phylum")
nrow(SCBI_melt)
site<-rep("SCBI",times=nrow(SCBI_melt))
SCBI_melt<-cbind(SCBI_melt,site)

SERC_tax_sum_phylum = summarize_taxonomy(SERC, level = 4, report_higher_tax = FALSE)
SERC_tax_sum_phylum$Phylum <-row.names(SERC_tax_sum_phylum)
SERC_melt<- melt(SERC_tax_sum_phylum, id.var="Phylum")
nrow(SERC_melt)
site<-rep("SERC",times=nrow(SERC_melt))
SERC_melt<-cbind(SERC_melt,site)

SJER_tax_sum_phylum = summarize_taxonomy(SJER, level = 4, report_higher_tax = FALSE)
SJER_tax_sum_phylum$Phylum <-row.names(SJER_tax_sum_phylum)
SJER_melt<- melt(SJER_tax_sum_phylum, id.var="Phylum")
nrow(SJER_melt)
site<-rep("SJER",times=nrow(SJER_melt))
SJER_melt<-cbind(SJER_melt,site)

SOAP_tax_sum_phylum = summarize_taxonomy(SOAP, level = 4, report_higher_tax = FALSE)
SOAP_tax_sum_phylum$Phylum <-row.names(SOAP_tax_sum_phylum)
SOAP_melt<- melt(SOAP_tax_sum_phylum, id.var="Phylum")
nrow(SOAP_melt)
site<-rep("SOAP",times=nrow(SOAP_melt))
SOAP_melt<-cbind(SOAP_melt,site)

SRER_tax_sum_phylum = summarize_taxonomy(SRER, level = 4, report_higher_tax = FALSE)
SRER_tax_sum_phylum$Phylum <-row.names(SRER_tax_sum_phylum)
SRER_melt<- melt(SRER_tax_sum_phylum, id.var="Phylum")
nrow(SRER_melt)
site<-rep("SRER",times=nrow(SRER_melt))
SRER_melt<-cbind(SRER_melt,site)

STEI_tax_sum_phylum = summarize_taxonomy(STEI, level = 4, report_higher_tax = FALSE)
STEI_tax_sum_phylum$Phylum <-row.names(STEI_tax_sum_phylum)
STEI_melt<- melt(STEI_tax_sum_phylum, id.var="Phylum")
nrow(STEI_melt)
site<-rep("STEI",times=nrow(STEI_melt))
STEI_melt<-cbind(STEI_melt,site)

STER_tax_sum_phylum = summarize_taxonomy(STER, level = 4, report_higher_tax = FALSE)
STER_tax_sum_phylum$Phylum <-row.names(STER_tax_sum_phylum)
STER_melt<- melt(STER_tax_sum_phylum, id.var="Phylum")
nrow(STER_melt)
site<-rep("STER",times=nrow(STER_melt))
STER_melt<-cbind(STER_melt,site)

TALL_tax_sum_phylum = summarize_taxonomy(TALL, level = 4, report_higher_tax = FALSE)
TALL_tax_sum_phylum$Phylum <-row.names(TALL_tax_sum_phylum)
TALL_melt<- melt(TALL_tax_sum_phylum, id.var="Phylum")
nrow(TALL_melt)
site<-rep("TALL",times=nrow(TALL_melt))
TALL_melt<-cbind(TALL_melt,site)

TOOL_tax_sum_phylum = summarize_taxonomy(TOOL, level = 4, report_higher_tax = FALSE)
TOOL_tax_sum_phylum$Phylum <-row.names(TOOL_tax_sum_phylum)
TOOL_melt<- melt(TOOL_tax_sum_phylum, id.var="Phylum")
nrow(TOOL_melt)
site<-rep("TOOL",times=nrow(TOOL_melt))
TOOL_melt<-cbind(TOOL_melt,site)

TREE_tax_sum_phylum = summarize_taxonomy(TREE, level = 4, report_higher_tax = FALSE)
TREE_tax_sum_phylum$Phylum <-row.names(TREE_tax_sum_phylum)
TREE_melt<- melt(TREE_tax_sum_phylum, id.var="Phylum")
nrow(TREE_melt)
site<-rep("TREE",times=nrow(TREE_melt))
TREE_melt<-cbind(TREE_melt,site)

UKFS_tax_sum_phylum = summarize_taxonomy(UKFS, level = 4, report_higher_tax = FALSE)
UKFS_tax_sum_phylum$Phylum <-row.names(UKFS_tax_sum_phylum)
UKFS_melt<- melt(UKFS_tax_sum_phylum, id.var="Phylum")
nrow(UKFS_melt)
site<-rep("UKFS",times=nrow(UKFS_melt))
UKFS_melt<-cbind(UKFS_melt,site)

UNDE_tax_sum_phylum = summarize_taxonomy(UNDE, level = 4, report_higher_tax = FALSE)
UNDE_tax_sum_phylum$Phylum <-row.names(UNDE_tax_sum_phylum)
UNDE_melt<- melt(UNDE_tax_sum_phylum, id.var="Phylum")
nrow(UNDE_melt)
site<-rep("UNDE",times=nrow(UNDE_melt))
UNDE_melt<-cbind(UNDE_melt,site)

WOOD_tax_sum_phylum = summarize_taxonomy(WOOD, level = 4, report_higher_tax = FALSE)
WOOD_tax_sum_phylum$Phylum <-row.names(WOOD_tax_sum_phylum)
WOOD_melt<- melt(WOOD_tax_sum_phylum, id.var="Phylum")
nrow(WOOD_melt)
site<-rep("WOOD",times=nrow(WOOD_melt))
WOOD_melt<-cbind(WOOD_melt,site)

WREF_tax_sum_phylum = summarize_taxonomy(WREF, level = 4, report_higher_tax = FALSE)
WREF_tax_sum_phylum$Phylum <-row.names(WREF_tax_sum_phylum)
WREF_melt<- melt(WREF_tax_sum_phylum, id.var="Phylum")
nrow(WREF_melt)
site<-rep("WREF",times=nrow(WREF_melt))
WREF_melt<-cbind(WREF_melt,site)

YELL_tax_sum_phylum = summarize_taxonomy(YELL, level = 4, report_higher_tax = FALSE)
YELL_tax_sum_phylum$Phylum <-row.names(YELL_tax_sum_phylum)
YELL_melt<- melt(YELL_tax_sum_phylum, id.var="Phylum")
nrow(YELL_melt)
site<-rep("YELL",times=nrow(YELL_melt))
YELL_melt<-cbind(YELL_melt,site)
```

```{r}
phylumfinalbind<-do.call("rbind",list(ABBY_melt, BARR_melt, BART_melt, BLAN_melt, BONA_melt, CLBJ_melt, CPER_melt, DCFS_melt, DEJU_melt, DELA_melt, DSNY_melt, GRSM_melt, GUAN_melt, HARV_melt, HEAL_melt, JERC_melt, JORN_melt, KONA_melt, KONZ_melt, LAJA_melt, LENO_melt, MLBS_melt, MOAB_melt, NIWO_melt, OAES_melt, ONAQ_melt, ORNL_melt, OSBS_melt, PUUM_melt, RMNP_melt, SCBI_melt, SERC_melt, SJER_melt, SOAP_melt, SRER_melt, STEI_melt, STER_melt, TALL_melt, TOOL_melt, TREE_melt, UKFS_melt, UNDE_melt, WOOD_melt, WREF_melt, YELL_melt))
```

phylum graph
```{r}
library(ggthemes)
b<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(23)) + 
  ggtitle("Fungi Graphs by Site")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

b
```

```{r}

ggsave(filename = "phylum_bysite_fungi_only.pdf",
      plot = b,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```
# Metazoa by size class graphs

First, need to fix taxonomy to upload to mctoolsr

```{r}
library(janitor)
setwd("/Users/leenavilonen/Documents/NEON")
taxonomy<-read.csv('sizeclass_taxonomy.csv',header=TRUE)
library(dplyr)
library(tidyr)
taxonomy<-unite(taxonomy, col='taxonomy', c('Domain', 'SuperGroup', 'Kingdom', 'Phylum', 'Class', 'Order', 'Genus','Size_Class'), sep=';')
otu_table<-read.csv('input_metazoa_otu.csv',header=TRUE)
otu_table_withtax <- merge(otu_table,taxonomy,by="X.OTU.ID")
write.csv(otu_table_withtax,'otu_table_withtax_metazoa_sizeclass.csv')
```


```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax_metazoa_sizeclass.txt", "input_metazoa_map.txt")
input = filter_data(input, 'nlcdClass', filter_vals = '0')


input_microfauna = filter_taxa_from_input(input, taxa_to_keep = 'Microfauna')
input_mesofauna = filter_taxa_from_input(input, taxa_to_keep = 'Mesofauna')
input_macrofauna = filter_taxa_from_input(input, taxa_to_keep = 'Macrofauna')
```

```{r}
library(labdsv)
otuTAB<-input_microfauna$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
input_microfauna$data_loaded<-t(hell)
input_microfauna$data_loaded<-input_microfauna$data_loaded[ , apply(input_microfauna$data_loaded, 2, function(x) !any(is.na(x)))]

otuTAB<-input_macrofauna$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
input_macrofauna$data_loaded<-t(hell)
input_macrofauna$data_loaded<-input_macrofauna$data_loaded[ , apply(input_macrofauna$data_loaded, 2, function(x) !any(is.na(x)))]

otuTAB<-input_mesofauna$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
input_mesofauna$data_loaded<-t(hell)
input_mesofauna$data_loaded<-input_mesofauna$data_loaded[ , apply(input_mesofauna$data_loaded, 2, function(x) !any(is.na(x)))]
```

```{r}
library(janitor)
#microfauna.merge<-merge(t(input_microfauna$data_loaded),input_microfauna$map_loaded,by = 'row.names')
#write.csv(microfauna.merge,file='microfauna.merge.csv', row.names=FALSE)
microfauna_data<-read.csv(file = 'microfauna.merge.csv',header = TRUE)
microfauna_map<-read.csv(file = 'microfauna_map.csv',header = TRUE)
model1<-adonis2(microfauna_data~nlcdClass,data=microfauna_map,permutations=999,method="jaccard")
model1
```

```{r}
library(ecole)
a<-permanova_pairwise(
  microfauna_data,
  microfauna_map$nlcdClass,
  permutations = 999,
  method = "jaccard",
  padj = "bonferroni"
)
write.csv(a,file='perm_metazoa_microfauna.csv', row.names=FALSE)
```

```{r}
#mesofauna.merge<-merge(t(input_mesofauna$data_loaded),input_mesofauna$map_loaded,by = 'row.names')
#write.csv(mesofauna.merge,file='mesofauna.merge.csv', row.names=FALSE)
mesofauna_data<-read.csv(file = 'mesofauna.merge.csv',header = TRUE)
mesofauna_map<-read.csv(file = 'mesofauna_map.csv',header = TRUE)
model1<-adonis2(mesofauna_data~nlcdClass,data=mesofauna_map,permutations=999,method="jaccard")
model1
```

```{r}
library(ecole)
a<-permanova_pairwise(
  mesofauna_data,
  mesofauna_map$nlcdClass,
  permutations = 999,
  method = "jaccard",
  padj = "bonferroni"
)
write.csv(a,file='perm_metazoa_mesofauna.csv', row.names=FALSE)
```

```{r}
#macrofauna.merge<-merge(t(input_macrofauna$data_loaded),input_macrofauna$map_loaded,by = 'row.names')
#write.csv(macrofauna.merge,file='macrofauna.merge.csv', row.names=FALSE)
macrofauna_data<-read.csv(file = 'macrofauna.merge.csv',header = TRUE)
macrofauna_map<-read.csv(file = 'macrofauna_map.csv',header = TRUE)
model1<-adonis2(macrofauna_data~nlcdClass,data=macrofauna_map,permutations=999,method="jaccard")
model1
```

```{r}
library(ecole)
a<-permanova_pairwise(
  macrofauna_data,
  macrofauna_map$nlcdClass,
  permutations = 999,
  method = "jaccard",
  padj = "bonferroni"
)
write.csv(a,file='perm_metazoa_macrofauna.csv', row.names=FALSE)
```

```{r}
microfauna.mds<-metaMDS(t(input_microfauna$data_loaded))
microfauna.scores<-vegan::scores(microfauna.mds)
data.scoresmicrofauna <- as.data.frame(microfauna.scores$sites) 
library(janitor)
data.scoresmicrofauna.merge<-merge(data.scoresmicrofauna,input_microfauna$map_loaded,by = 'row.names')
```

```{r}
mesofauna.mds<-metaMDS(t(input_mesofauna$data_loaded))
mesofauna.scores<-vegan::scores(mesofauna.mds)
data.scoresmesofauna <- as.data.frame(mesofauna.scores$sites) 
library(janitor)
data.scoresmesofauna.merge<-merge(data.scoresmesofauna,input_mesofauna$map_loaded,by = 'row.names')
```

```{r}
macrofauna.mds<-metaMDS(t(input_macrofauna$data_loaded))
macrofauna.scores<-vegan::scores(macrofauna.mds)
data.scoresmacrofauna <- as.data.frame(macrofauna.scores$sites) 
library(janitor)
data.scoresmacrofauna.merge<-merge(data.scoresmacrofauna,input_macrofauna$map_loaded,by = 'row.names')
```

Get rid of outliers - micro
```{r}
data.scoresmicrofauna.merge<-subset(data.scoresmicrofauna.merge,NMDS1<0)
data.scoresmicrofauna.merge<-subset(data.scoresmicrofauna.merge,NMDS1>(-0.1))
data.scoresmicrofauna.merge<-subset(data.scoresmicrofauna.merge,NMDS2<0.04)
```

```{r}
library(ggplot2)
micro<-ggplot(data=data.scoresmicrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmicrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  ggtitle("Microfauna")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
micro

```



Get rid of outliers - meso
```{r}
data.scoresmesofauna.merge<-subset(data.scoresmesofauna.merge,NMDS1<0)
data.scoresmesofauna.merge<-subset(data.scoresmesofauna.merge,NMDS1>(-0.075))
data.scoresmesofauna.merge<-subset(data.scoresmesofauna.merge,NMDS2<(-0.35))
data.scoresmesofauna.merge<-subset(data.scoresmesofauna.merge,NMDS2>(-0.45))
```

```{r}
library(ggplot2)
meso<-ggplot(data=data.scoresmesofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmesofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  ggtitle("Mesofauna")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
meso

```

Get rid of outliers - macro
```{r}
data.scoresmacrofauna.merge<-subset(data.scoresmacrofauna.merge,NMDS1<0.001)
data.scoresmacrofauna.merge<-subset(data.scoresmacrofauna.merge,NMDS1>(-0.001))
data.scoresmacrofauna.merge<-subset(data.scoresmacrofauna.merge,NMDS2<0)
data.scoresmacrofauna.merge<-subset(data.scoresmacrofauna.merge,NMDS2>(-0.002))
```

```{r}
library(ggplot2)
macro<-ggplot(data=data.scoresmacrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmacrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  ggtitle("Macrofauna")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
macro

```

```{r}
library(patchwork)
jointgraph<-micro+meso+macro
jointgraph
```

```{r}
ggsave(filename = "Biome_nmds_metazoa_bysize.pdf",
      plot = jointgraph,
      bg = "transparent",
      width = 18, height = 6, units = "in",
      dpi = 600)
```

```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary_macro <- summarySE(data.scoresmacrofauna.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds2_summary_macro <- summarySE(data.scoresmacrofauna.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
library(dplyr)
nmds1_summary_macro<-select(nmds1_summary_macro,-c(N,sd,ci))
nmds1_summary_macro<-nmds1_summary_macro %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary_macro<-select(nmds2_summary_macro,-c(N,sd,ci))
nmds2_summary_macro<-nmds2_summary_macro %>% dplyr::rename("se_NMDS2"= se)
nmds_summary_macro<-merge(nmds1_summary_macro,nmds2_summary_macro,by="nlcdClass")

nmds1_summary_meso <- summarySE(data.scoresmesofauna.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds2_summary_meso <- summarySE(data.scoresmesofauna.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
library(dplyr)
nmds1_summary_meso<-select(nmds1_summary_meso,-c(N,sd,ci))
nmds1_summary_meso<-nmds1_summary_meso %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary_meso<-select(nmds2_summary_meso,-c(N,sd,ci))
nmds2_summary_meso<-nmds2_summary_meso %>% dplyr::rename("se_NMDS2"= se)
nmds_summary_meso<-merge(nmds1_summary_meso,nmds2_summary_meso,by="nlcdClass")

nmds1_summary_micro <- summarySE(data.scoresmicrofauna.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds2_summary_micro <- summarySE(data.scoresmicrofauna.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
library(dplyr)
nmds1_summary_micro<-select(nmds1_summary_micro,-c(N,sd,ci))
nmds1_summary_micro<-nmds1_summary_micro %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary_micro<-select(nmds2_summary_micro,-c(N,sd,ci))
nmds2_summary_micro<-nmds2_summary_micro %>% dplyr::rename("se_NMDS2"= se)
nmds_summary_micro<-merge(nmds1_summary_micro,nmds2_summary_micro,by="nlcdClass")
```

micro
```{r}
library(ggplot2)
microsum<-ggplot(data=nmds_summary_micro,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmicrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary_micro,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary_micro,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.0005) +
  geom_errorbar(data=nmds_summary_micro,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.0005) +
  ggtitle("Microfauna")+
  xlim(-0.065,-0.045)+
  ylim(-0.02,0)+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
microsum

```

meso
```{r}
library(ggplot2)
mesosum<-ggplot(data=nmds_summary_meso,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmesofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary_meso,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary_meso,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.001) +
  geom_errorbar(data=nmds_summary_meso,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.001) +
  ggtitle("Mesofauna")+
  xlim(-0.05,-0.025)+
  ylim(-0.39,-0.36)+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
mesosum

```

macro
```{r}
library(ggplot2)
macrosum<-ggplot(data=nmds_summary_macro,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmacrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary_macro,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary_macro,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.00005) +
  geom_errorbar(data=nmds_summary_macro,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.00005) +
  ggtitle("macrofauna")+
  #xlim(-0.05,-0.025)+
  ylim(-0.0014,-0.0003)+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
macrosum

```

```{r}
library(patchwork)
joinedgraphs_sum<-microsum+mesosum+macrosum
joinedgraphs_sum
```


```{r}
ggsave(filename = "Biome_nmds_metazoa_bysize_average_se.pdf",
      plot = joinedgraphs_sum,
      bg = "transparent",
      width = 18, height = 6, units = "in",
      dpi = 600)
```

comm = t(biome$data_loaded),distance = "jaccard", k=2

# Do the same but with jaccard
```{r}
microfauna.mds<-metaMDS(comm = t(input_microfauna$data_loaded),distance = "jaccard", k=2)
microfauna.scores<-vegan::scores(microfauna.mds)
data.scoresmicrofauna <- as.data.frame(microfauna.scores$sites) 
library(janitor)
data.scoresmicrofauna.merge<-merge(data.scoresmicrofauna,input_microfauna$map_loaded,by = 'row.names')
```

```{r}
mesofauna.mds<-metaMDS(comm = t(input_mesofauna$data_loaded),distance = "jaccard", k=2)
mesofauna.scores<-vegan::scores(mesofauna.mds)
data.scoresmesofauna <- as.data.frame(mesofauna.scores$sites) 
library(janitor)
data.scoresmesofauna.merge<-merge(data.scoresmesofauna,input_mesofauna$map_loaded,by = 'row.names')
```

```{r}
macrofauna.mds<-metaMDS(comm = t(input_macrofauna$data_loaded),distance = "jaccard", k=2)
macrofauna.scores<-vegan::scores(macrofauna.mds)
data.scoresmacrofauna <- as.data.frame(macrofauna.scores$sites) 
library(janitor)
data.scoresmacrofauna.merge<-merge(data.scoresmacrofauna,input_macrofauna$map_loaded,by = 'row.names')
```

Get rid of outliers - micro
```{r}
data.scoresmicrofauna.merge<-subset(data.scoresmicrofauna.merge,NMDS1<0.5)
data.scoresmicrofauna.merge<-subset(data.scoresmicrofauna.merge,NMDS1>(-0.8))
data.scoresmicrofauna.merge<-subset(data.scoresmicrofauna.merge,NMDS2<0.4)
data.scoresmicrofauna.merge<-subset(data.scoresmicrofauna.merge,NMDS2>(-0.5))
```

```{r}
library(ggplot2)
micro<-ggplot(data=data.scoresmicrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmicrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  ggtitle("Microfauna")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
micro

```



Get rid of outliers - meso
```{r}
data.scoresmesofauna.merge<-subset(data.scoresmesofauna.merge,NMDS1<(-0.2))
data.scoresmesofauna.merge<-subset(data.scoresmesofauna.merge,NMDS1>(-0.4))
data.scoresmesofauna.merge<-subset(data.scoresmesofauna.merge,NMDS2<0.25)
data.scoresmesofauna.merge<-subset(data.scoresmesofauna.merge,NMDS2>(-0.25))
```

```{r}
library(ggplot2)
meso<-ggplot(data=data.scoresmesofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmesofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  ggtitle("Mesofauna")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
meso

```

Get rid of outliers - macro
```{r}
data.scoresmacrofauna.merge<-subset(data.scoresmacrofauna.merge,NMDS1<(-0.0001))
data.scoresmacrofauna.merge<-subset(data.scoresmacrofauna.merge,NMDS1>(-0.001))
data.scoresmacrofauna.merge<-subset(data.scoresmacrofauna.merge,NMDS2<0.0001)
data.scoresmacrofauna.merge<-subset(data.scoresmacrofauna.merge,NMDS2>(-0.0001))
```

```{r}
library(ggplot2)
macro<-ggplot(data=data.scoresmacrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmacrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + # add the point markers
  ggtitle("Macrofauna")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
macro

```

```{r}
library(patchwork)
jointgraph<-micro+meso+macro
jointgraph
```

```{r}
ggsave(filename = "Biome_nmds_metazoa_bysize_presenceabsence.pdf",
      plot = jointgraph,
      bg = "transparent",
      width = 18, height = 6, units = "in",
      dpi = 600)
```

```{r}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    library(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This does the summary. For each group's data frame, return a vector with
    # N, mean, and sd
    datac <- ddply(data, groupvars, .drop=.drop,
      .fun = function(xx, col) {
        c(N    = length2(xx[[col]], na.rm=na.rm),
          mean = mean   (xx[[col]], na.rm=na.rm),
          sd   = sd     (xx[[col]], na.rm=na.rm)
        )
      },
      measurevar
    )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean" = measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}
```

```{r}
nmds1_summary_macro <- summarySE(data.scoresmacrofauna.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds2_summary_macro <- summarySE(data.scoresmacrofauna.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
library(dplyr)
nmds1_summary_macro<-select(nmds1_summary_macro,-c(N,sd,ci))
nmds1_summary_macro<-nmds1_summary_macro %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary_macro<-select(nmds2_summary_macro,-c(N,sd,ci))
nmds2_summary_macro<-nmds2_summary_macro %>% dplyr::rename("se_NMDS2"= se)
nmds_summary_macro<-merge(nmds1_summary_macro,nmds2_summary_macro,by="nlcdClass")

nmds1_summary_meso <- summarySE(data.scoresmesofauna.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds2_summary_meso <- summarySE(data.scoresmesofauna.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
library(dplyr)
nmds1_summary_meso<-select(nmds1_summary_meso,-c(N,sd,ci))
nmds1_summary_meso<-nmds1_summary_meso %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary_meso<-select(nmds2_summary_meso,-c(N,sd,ci))
nmds2_summary_meso<-nmds2_summary_meso %>% dplyr::rename("se_NMDS2"= se)
nmds_summary_meso<-merge(nmds1_summary_meso,nmds2_summary_meso,by="nlcdClass")

nmds1_summary_micro <- summarySE(data.scoresmicrofauna.merge, measurevar=c("NMDS1"), groupvars=("nlcdClass"))
nmds2_summary_micro <- summarySE(data.scoresmicrofauna.merge, measurevar=c("NMDS2"), groupvars=("nlcdClass"))
library(dplyr)
nmds1_summary_micro<-select(nmds1_summary_micro,-c(N,sd,ci))
nmds1_summary_micro<-nmds1_summary_micro %>% dplyr::rename("se_NMDS1"= se)
nmds2_summary_micro<-select(nmds2_summary_micro,-c(N,sd,ci))
nmds2_summary_micro<-nmds2_summary_micro %>% dplyr::rename("se_NMDS2"= se)
nmds_summary_micro<-merge(nmds1_summary_micro,nmds2_summary_micro,by="nlcdClass")
```

micro
```{r}
library(ggplot2)
microsum<-ggplot(data=nmds_summary_micro,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmicrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary_micro,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary_micro,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.0005) +
  geom_errorbar(data=nmds_summary_micro,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.0005) +
  ggtitle("Microfauna")+
  xlim(-0.6,-0.3)+
  ylim(-0.15,0.15)+
 
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
microsum

```

meso
```{r}
library(ggplot2)
mesosum<-ggplot(data=nmds_summary_meso,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmesofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary_meso,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary_meso,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.001) +
  geom_errorbar(data=nmds_summary_meso,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.001) +
  ggtitle("Mesofauna")+
  #xlim(-0.05,-0.025)+
  #ylim(-0.39,-0.36)+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
mesosum

```

macro
```{r}
library(ggplot2)
macrosum<-ggplot(data=nmds_summary_macro,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass))) + 
  geom_point(data=data.scoresmacrofauna.merge,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),alpha = 0.1) +
  
  geom_point(data=nmds_summary_macro,aes(x=NMDS1,y=NMDS2,color=factor(nlcdClass)),size=3,alpha = 1) + # add the point markers
  geom_errorbar(data=nmds_summary_macro,aes(ymin=NMDS2-se_NMDS2, ymax=NMDS2+se_NMDS2), width=.000005) +
  geom_errorbar(data=nmds_summary_macro,aes(xmin=NMDS1-se_NMDS1, xmax=NMDS1+se_NMDS1), width=.000005) +
  ggtitle("macrofauna")+
  xlim(-0.0003,-0.0002)+
  #ylim(-0.0014,-0.0003)+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
macrosum

```

```{r}
library(patchwork)
joinedgraphs_sum<-microsum+mesosum+macrosum
joinedgraphs_sum
```


```{r}
ggsave(filename = "Biome_nmds_metazoa_bysize_average_se_presenceabsence.pdf",
      plot = joinedgraphs_sum,
      bg = "transparent",
      width = 18, height = 6, units = "in",
      dpi = 600)
```

```{r}
library(janitor)
setwd("/Users/leenavilonen/Documents/NEON")
taxonomy<-read.csv('eukDiv_tax_functionalgroup.csv',header=TRUE)
library(dplyr)
library(tidyr)
taxonomy<-unite(taxonomy, col='taxonomy', c('Domain', 'SuperGroup', 'Kingdom', 'Phylum', 'Class', 'Order', 'Genus','Functional_Group'), sep=';')
otu_table<-read.csv('otu_table.csv',header=TRUE)
otu_table_withtax <- merge(otu_table,taxonomy,by="X.OTU.ID")
write.csv(otu_table_withtax,'otu_table_withtax_functionalgroup.csv')
```



```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax_functionalgroup.txt", "eukDiv_mta.txt")
input = filter_data(input, 'nlcdClass', filter_vals = '0')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
```


Filter by site
```{r}
MLBS = filter_data(input, 'Site', keep_vals = 'MLBS')
HARV = filter_data(input, 'Site', keep_vals = 'HARV')
WOOD = filter_data(input, 'Site', keep_vals = 'WOOD')
LAJA = filter_data(input, 'Site', keep_vals = 'LAJA')
UNDE = filter_data(input, 'Site', keep_vals = 'UNDE')
SRER = filter_data(input, 'Site', keep_vals = 'SRER')
DSNY = filter_data(input, 'Site', keep_vals = 'DSNY')
BONA = filter_data(input, 'Site', keep_vals = 'BONA')
GUAN = filter_data(input, 'Site', keep_vals = 'GUAN')
ABBY = filter_data(input, 'Site', keep_vals = 'ABBY')
BARR = filter_data(input, 'Site', keep_vals = 'BARR')
BART = filter_data(input, 'Site', keep_vals = 'BART')
BLAN = filter_data(input, 'Site', keep_vals = 'BLAN')
CLBJ = filter_data(input, 'Site', keep_vals = 'CLBJ')
CPER = filter_data(input, 'Site', keep_vals = 'CPER')
DCFS = filter_data(input, 'Site', keep_vals = 'DCFS')
DEJU = filter_data(input, 'Site', keep_vals = 'DEJU')
DELA = filter_data(input, 'Site', keep_vals = 'DELA')
GRSM = filter_data(input, 'Site', keep_vals = 'GRSM')
HEAL = filter_data(input, 'Site', keep_vals = 'HEAL')
JERC = filter_data(input, 'Site', keep_vals = 'JERC')
JORN = filter_data(input, 'Site', keep_vals = 'JORN')
KONA = filter_data(input, 'Site', keep_vals = 'KONA')
KONZ = filter_data(input, 'Site', keep_vals = 'KONZ')
LENO = filter_data(input, 'Site', keep_vals = 'LENO')
MOAB = filter_data(input, 'Site', keep_vals = 'MOAB')
NIWO = filter_data(input, 'Site', keep_vals = 'NIWO')
OAES = filter_data(input, 'Site', keep_vals = 'OAES')
ONAQ = filter_data(input, 'Site', keep_vals = 'ONAQ')
ORNL = filter_data(input, 'Site', keep_vals = 'ORNL')
OSBS = filter_data(input, 'Site', keep_vals = 'OSBS')
PUUM = filter_data(input, 'Site', keep_vals = 'PUUM')
RMNP = filter_data(input, 'Site', keep_vals = 'RMNP')
SCBI = filter_data(input, 'Site', keep_vals = 'SCBI')
SERC = filter_data(input, 'Site', keep_vals = 'SERC')
SJER = filter_data(input, 'Site', keep_vals = 'SJER')
SOAP = filter_data(input, 'Site', keep_vals = 'SOAP')
STEI = filter_data(input, 'Site', keep_vals = 'STEI')
STER = filter_data(input, 'Site', keep_vals = 'STER')
TALL = filter_data(input, 'Site', keep_vals = 'TALL')
TOOL = filter_data(input, 'Site', keep_vals = 'TOOL')
TREE = filter_data(input, 'Site', keep_vals = 'TREE')
UKFS = filter_data(input, 'Site', keep_vals = 'UKFS')
WREF = filter_data(input, 'Site', keep_vals = 'WREF')
YELL = filter_data(input, 'Site', keep_vals = 'YELL')
```


```{r}
otuTAB<-ABBY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ABBY$data_loaded<-t(hell)

otuTAB<-BARR$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BARR$data_loaded<-t(hell)

otuTAB<-BART$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BART$data_loaded<-t(hell)

otuTAB<-BLAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BLAN$data_loaded<-t(hell)

otuTAB<-BONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BONA$data_loaded<-t(hell)

otuTAB<-CLBJ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CLBJ$data_loaded<-t(hell)

otuTAB<-CPER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CPER$data_loaded<-t(hell)

otuTAB<-DCFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DCFS$data_loaded<-t(hell)

otuTAB<-DELA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DELA$data_loaded<-t(hell)

otuTAB<-DSNY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DSNY$data_loaded<-t(hell)

otuTAB<-GRSM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GRSM$data_loaded<-t(hell)

otuTAB<-GUAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GUAN$data_loaded<-t(hell)

otuTAB<-HARV$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HARV$data_loaded<-t(hell)

otuTAB<-HEAL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HEAL$data_loaded<-t(hell)

otuTAB<-JERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JERC$data_loaded<-t(hell)

otuTAB<-JORN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JORN$data_loaded<-t(hell)

otuTAB<-KONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONA$data_loaded<-t(hell)

otuTAB<-KONZ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONZ$data_loaded<-t(hell)

otuTAB<-LAJA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LAJA$data_loaded<-t(hell)

otuTAB<-LENO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LENO$data_loaded<-t(hell)

otuTAB<-MLBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MLBS$data_loaded<-t(hell)

otuTAB<-MOAB$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MOAB$data_loaded<-t(hell)

otuTAB<-NIWO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
NIWO$data_loaded<-t(hell)

otuTAB<-OAES$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OAES$data_loaded<-t(hell)

otuTAB<-ONAQ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ONAQ$data_loaded<-t(hell)

otuTAB<-ORNL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ORNL$data_loaded<-t(hell)

otuTAB<-OSBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OSBS$data_loaded<-t(hell)

otuTAB<-PUUM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
PUUM$data_loaded<-t(hell)

otuTAB<-RMNP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
RMNP$data_loaded<-t(hell)

otuTAB<-SCBI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SCBI$data_loaded<-t(hell)

otuTAB<-SERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SERC$data_loaded<-t(hell)

otuTAB<-SJER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SJER$data_loaded<-t(hell)

otuTAB<-SOAP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SOAP$data_loaded<-t(hell)

otuTAB<-SRER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SRER$data_loaded<-t(hell)

otuTAB<-STEI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STEI$data_loaded<-t(hell)

otuTAB<-STER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STER$data_loaded<-t(hell)

otuTAB<-TALL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TALL$data_loaded<-t(hell)

otuTAB<-TOOL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TOOL$data_loaded<-t(hell)

otuTAB<-TREE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TREE$data_loaded<-t(hell)

otuTAB<-UKFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UKFS$data_loaded<-t(hell)

otuTAB<-UNDE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UNDE$data_loaded<-t(hell)

otuTAB<-WOOD$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WOOD$data_loaded<-t(hell)

otuTAB<-WREF$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WREF$data_loaded<-t(hell)

otuTAB<-YELL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
YELL$data_loaded<-t(hell)
```

```{r}
library(reshape2)
ABBY_tax_sum_phylum = summarize_taxonomy(ABBY, level = 8, report_higher_tax = FALSE)
ABBY_tax_sum_phylum$Phylum <-row.names(ABBY_tax_sum_phylum)
ABBY_melt<- melt(ABBY_tax_sum_phylum, id.var="Phylum")
site<-rep("ABBY",times=nrow(ABBY_melt))
ABBY_melt<-cbind(ABBY_melt,site)

BARR_tax_sum_phylum = summarize_taxonomy(BARR, level = 8, report_higher_tax = FALSE)
BARR_tax_sum_phylum$Phylum <-row.names(BARR_tax_sum_phylum)
BARR_melt<- melt(BARR_tax_sum_phylum, id.var="Phylum")
nrow(BARR_melt)
site<-rep("BARR",times=nrow(BARR_melt))
BARR_melt<-cbind(BARR_melt,site)

BART_tax_sum_phylum = summarize_taxonomy(BART, level = 8, report_higher_tax = FALSE)
BART_tax_sum_phylum$Phylum <-row.names(BART_tax_sum_phylum)
BART_melt<- melt(BART_tax_sum_phylum, id.var="Phylum")
nrow(BART_melt)
site<-rep("BART",times=nrow(BART_melt))
BART_melt<-cbind(BART_melt,site)

BLAN_tax_sum_phylum = summarize_taxonomy(BLAN, level = 8, report_higher_tax = FALSE)
BLAN_tax_sum_phylum$Phylum <-row.names(BLAN_tax_sum_phylum)
BLAN_melt<- melt(BLAN_tax_sum_phylum, id.var="Phylum")
nrow(BLAN_melt)
site<-rep("BLAN",times=nrow(BLAN_melt))
BLAN_melt<-cbind(BLAN_melt,site)

BONA_tax_sum_phylum = summarize_taxonomy(BONA, level = 8, report_higher_tax = FALSE)
BONA_tax_sum_phylum$Phylum <-row.names(BONA_tax_sum_phylum)
BONA_melt<- melt(BONA_tax_sum_phylum, id.var="Phylum")
nrow(BONA_melt)
site<-rep("BONA",times=nrow(BONA_melt))
BONA_melt<-cbind(BONA_melt,site)

CLBJ_tax_sum_phylum = summarize_taxonomy(CLBJ, level = 8, report_higher_tax = FALSE)
CLBJ_tax_sum_phylum$Phylum <-row.names(CLBJ_tax_sum_phylum)
CLBJ_melt<- melt(CLBJ_tax_sum_phylum, id.var="Phylum")
nrow(CLBJ_melt)
site<-rep("CLBJ",times=nrow(CLBJ_melt))
CLBJ_melt<-cbind(CLBJ_melt,site)

CPER_tax_sum_phylum = summarize_taxonomy(CPER, level = 8, report_higher_tax = FALSE)
CPER_tax_sum_phylum$Phylum <-row.names(CPER_tax_sum_phylum)
CPER_melt<- melt(CPER_tax_sum_phylum, id.var="Phylum")
nrow(CPER_melt)
site<-rep("CPER",times=nrow(CPER_melt))
CPER_melt<-cbind(CPER_melt,site)

DCFS_tax_sum_phylum = summarize_taxonomy(DCFS, level = 8, report_higher_tax = FALSE)
DCFS_tax_sum_phylum$Phylum <-row.names(DCFS_tax_sum_phylum)
DCFS_melt<- melt(DCFS_tax_sum_phylum, id.var="Phylum")
nrow(DCFS_melt)
site<-rep("DCFS",times=nrow(DCFS_melt))
DCFS_melt<-cbind(DCFS_melt,site)

DEJU_tax_sum_phylum = summarize_taxonomy(DEJU, level = 8, report_higher_tax = FALSE)
DEJU_tax_sum_phylum$Phylum <-row.names(DEJU_tax_sum_phylum)
DEJU_melt<- melt(DEJU_tax_sum_phylum, id.var="Phylum")
nrow(DEJU_melt)
site<-rep("DEJU",times=nrow(DEJU_melt))
DEJU_melt<-cbind(DEJU_melt,site)

DELA_tax_sum_phylum = summarize_taxonomy(DELA, level = 8, report_higher_tax = FALSE)
DELA_tax_sum_phylum$Phylum <-row.names(DELA_tax_sum_phylum)
DELA_melt<- melt(DELA_tax_sum_phylum, id.var="Phylum")
nrow(DELA_melt)
site<-rep("DELA",times=nrow(DELA_melt))
DELA_melt<-cbind(DELA_melt,site)

DSNY_tax_sum_phylum = summarize_taxonomy(DSNY, level = 8, report_higher_tax = FALSE)
DSNY_tax_sum_phylum$Phylum <-row.names(DSNY_tax_sum_phylum)
DSNY_melt<- melt(DSNY_tax_sum_phylum, id.var="Phylum")
nrow(DSNY_melt)
site<-rep("DSNY",times=nrow(DSNY_melt))
DSNY_melt<-cbind(DSNY_melt,site)

GRSM_tax_sum_phylum = summarize_taxonomy(GRSM, level = 8, report_higher_tax = FALSE)
GRSM_tax_sum_phylum$Phylum <-row.names(GRSM_tax_sum_phylum)
GRSM_melt<- melt(GRSM_tax_sum_phylum, id.var="Phylum")
nrow(GRSM_melt)
site<-rep("GRSM",times=nrow(GRSM_melt))
GRSM_melt<-cbind(GRSM_melt,site)

GUAN_tax_sum_phylum = summarize_taxonomy(GUAN, level = 8, report_higher_tax = FALSE)
GUAN_tax_sum_phylum$Phylum <-row.names(GUAN_tax_sum_phylum)
GUAN_melt<- melt(GUAN_tax_sum_phylum, id.var="Phylum")
nrow(GUAN_melt)
site<-rep("GUAN",times=nrow(GUAN_melt))
GUAN_melt<-cbind(GUAN_melt,site)

HARV_tax_sum_phylum = summarize_taxonomy(HARV, level = 8, report_higher_tax = FALSE)
HARV_tax_sum_phylum$Phylum <-row.names(HARV_tax_sum_phylum)
HARV_melt<- melt(HARV_tax_sum_phylum, id.var="Phylum")
nrow(HARV_melt)
site<-rep("HARV",times=nrow(HARV_melt))
HARV_melt<-cbind(HARV_melt,site)

HEAL_tax_sum_phylum = summarize_taxonomy(HEAL, level = 8, report_higher_tax = FALSE)
HEAL_tax_sum_phylum$Phylum <-row.names(HEAL_tax_sum_phylum)
HEAL_melt<- melt(HEAL_tax_sum_phylum, id.var="Phylum")
nrow(HEAL_melt)
site<-rep("HEAL",times=nrow(HEAL_melt))
HEAL_melt<-cbind(HEAL_melt,site)

JERC_tax_sum_phylum = summarize_taxonomy(JERC, level = 8, report_higher_tax = FALSE)
JERC_tax_sum_phylum$Phylum <-row.names(JERC_tax_sum_phylum)
JERC_melt<- melt(JERC_tax_sum_phylum, id.var="Phylum")
nrow(JERC_melt)
site<-rep("JERC",times=nrow(JERC_melt))
JERC_melt<-cbind(JERC_melt,site)

JORN_tax_sum_phylum = summarize_taxonomy(JORN, level = 8, report_higher_tax = FALSE)
JORN_tax_sum_phylum$Phylum <-row.names(JORN_tax_sum_phylum)
JORN_melt<- melt(JORN_tax_sum_phylum, id.var="Phylum")
nrow(JORN_melt)
site<-rep("JORN",times=nrow(JORN_melt))
JORN_melt<-cbind(JORN_melt,site)

KONA_tax_sum_phylum = summarize_taxonomy(KONA, level = 8, report_higher_tax = FALSE)
KONA_tax_sum_phylum$Phylum <-row.names(KONA_tax_sum_phylum)
KONA_melt<- melt(KONA_tax_sum_phylum, id.var="Phylum")
nrow(KONA_melt)
site<-rep("KONA",times=nrow(KONA_melt))
KONA_melt<-cbind(KONA_melt,site)

KONZ_tax_sum_phylum = summarize_taxonomy(KONZ, level = 8, report_higher_tax = FALSE)
KONZ_tax_sum_phylum$Phylum <-row.names(KONZ_tax_sum_phylum)
KONZ_melt<- melt(KONZ_tax_sum_phylum, id.var="Phylum")
nrow(KONZ_melt)
site<-rep("KONZ",times=nrow(KONZ_melt))
KONZ_melt<-cbind(KONZ_melt,site)

LAJA_tax_sum_phylum = summarize_taxonomy(LAJA, level = 8, report_higher_tax = FALSE)
LAJA_tax_sum_phylum$Phylum <-row.names(LAJA_tax_sum_phylum)
LAJA_melt<- melt(LAJA_tax_sum_phylum, id.var="Phylum")
nrow(LAJA_melt)
site<-rep("LAJA",times=nrow(LAJA_melt))
LAJA_melt<-cbind(LAJA_melt,site)

LENO_tax_sum_phylum = summarize_taxonomy(LENO, level = 8, report_higher_tax = FALSE)
LENO_tax_sum_phylum$Phylum <-row.names(LENO_tax_sum_phylum)
LENO_melt<- melt(LENO_tax_sum_phylum, id.var="Phylum")
nrow(LENO_melt)
site<-rep("LENO",times=nrow(LENO_melt))
LENO_melt<-cbind(LENO_melt,site)

MLBS_tax_sum_phylum = summarize_taxonomy(MLBS, level = 8, report_higher_tax = FALSE)
MLBS_tax_sum_phylum$Phylum <-row.names(MLBS_tax_sum_phylum)
MLBS_melt<- melt(MLBS_tax_sum_phylum, id.var="Phylum")
nrow(MLBS_melt)
site<-rep("MLBS",times=nrow(MLBS_melt))
MLBS_melt<-cbind(MLBS_melt,site)

MOAB_tax_sum_phylum = summarize_taxonomy(MOAB, level = 8, report_higher_tax = FALSE)
MOAB_tax_sum_phylum$Phylum <-row.names(MOAB_tax_sum_phylum)
MOAB_melt<- melt(MOAB_tax_sum_phylum, id.var="Phylum")
nrow(MOAB_melt)
site<-rep("MOAB",times=nrow(MOAB_melt))
MOAB_melt<-cbind(MOAB_melt,site)

NIWO_tax_sum_phylum = summarize_taxonomy(NIWO, level = 8, report_higher_tax = FALSE)
NIWO_tax_sum_phylum$Phylum <-row.names(NIWO_tax_sum_phylum)
NIWO_melt<- melt(NIWO_tax_sum_phylum, id.var="Phylum")
nrow(NIWO_melt)
site<-rep("NIWO",times=nrow(NIWO_melt))
NIWO_melt<-cbind(NIWO_melt,site)

OAES_tax_sum_phylum = summarize_taxonomy(OAES, level = 8, report_higher_tax = FALSE)
OAES_tax_sum_phylum$Phylum <-row.names(OAES_tax_sum_phylum)
OAES_melt<- melt(OAES_tax_sum_phylum, id.var="Phylum")
nrow(OAES_melt)
site<-rep("OAES",times=nrow(OAES_melt))
OAES_melt<-cbind(OAES_melt,site)

ONAQ_tax_sum_phylum = summarize_taxonomy(ONAQ, level = 8, report_higher_tax = FALSE)
ONAQ_tax_sum_phylum$Phylum <-row.names(ONAQ_tax_sum_phylum)
ONAQ_melt<- melt(ONAQ_tax_sum_phylum, id.var="Phylum")
nrow(ONAQ_melt)
site<-rep("ONAQ",times=nrow(ONAQ_melt))
ONAQ_melt<-cbind(ONAQ_melt,site)

ORNL_tax_sum_phylum = summarize_taxonomy(ORNL, level = 8, report_higher_tax = FALSE)
ORNL_tax_sum_phylum$Phylum <-row.names(ORNL_tax_sum_phylum)
ORNL_melt<- melt(ORNL_tax_sum_phylum, id.var="Phylum")
nrow(ORNL_melt)
site<-rep("ORNL",times=nrow(ORNL_melt))
ORNL_melt<-cbind(ORNL_melt,site)

OSBS_tax_sum_phylum = summarize_taxonomy(OSBS, level = 8, report_higher_tax = FALSE)
OSBS_tax_sum_phylum$Phylum <-row.names(OSBS_tax_sum_phylum)
OSBS_melt<- melt(OSBS_tax_sum_phylum, id.var="Phylum")
nrow(OSBS_melt)
site<-rep("OSBS",times=nrow(OSBS_melt))
OSBS_melt<-cbind(OSBS_melt,site)

PUUM_tax_sum_phylum = summarize_taxonomy(PUUM, level = 8, report_higher_tax = FALSE)
PUUM_tax_sum_phylum$Phylum <-row.names(PUUM_tax_sum_phylum)
PUUM_melt<- melt(PUUM_tax_sum_phylum, id.var="Phylum")
nrow(PUUM_melt)
site<-rep("PUUM",times=nrow(PUUM_melt))
PUUM_melt<-cbind(PUUM_melt,site)

RMNP_tax_sum_phylum = summarize_taxonomy(RMNP, level = 8, report_higher_tax = FALSE)
RMNP_tax_sum_phylum$Phylum <-row.names(RMNP_tax_sum_phylum)
RMNP_melt<- melt(RMNP_tax_sum_phylum, id.var="Phylum")
nrow(RMNP_melt)
site<-rep("RMNP",times=nrow(RMNP_melt))
RMNP_melt<-cbind(RMNP_melt,site)

SCBI_tax_sum_phylum = summarize_taxonomy(SCBI, level = 8, report_higher_tax = FALSE)
SCBI_tax_sum_phylum$Phylum <-row.names(SCBI_tax_sum_phylum)
SCBI_melt<- melt(SCBI_tax_sum_phylum, id.var="Phylum")
nrow(SCBI_melt)
site<-rep("SCBI",times=nrow(SCBI_melt))
SCBI_melt<-cbind(SCBI_melt,site)

SERC_tax_sum_phylum = summarize_taxonomy(SERC, level = 8, report_higher_tax = FALSE)
SERC_tax_sum_phylum$Phylum <-row.names(SERC_tax_sum_phylum)
SERC_melt<- melt(SERC_tax_sum_phylum, id.var="Phylum")
nrow(SERC_melt)
site<-rep("SERC",times=nrow(SERC_melt))
SERC_melt<-cbind(SERC_melt,site)

SJER_tax_sum_phylum = summarize_taxonomy(SJER, level = 8, report_higher_tax = FALSE)
SJER_tax_sum_phylum$Phylum <-row.names(SJER_tax_sum_phylum)
SJER_melt<- melt(SJER_tax_sum_phylum, id.var="Phylum")
nrow(SJER_melt)
site<-rep("SJER",times=nrow(SJER_melt))
SJER_melt<-cbind(SJER_melt,site)

SOAP_tax_sum_phylum = summarize_taxonomy(SOAP, level = 8, report_higher_tax = FALSE)
SOAP_tax_sum_phylum$Phylum <-row.names(SOAP_tax_sum_phylum)
SOAP_melt<- melt(SOAP_tax_sum_phylum, id.var="Phylum")
nrow(SOAP_melt)
site<-rep("SOAP",times=nrow(SOAP_melt))
SOAP_melt<-cbind(SOAP_melt,site)

SRER_tax_sum_phylum = summarize_taxonomy(SRER, level = 8, report_higher_tax = FALSE)
SRER_tax_sum_phylum$Phylum <-row.names(SRER_tax_sum_phylum)
SRER_melt<- melt(SRER_tax_sum_phylum, id.var="Phylum")
nrow(SRER_melt)
site<-rep("SRER",times=nrow(SRER_melt))
SRER_melt<-cbind(SRER_melt,site)

STEI_tax_sum_phylum = summarize_taxonomy(STEI, level = 8, report_higher_tax = FALSE)
STEI_tax_sum_phylum$Phylum <-row.names(STEI_tax_sum_phylum)
STEI_melt<- melt(STEI_tax_sum_phylum, id.var="Phylum")
nrow(STEI_melt)
site<-rep("STEI",times=nrow(STEI_melt))
STEI_melt<-cbind(STEI_melt,site)

STER_tax_sum_phylum = summarize_taxonomy(STER, level = 8, report_higher_tax = FALSE)
STER_tax_sum_phylum$Phylum <-row.names(STER_tax_sum_phylum)
STER_melt<- melt(STER_tax_sum_phylum, id.var="Phylum")
nrow(STER_melt)
site<-rep("STER",times=nrow(STER_melt))
STER_melt<-cbind(STER_melt,site)

TALL_tax_sum_phylum = summarize_taxonomy(TALL, level = 8, report_higher_tax = FALSE)
TALL_tax_sum_phylum$Phylum <-row.names(TALL_tax_sum_phylum)
TALL_melt<- melt(TALL_tax_sum_phylum, id.var="Phylum")
nrow(TALL_melt)
site<-rep("TALL",times=nrow(TALL_melt))
TALL_melt<-cbind(TALL_melt,site)

TOOL_tax_sum_phylum = summarize_taxonomy(TOOL, level = 8, report_higher_tax = FALSE)
TOOL_tax_sum_phylum$Phylum <-row.names(TOOL_tax_sum_phylum)
TOOL_melt<- melt(TOOL_tax_sum_phylum, id.var="Phylum")
nrow(TOOL_melt)
site<-rep("TOOL",times=nrow(TOOL_melt))
TOOL_melt<-cbind(TOOL_melt,site)

TREE_tax_sum_phylum = summarize_taxonomy(TREE, level = 8, report_higher_tax = FALSE)
TREE_tax_sum_phylum$Phylum <-row.names(TREE_tax_sum_phylum)
TREE_melt<- melt(TREE_tax_sum_phylum, id.var="Phylum")
nrow(TREE_melt)
site<-rep("TREE",times=nrow(TREE_melt))
TREE_melt<-cbind(TREE_melt,site)

UKFS_tax_sum_phylum = summarize_taxonomy(UKFS, level = 8, report_higher_tax = FALSE)
UKFS_tax_sum_phylum$Phylum <-row.names(UKFS_tax_sum_phylum)
UKFS_melt<- melt(UKFS_tax_sum_phylum, id.var="Phylum")
nrow(UKFS_melt)
site<-rep("UKFS",times=nrow(UKFS_melt))
UKFS_melt<-cbind(UKFS_melt,site)

UNDE_tax_sum_phylum = summarize_taxonomy(UNDE, level = 8, report_higher_tax = FALSE)
UNDE_tax_sum_phylum$Phylum <-row.names(UNDE_tax_sum_phylum)
UNDE_melt<- melt(UNDE_tax_sum_phylum, id.var="Phylum")
nrow(UNDE_melt)
site<-rep("UNDE",times=nrow(UNDE_melt))
UNDE_melt<-cbind(UNDE_melt,site)

WOOD_tax_sum_phylum = summarize_taxonomy(WOOD, level = 8, report_higher_tax = FALSE)
WOOD_tax_sum_phylum$Phylum <-row.names(WOOD_tax_sum_phylum)
WOOD_melt<- melt(WOOD_tax_sum_phylum, id.var="Phylum")
nrow(WOOD_melt)
site<-rep("WOOD",times=nrow(WOOD_melt))
WOOD_melt<-cbind(WOOD_melt,site)

WREF_tax_sum_phylum = summarize_taxonomy(WREF, level = 8, report_higher_tax = FALSE)
WREF_tax_sum_phylum$Phylum <-row.names(WREF_tax_sum_phylum)
WREF_melt<- melt(WREF_tax_sum_phylum, id.var="Phylum")
nrow(WREF_melt)
site<-rep("WREF",times=nrow(WREF_melt))
WREF_melt<-cbind(WREF_melt,site)

YELL_tax_sum_phylum = summarize_taxonomy(YELL, level = 8, report_higher_tax = FALSE)
YELL_tax_sum_phylum$Phylum <-row.names(YELL_tax_sum_phylum)
YELL_melt<- melt(YELL_tax_sum_phylum, id.var="Phylum")
nrow(YELL_melt)
site<-rep("YELL",times=nrow(YELL_melt))
YELL_melt<-cbind(YELL_melt,site)
```

```{r}
phylumfinalbind<-do.call("rbind",list(ABBY_melt, BARR_melt, BART_melt, BLAN_melt, BONA_melt, CLBJ_melt, CPER_melt, DCFS_melt, DEJU_melt, DELA_melt, DSNY_melt, GRSM_melt, GUAN_melt, HARV_melt, HEAL_melt, JERC_melt, JORN_melt, KONA_melt, KONZ_melt, LAJA_melt, LENO_melt, MLBS_melt, MOAB_melt, NIWO_melt, OAES_melt, ONAQ_melt, ORNL_melt, OSBS_melt, PUUM_melt, RMNP_melt, SCBI_melt, SERC_melt, SJER_melt, SOAP_melt, SRER_melt, STEI_melt, STER_melt, TALL_melt, TOOL_melt, TREE_melt, UKFS_melt, UNDE_melt, WOOD_melt, WREF_melt, YELL_melt))
```

phylum graph
```{r}

b<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(23)) + 
  ggtitle("Functional Group")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

b
```

```{r}

ggsave(filename = "Functionalgroupbysite.pdf",
      plot = b,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax_functionalgroup.txt", "eukDiv_mta.txt")
input = filter_data(input, 'nlcdClass', filter_vals = '0')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
```

```{r}
cultivatedCrops = filter_data(input, 'nlcdClass', keep_vals = 'cultivatedCrops')
deciduousForest = filter_data(input, 'nlcdClass', keep_vals = 'deciduousForest')
dwarfScrub = filter_data(input, 'nlcdClass', keep_vals = 'dwarfScrub')
emergentHerbaceousWetlands = filter_data(input, 'nlcdClass', keep_vals = 'emergentHerbaceousWetlands')
evergreenForest = filter_data(input, 'nlcdClass', keep_vals = 'evergreenForest')
grasslandHerbaceous = filter_data(input, 'nlcdClass', keep_vals = 'grasslandHerbaceous')
mixedForest = filter_data(input, 'nlcdClass', keep_vals = 'mixedForest')
pastureHay = filter_data(input, 'nlcdClass', keep_vals = 'pastureHay')
sedgeHerbaceous = filter_data(input, 'nlcdClass', keep_vals = 'sedgeHerbaceous')
shrubScrub = filter_data(input, 'nlcdClass', keep_vals = 'shrubScrub')
woodyWetlands = filter_data(input, 'nlcdClass', keep_vals = 'woodyWetlands')
```

```{r}
otuTAB<-cultivatedCrops$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
cultivatedCrops$data_loaded<-t(hell)

otuTAB<-deciduousForest$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
deciduousForest$data_loaded<-t(hell)

otuTAB<-dwarfScrub$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
dwarfScrub$data_loaded<-t(hell)

otuTAB<-emergentHerbaceousWetlands$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
emergentHerbaceousWetlands$data_loaded<-t(hell)

otuTAB<-evergreenForest$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
evergreenForest$data_loaded<-t(hell)

otuTAB<-grasslandHerbaceous$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
grasslandHerbaceous$data_loaded<-t(hell)

otuTAB<-mixedForest$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
mixedForest$data_loaded<-t(hell)

otuTAB<-pastureHay$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
pastureHay$data_loaded<-t(hell)

otuTAB<-sedgeHerbaceous$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
sedgeHerbaceous$data_loaded<-t(hell)

otuTAB<-shrubScrub$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
shrubScrub$data_loaded<-t(hell)

otuTAB<-woodyWetlands$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
woodyWetlands$data_loaded<-t(hell)
```

```{r}
library(reshape2)
cultivatedCrops_tax_sum_phylum = summarize_taxonomy(cultivatedCrops, level = 8, report_higher_tax = FALSE)
cultivatedCrops_tax_sum_phylum$Phylum <-row.names(cultivatedCrops_tax_sum_phylum)
cultivatedCrops_melt<- melt(cultivatedCrops_tax_sum_phylum, id.var="Phylum")
nrow(cultivatedCrops_melt)
site<-rep("cultivatedCrops",times=nrow(cultivatedCrops_melt))
cultivatedCrops_melt<-cbind(cultivatedCrops_melt,site)

deciduousForest_tax_sum_phylum = summarize_taxonomy(deciduousForest, level = 8, report_higher_tax = FALSE)
deciduousForest_tax_sum_phylum$Phylum <-row.names(deciduousForest_tax_sum_phylum)
deciduousForest_melt<- melt(deciduousForest_tax_sum_phylum, id.var="Phylum")
nrow(deciduousForest_melt)
site<-rep("deciduousForest",times=nrow(deciduousForest_melt))
deciduousForest_melt<-cbind(deciduousForest_melt,site)

dwarfScrub_tax_sum_phylum = summarize_taxonomy(dwarfScrub, level = 8, report_higher_tax = FALSE)
dwarfScrub_tax_sum_phylum$Phylum <-row.names(dwarfScrub_tax_sum_phylum)
dwarfScrub_melt<- melt(dwarfScrub_tax_sum_phylum, id.var="Phylum")
nrow(dwarfScrub_melt)
site<-rep("dwarfScrub",times=nrow(dwarfScrub_melt))
dwarfScrub_melt<-cbind(dwarfScrub_melt,site)

emergentHerbaceousWetlands_tax_sum_phylum = summarize_taxonomy(emergentHerbaceousWetlands, level = 8, report_higher_tax = FALSE)
emergentHerbaceousWetlands_tax_sum_phylum$Phylum <-row.names(emergentHerbaceousWetlands_tax_sum_phylum)
emergentHerbaceousWetlands_melt<- melt(emergentHerbaceousWetlands_tax_sum_phylum, id.var="Phylum")
nrow(emergentHerbaceousWetlands_melt)
site<-rep("emergentHerbaceousWetlands",times=nrow(emergentHerbaceousWetlands_melt))
emergentHerbaceousWetlands_melt<-cbind(emergentHerbaceousWetlands_melt,site)

evergreenForest_tax_sum_phylum = summarize_taxonomy(evergreenForest, level = 8, report_higher_tax = FALSE)
evergreenForest_tax_sum_phylum$Phylum <-row.names(evergreenForest_tax_sum_phylum)
evergreenForest_melt<- melt(evergreenForest_tax_sum_phylum, id.var="Phylum")
nrow(evergreenForest_melt)
site<-rep("evergreenForest",times=nrow(evergreenForest_melt))
evergreenForest_melt<-cbind(evergreenForest_melt,site)

grasslandHerbaceous_tax_sum_phylum = summarize_taxonomy(grasslandHerbaceous, level = 8, report_higher_tax = FALSE)
grasslandHerbaceous_tax_sum_phylum$Phylum <-row.names(grasslandHerbaceous_tax_sum_phylum)
grasslandHerbaceous_melt<- melt(grasslandHerbaceous_tax_sum_phylum, id.var="Phylum")
nrow(grasslandHerbaceous_melt)
site<-rep("grasslandHerbaceous",times=nrow(grasslandHerbaceous_melt))
grasslandHerbaceous_melt<-cbind(grasslandHerbaceous_melt,site)

mixedForest_tax_sum_phylum = summarize_taxonomy(mixedForest, level = 8, report_higher_tax = FALSE)
mixedForest_tax_sum_phylum$Phylum <-row.names(mixedForest_tax_sum_phylum)
mixedForest_melt<- melt(mixedForest_tax_sum_phylum, id.var="Phylum")
nrow(mixedForest_melt)
site<-rep("mixedForest",times=nrow(mixedForest_melt))
mixedForest_melt<-cbind(mixedForest_melt,site)

pastureHay_tax_sum_phylum = summarize_taxonomy(pastureHay, level = 8, report_higher_tax = FALSE)
pastureHay_tax_sum_phylum$Phylum <-row.names(pastureHay_tax_sum_phylum)
pastureHay_melt<- melt(pastureHay_tax_sum_phylum, id.var="Phylum")
nrow(pastureHay_melt)
site<-rep("pastureHay",times=nrow(pastureHay_melt))
pastureHay_melt<-cbind(pastureHay_melt,site)

sedgeHerbaceous_tax_sum_phylum = summarize_taxonomy(sedgeHerbaceous, level = 8, report_higher_tax = FALSE)
sedgeHerbaceous_tax_sum_phylum$Phylum <-row.names(sedgeHerbaceous_tax_sum_phylum)
sedgeHerbaceous_melt<- melt(sedgeHerbaceous_tax_sum_phylum, id.var="Phylum")
nrow(sedgeHerbaceous_melt)
site<-rep("sedgeHerbaceous",times=nrow(sedgeHerbaceous_melt))
sedgeHerbaceous_melt<-cbind(sedgeHerbaceous_melt,site)

shrubScrub_tax_sum_phylum = summarize_taxonomy(shrubScrub, level = 8, report_higher_tax = FALSE)
shrubScrub_tax_sum_phylum$Phylum <-row.names(shrubScrub_tax_sum_phylum)
shrubScrub_melt<- melt(shrubScrub_tax_sum_phylum, id.var="Phylum")
nrow(shrubScrub_melt)
site<-rep("shrubScrub",times=nrow(shrubScrub_melt))
shrubScrub_melt<-cbind(shrubScrub_melt,site)

woodyWetlands_tax_sum_phylum = summarize_taxonomy(woodyWetlands, level = 8, report_higher_tax = FALSE)
woodyWetlands_tax_sum_phylum$Phylum <-row.names(woodyWetlands_tax_sum_phylum)
woodyWetlands_melt<- melt(woodyWetlands_tax_sum_phylum, id.var="Phylum")
nrow(woodyWetlands_melt)
site<-rep("woodyWetlands",times=nrow(woodyWetlands_melt))
woodyWetlands_melt<-cbind(woodyWetlands_melt,site)
```

```{r}
phylumbiomebind<-do.call("rbind",list(cultivatedCrops_melt,deciduousForest_melt,dwarfScrub_melt,emergentHerbaceousWetlands_melt,evergreenForest_melt,grasslandHerbaceous_melt,mixedForest_melt,pastureHay_melt,sedgeHerbaceous_melt,shrubScrub_melt,woodyWetlands_melt))
```

phylum graph
```{r}
library(ggthemes)
z<-ggplot(phylumbiomebind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(6))(23)) + 
  ggtitle("Metazoa Only by Biome")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

z
```

```{r}

ggsave(filename = "Functionalgroupbybiome.pdf",
      plot = z,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

# Regression Plots
```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
```

Filter out chro/mito, which are problems with binning from the pipeline
```{r}
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Fungi:mito')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')

```



```{r}
library(labdsv)
otuTAB<-input$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
input$data_loaded<-t(hell)
```


```{r}
library(vegan)
regressions<-specnumber(t(input$data_loaded))
regressions<-as.data.frame(regressions)
colnames(regressions)[1] ="rich"
regressions<-cbind(regressions,input$map_loaded$Site)
regressions<-cbind(regressions,input$map_loaded$latitude)
regressions<-cbind(regressions,input$map_loaded$soilTemp)
regressions<-cbind(regressions,input$map_loaded$soilMoisture)
regressions<-cbind(regressions,input$map_loaded$organicCPercent)
colnames(regressions)[2] ="Site"
colnames(regressions)[3] ="Latitude"
colnames(regressions)[4] ="Soil_Temp"
colnames(regressions)[5] ="Soil_Moisture"
colnames(regressions)[6] ="Organic_C_Percent"
```

Add MAP and MAT
```{r}
MAPMAT<-read.csv('sites_map_mat.csv',header = TRUE)
library(dplyr)
regressions<-merge(regressions,MAPMAT,by="Site")
```


```{r}
library(ggplot2)
regressions_OC<-subset(regressions,Organic_C_Percent>0)

y<-ggplot(regressions_OC, aes(x = Organic_C_Percent, y = rich, color= Latitude)) + 
  geom_point() + 
  
  ggtitle("Richness vs. Organic C")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
y

ggsave(filename = "richness_organicC.pdf",
      plot = y,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions_OC,rich~log10(Organic_C_Percent))
int
summary(int)
```


```{r}
g<-ggplot(regressions, aes(x = Soil_Temp, y = rich, color= Site)) + 
  geom_point() + 
  
  ggtitle("Richness vs. Soil Temp")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
g

ggsave(filename = "richness_soiltemp.pdf",
      plot = g,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions,rich~Soil_Temp)
int
summary(int)
```

```{r}

g<-ggplot(regressions, aes(x = MAT, y = rich)) + 
  geom_point() + 
  
  ggtitle("Richness vs. Mean Annual Temperature")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
g

ggsave(filename = "richness_MAT.pdf",
      plot = g,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions,rich~MAT)
int
summary(int)
```


```{r}
g<-ggplot(regressions, aes(x = Soil_Moisture, y = rich, color= Site)) + 
  geom_point() + 
  
  ggtitle("Richness vs. Soil Moisture")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
g

ggsave(filename = "richness_soilmoisture.pdf",
      plot = g,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions,rich~Soil_Moisture)
int
summary(int)
```


```{r}
setwd("/Users/leenavilonen/Documents/NEON")
g<-ggplot(regressions, aes(x = MAP, y = rich)) + 
  geom_point() + 
  
  ggtitle("Richness vs. Mean Annual Precipitation")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
g

ggsave(filename = "richness_MAP.pdf",
      plot = g,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions,rich~MAP)
int
summary(int)
```


# Regression Plots - Nematodes ONLY 
```{r, message=FALSE, warning=FALSE}
library(mctoolsr)
library(vegan)
library(car)
library(metacoder)
library(emmeans)
library(multcompView)
library(dplyr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")
```

Filter out chro/mito, which are problems with binning from the pipeline
```{r}
input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Fungi:mito')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input <- filter_taxa_from_input(input, taxa_to_keep = 'Metazoa')
input <- filter_taxa_from_input(input, taxa_to_keep = 'Nematoda')
```



```{r}
library(labdsv)
otuTAB<-input$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
#View(hell)
input$data_loaded<-t(hell)
```


```{r}
library(vegan)
regressions<-specnumber(t(input$data_loaded))
regressions<-as.data.frame(regressions)
colnames(regressions)[1] ="rich"
regressions<-cbind(regressions,input$map_loaded$Site)
regressions<-cbind(regressions,input$map_loaded$latitude)
regressions<-cbind(regressions,input$map_loaded$soilTemp)
regressions<-cbind(regressions,input$map_loaded$soilMoisture)
regressions<-cbind(regressions,input$map_loaded$organicCPercent)
colnames(regressions)[2] ="Site"
colnames(regressions)[3] ="Latitude"
colnames(regressions)[4] ="Soil_Temp"
colnames(regressions)[5] ="Soil_Moisture"
colnames(regressions)[6] ="Organic_C_Percent"
```

Add MAP and MAT
```{r}
MAPMAT<-read.csv('sites_map_mat.csv',header = TRUE)
library(dplyr)
regressions<-merge(regressions,MAPMAT,by="Site")
```

# no outliers detected!!
```{r}
regressions_OC<-subset(regressions,Organic_C_Percent>0)
library(EnvStats)
test <- rosnerTest(regressions_OC$Organic_C_Percent,
  k = 3
)
test
test$all.stats
```

```{r}
library(ggplot2)

regressions_OC<-subset(regressions,Organic_C_Percent>0)
y<-ggplot(regressions_OC, aes(x = log10(Organic_C_Percent), y = rich)) + 
  geom_point() + 
  geom_smooth(method='lm', formula= y~x)+
  ggtitle("Nematodes only - Richness vs. Organic C")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
y
```

```{r}
ggsave(filename = "nematodesonly_richness_organicC_log10scale.pdf",
      plot = y,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions_OC,rich~log10(Organic_C_Percent))
int
summary(int)
int<-lm(data=regressions_OC,log10(Organic_C_Percent)~rich)
int
summary(int)
```


```{r}
g<-ggplot(regressions, aes(x = Soil_Temp, y = rich)) + 
  geom_point() + 
  geom_smooth(method='lm', formula= y~x)+
  ggtitle("Nematodes only_Richness vs. Soil Temp")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
g

ggsave(filename = "nematodes_richness_soiltemp.pdf",
      plot = g,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions,rich~Soil_Temp)
int
summary(int)
```

```{r}

g<-ggplot(regressions, aes(x = MAT, y = rich)) + 
  geom_point() + 
  geom_smooth(method='lm', formula= y~x)+
  ggtitle(" Nematode Richness vs. Mean Annual Temperature")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
g

ggsave(filename = "nematodes_richness_MAT.pdf",
      plot = g,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions,rich~MAT)
int
summary(int)
```


```{r}
g<-ggplot(regressions, aes(x = Soil_Moisture, y = rich)) + 
  geom_point() + 
  geom_smooth(method='lm', formula= y~x)+
  ggtitle("Nematode Richness vs. Soil Moisture")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
g

ggsave(filename = "nematode_richness_soilmoisture.pdf",
      plot = g,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions,rich~Soil_Moisture)
int
summary(int)
```


```{r}
setwd("/Users/leenavilonen/Documents/NEON")
g<-ggplot(regressions, aes(x = MAP, y = rich)) + 
  geom_point() + 
  geom_smooth(method='lm', formula= y~x)+
  ggtitle("Nematode Richness vs. Mean Annual Precipitation")+
  

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))
g

ggsave(filename = "nematode_richness_MAP.pdf",
      plot = g,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600)
```

```{r}
int<-lm(data=regressions,rich~MAP)
int
summary(int)
```

Sorted by MAP, MAT, and latitude. After this do with management.

***FUNGI*** 

Filter by site
```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input <- filter_taxa_from_input(input, taxa_to_keep = 'Fungi')
MLBS = filter_data(input, 'Site', keep_vals = 'MLBS')
HARV = filter_data(input, 'Site', keep_vals = 'HARV')
WOOD = filter_data(input, 'Site', keep_vals = 'WOOD')
LAJA = filter_data(input, 'Site', keep_vals = 'LAJA')
UNDE = filter_data(input, 'Site', keep_vals = 'UNDE')
SRER = filter_data(input, 'Site', keep_vals = 'SRER')
DSNY = filter_data(input, 'Site', keep_vals = 'DSNY')
BONA = filter_data(input, 'Site', keep_vals = 'BONA')
GUAN = filter_data(input, 'Site', keep_vals = 'GUAN')
ABBY = filter_data(input, 'Site', keep_vals = 'ABBY')
BARR = filter_data(input, 'Site', keep_vals = 'BARR')
BART = filter_data(input, 'Site', keep_vals = 'BART')
BLAN = filter_data(input, 'Site', keep_vals = 'BLAN')
CLBJ = filter_data(input, 'Site', keep_vals = 'CLBJ')
CPER = filter_data(input, 'Site', keep_vals = 'CPER')
DCFS = filter_data(input, 'Site', keep_vals = 'DCFS')
DEJU = filter_data(input, 'Site', keep_vals = 'DEJU')
DELA = filter_data(input, 'Site', keep_vals = 'DELA')
GRSM = filter_data(input, 'Site', keep_vals = 'GRSM')
HEAL = filter_data(input, 'Site', keep_vals = 'HEAL')
JERC = filter_data(input, 'Site', keep_vals = 'JERC')
JORN = filter_data(input, 'Site', keep_vals = 'JORN')
KONA = filter_data(input, 'Site', keep_vals = 'KONA')
KONZ = filter_data(input, 'Site', keep_vals = 'KONZ')
LENO = filter_data(input, 'Site', keep_vals = 'LENO')
MOAB = filter_data(input, 'Site', keep_vals = 'MOAB')
NIWO = filter_data(input, 'Site', keep_vals = 'NIWO')
OAES = filter_data(input, 'Site', keep_vals = 'OAES')
ONAQ = filter_data(input, 'Site', keep_vals = 'ONAQ')
ORNL = filter_data(input, 'Site', keep_vals = 'ORNL')
OSBS = filter_data(input, 'Site', keep_vals = 'OSBS')
PUUM = filter_data(input, 'Site', keep_vals = 'PUUM')
RMNP = filter_data(input, 'Site', keep_vals = 'RMNP')
SCBI = filter_data(input, 'Site', keep_vals = 'SCBI')
SERC = filter_data(input, 'Site', keep_vals = 'SERC')
SJER = filter_data(input, 'Site', keep_vals = 'SJER')
SOAP = filter_data(input, 'Site', keep_vals = 'SOAP')
STEI = filter_data(input, 'Site', keep_vals = 'STEI')
STER = filter_data(input, 'Site', keep_vals = 'STER')
TALL = filter_data(input, 'Site', keep_vals = 'TALL')
TOOL = filter_data(input, 'Site', keep_vals = 'TOOL')
TREE = filter_data(input, 'Site', keep_vals = 'TREE')
UKFS = filter_data(input, 'Site', keep_vals = 'UKFS')
WREF = filter_data(input, 'Site', keep_vals = 'WREF')
YELL = filter_data(input, 'Site', keep_vals = 'YELL')
```

```{r}
library(labdsv)
otuTAB<-ABBY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ABBY$data_loaded<-t(hell)

otuTAB<-BARR$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BARR$data_loaded<-t(hell)

otuTAB<-BART$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BART$data_loaded<-t(hell)

otuTAB<-BLAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BLAN$data_loaded<-t(hell)

otuTAB<-BONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BONA$data_loaded<-t(hell)

otuTAB<-CLBJ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CLBJ$data_loaded<-t(hell)

otuTAB<-CPER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CPER$data_loaded<-t(hell)

otuTAB<-DCFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DCFS$data_loaded<-t(hell)

otuTAB<-DELA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DELA$data_loaded<-t(hell)

otuTAB<-DSNY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DSNY$data_loaded<-t(hell)

otuTAB<-GRSM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GRSM$data_loaded<-t(hell)

otuTAB<-GUAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GUAN$data_loaded<-t(hell)

otuTAB<-HARV$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HARV$data_loaded<-t(hell)

otuTAB<-HEAL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HEAL$data_loaded<-t(hell)

otuTAB<-JERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JERC$data_loaded<-t(hell)

otuTAB<-JORN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JORN$data_loaded<-t(hell)

otuTAB<-KONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONA$data_loaded<-t(hell)

otuTAB<-KONZ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONZ$data_loaded<-t(hell)

otuTAB<-LAJA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LAJA$data_loaded<-t(hell)

otuTAB<-LENO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LENO$data_loaded<-t(hell)

otuTAB<-MLBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MLBS$data_loaded<-t(hell)

otuTAB<-MOAB$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MOAB$data_loaded<-t(hell)

otuTAB<-NIWO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
NIWO$data_loaded<-t(hell)

otuTAB<-OAES$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OAES$data_loaded<-t(hell)

otuTAB<-ONAQ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ONAQ$data_loaded<-t(hell)

otuTAB<-ORNL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ORNL$data_loaded<-t(hell)

otuTAB<-OSBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OSBS$data_loaded<-t(hell)

otuTAB<-PUUM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
PUUM$data_loaded<-t(hell)

otuTAB<-RMNP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
RMNP$data_loaded<-t(hell)

otuTAB<-SCBI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SCBI$data_loaded<-t(hell)

otuTAB<-SERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SERC$data_loaded<-t(hell)

otuTAB<-SJER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SJER$data_loaded<-t(hell)

otuTAB<-SOAP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SOAP$data_loaded<-t(hell)

otuTAB<-SRER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SRER$data_loaded<-t(hell)

otuTAB<-STEI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STEI$data_loaded<-t(hell)

otuTAB<-STER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STER$data_loaded<-t(hell)

otuTAB<-TALL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TALL$data_loaded<-t(hell)

otuTAB<-TOOL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TOOL$data_loaded<-t(hell)

otuTAB<-TREE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TREE$data_loaded<-t(hell)

otuTAB<-UKFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UKFS$data_loaded<-t(hell)

otuTAB<-UNDE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UNDE$data_loaded<-t(hell)

otuTAB<-WOOD$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WOOD$data_loaded<-t(hell)

otuTAB<-WREF$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WREF$data_loaded<-t(hell)

otuTAB<-YELL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
YELL$data_loaded<-t(hell)
```


```{r}
library(reshape2)
ABBY_tax_sum_phylum = summarize_taxonomy(ABBY, level = 4, report_higher_tax = FALSE)
ABBY_tax_sum_phylum$Phylum <-row.names(ABBY_tax_sum_phylum)
ABBY_melt<- melt(ABBY_tax_sum_phylum, id.var="Phylum")
site<-rep("ABBY",times=nrow(ABBY_melt))
ABBY_melt<-cbind(ABBY_melt,site)

BARR_tax_sum_phylum = summarize_taxonomy(BARR, level = 4, report_higher_tax = FALSE)
BARR_tax_sum_phylum$Phylum <-row.names(BARR_tax_sum_phylum)
BARR_melt<- melt(BARR_tax_sum_phylum, id.var="Phylum")
nrow(BARR_melt)
site<-rep("BARR",times=nrow(BARR_melt))
BARR_melt<-cbind(BARR_melt,site)

BART_tax_sum_phylum = summarize_taxonomy(BART, level = 4, report_higher_tax = FALSE)
BART_tax_sum_phylum$Phylum <-row.names(BART_tax_sum_phylum)
BART_melt<- melt(BART_tax_sum_phylum, id.var="Phylum")
nrow(BART_melt)
site<-rep("BART",times=nrow(BART_melt))
BART_melt<-cbind(BART_melt,site)

BLAN_tax_sum_phylum = summarize_taxonomy(BLAN, level = 4, report_higher_tax = FALSE)
BLAN_tax_sum_phylum$Phylum <-row.names(BLAN_tax_sum_phylum)
BLAN_melt<- melt(BLAN_tax_sum_phylum, id.var="Phylum")
nrow(BLAN_melt)
site<-rep("BLAN",times=nrow(BLAN_melt))
BLAN_melt<-cbind(BLAN_melt,site)

BONA_tax_sum_phylum = summarize_taxonomy(BONA, level = 4, report_higher_tax = FALSE)
BONA_tax_sum_phylum$Phylum <-row.names(BONA_tax_sum_phylum)
BONA_melt<- melt(BONA_tax_sum_phylum, id.var="Phylum")
nrow(BONA_melt)
site<-rep("BONA",times=nrow(BONA_melt))
BONA_melt<-cbind(BONA_melt,site)

CLBJ_tax_sum_phylum = summarize_taxonomy(CLBJ, level = 4, report_higher_tax = FALSE)
CLBJ_tax_sum_phylum$Phylum <-row.names(CLBJ_tax_sum_phylum)
CLBJ_melt<- melt(CLBJ_tax_sum_phylum, id.var="Phylum")
nrow(CLBJ_melt)
site<-rep("CLBJ",times=nrow(CLBJ_melt))
CLBJ_melt<-cbind(CLBJ_melt,site)

CPER_tax_sum_phylum = summarize_taxonomy(CPER, level = 4, report_higher_tax = FALSE)
CPER_tax_sum_phylum$Phylum <-row.names(CPER_tax_sum_phylum)
CPER_melt<- melt(CPER_tax_sum_phylum, id.var="Phylum")
nrow(CPER_melt)
site<-rep("CPER",times=nrow(CPER_melt))
CPER_melt<-cbind(CPER_melt,site)

DCFS_tax_sum_phylum = summarize_taxonomy(DCFS, level = 4, report_higher_tax = FALSE)
DCFS_tax_sum_phylum$Phylum <-row.names(DCFS_tax_sum_phylum)
DCFS_melt<- melt(DCFS_tax_sum_phylum, id.var="Phylum")
nrow(DCFS_melt)
site<-rep("DCFS",times=nrow(DCFS_melt))
DCFS_melt<-cbind(DCFS_melt,site)

DEJU_tax_sum_phylum = summarize_taxonomy(DEJU, level = 4, report_higher_tax = FALSE)
DEJU_tax_sum_phylum$Phylum <-row.names(DEJU_tax_sum_phylum)
DEJU_melt<- melt(DEJU_tax_sum_phylum, id.var="Phylum")
nrow(DEJU_melt)
site<-rep("DEJU",times=nrow(DEJU_melt))
DEJU_melt<-cbind(DEJU_melt,site)

DELA_tax_sum_phylum = summarize_taxonomy(DELA, level = 4, report_higher_tax = FALSE)
DELA_tax_sum_phylum$Phylum <-row.names(DELA_tax_sum_phylum)
DELA_melt<- melt(DELA_tax_sum_phylum, id.var="Phylum")
nrow(DELA_melt)
site<-rep("DELA",times=nrow(DELA_melt))
DELA_melt<-cbind(DELA_melt,site)

DSNY_tax_sum_phylum = summarize_taxonomy(DSNY, level = 4, report_higher_tax = FALSE)
DSNY_tax_sum_phylum$Phylum <-row.names(DSNY_tax_sum_phylum)
DSNY_melt<- melt(DSNY_tax_sum_phylum, id.var="Phylum")
nrow(DSNY_melt)
site<-rep("DSNY",times=nrow(DSNY_melt))
DSNY_melt<-cbind(DSNY_melt,site)

GRSM_tax_sum_phylum = summarize_taxonomy(GRSM, level = 4, report_higher_tax = FALSE)
GRSM_tax_sum_phylum$Phylum <-row.names(GRSM_tax_sum_phylum)
GRSM_melt<- melt(GRSM_tax_sum_phylum, id.var="Phylum")
nrow(GRSM_melt)
site<-rep("GRSM",times=nrow(GRSM_melt))
GRSM_melt<-cbind(GRSM_melt,site)

GUAN_tax_sum_phylum = summarize_taxonomy(GUAN, level = 4, report_higher_tax = FALSE)
GUAN_tax_sum_phylum$Phylum <-row.names(GUAN_tax_sum_phylum)
GUAN_melt<- melt(GUAN_tax_sum_phylum, id.var="Phylum")
nrow(GUAN_melt)
site<-rep("GUAN",times=nrow(GUAN_melt))
GUAN_melt<-cbind(GUAN_melt,site)

HARV_tax_sum_phylum = summarize_taxonomy(HARV, level = 4, report_higher_tax = FALSE)
HARV_tax_sum_phylum$Phylum <-row.names(HARV_tax_sum_phylum)
HARV_melt<- melt(HARV_tax_sum_phylum, id.var="Phylum")
nrow(HARV_melt)
site<-rep("HARV",times=nrow(HARV_melt))
HARV_melt<-cbind(HARV_melt,site)

HEAL_tax_sum_phylum = summarize_taxonomy(HEAL, level = 4, report_higher_tax = FALSE)
HEAL_tax_sum_phylum$Phylum <-row.names(HEAL_tax_sum_phylum)
HEAL_melt<- melt(HEAL_tax_sum_phylum, id.var="Phylum")
nrow(HEAL_melt)
site<-rep("HEAL",times=nrow(HEAL_melt))
HEAL_melt<-cbind(HEAL_melt,site)

JERC_tax_sum_phylum = summarize_taxonomy(JERC, level = 4, report_higher_tax = FALSE)
JERC_tax_sum_phylum$Phylum <-row.names(JERC_tax_sum_phylum)
JERC_melt<- melt(JERC_tax_sum_phylum, id.var="Phylum")
nrow(JERC_melt)
site<-rep("JERC",times=nrow(JERC_melt))
JERC_melt<-cbind(JERC_melt,site)

JORN_tax_sum_phylum = summarize_taxonomy(JORN, level = 4, report_higher_tax = FALSE)
JORN_tax_sum_phylum$Phylum <-row.names(JORN_tax_sum_phylum)
JORN_melt<- melt(JORN_tax_sum_phylum, id.var="Phylum")
nrow(JORN_melt)
site<-rep("JORN",times=nrow(JORN_melt))
JORN_melt<-cbind(JORN_melt,site)

KONA_tax_sum_phylum = summarize_taxonomy(KONA, level = 4, report_higher_tax = FALSE)
KONA_tax_sum_phylum$Phylum <-row.names(KONA_tax_sum_phylum)
KONA_melt<- melt(KONA_tax_sum_phylum, id.var="Phylum")
nrow(KONA_melt)
site<-rep("KONA",times=nrow(KONA_melt))
KONA_melt<-cbind(KONA_melt,site)

KONZ_tax_sum_phylum = summarize_taxonomy(KONZ, level = 4, report_higher_tax = FALSE)
KONZ_tax_sum_phylum$Phylum <-row.names(KONZ_tax_sum_phylum)
KONZ_melt<- melt(KONZ_tax_sum_phylum, id.var="Phylum")
nrow(KONZ_melt)
site<-rep("KONZ",times=nrow(KONZ_melt))
KONZ_melt<-cbind(KONZ_melt,site)

LAJA_tax_sum_phylum = summarize_taxonomy(LAJA, level = 4, report_higher_tax = FALSE)
LAJA_tax_sum_phylum$Phylum <-row.names(LAJA_tax_sum_phylum)
LAJA_melt<- melt(LAJA_tax_sum_phylum, id.var="Phylum")
nrow(LAJA_melt)
site<-rep("LAJA",times=nrow(LAJA_melt))
LAJA_melt<-cbind(LAJA_melt,site)

LENO_tax_sum_phylum = summarize_taxonomy(LENO, level = 4, report_higher_tax = FALSE)
LENO_tax_sum_phylum$Phylum <-row.names(LENO_tax_sum_phylum)
LENO_melt<- melt(LENO_tax_sum_phylum, id.var="Phylum")
nrow(LENO_melt)
site<-rep("LENO",times=nrow(LENO_melt))
LENO_melt<-cbind(LENO_melt,site)

MLBS_tax_sum_phylum = summarize_taxonomy(MLBS, level = 4, report_higher_tax = FALSE)
MLBS_tax_sum_phylum$Phylum <-row.names(MLBS_tax_sum_phylum)
MLBS_melt<- melt(MLBS_tax_sum_phylum, id.var="Phylum")
nrow(MLBS_melt)
site<-rep("MLBS",times=nrow(MLBS_melt))
MLBS_melt<-cbind(MLBS_melt,site)

MOAB_tax_sum_phylum = summarize_taxonomy(MOAB, level = 4, report_higher_tax = FALSE)
MOAB_tax_sum_phylum$Phylum <-row.names(MOAB_tax_sum_phylum)
MOAB_melt<- melt(MOAB_tax_sum_phylum, id.var="Phylum")
nrow(MOAB_melt)
site<-rep("MOAB",times=nrow(MOAB_melt))
MOAB_melt<-cbind(MOAB_melt,site)

NIWO_tax_sum_phylum = summarize_taxonomy(NIWO, level = 4, report_higher_tax = FALSE)
NIWO_tax_sum_phylum$Phylum <-row.names(NIWO_tax_sum_phylum)
NIWO_melt<- melt(NIWO_tax_sum_phylum, id.var="Phylum")
nrow(NIWO_melt)
site<-rep("NIWO",times=nrow(NIWO_melt))
NIWO_melt<-cbind(NIWO_melt,site)

OAES_tax_sum_phylum = summarize_taxonomy(OAES, level = 4, report_higher_tax = FALSE)
OAES_tax_sum_phylum$Phylum <-row.names(OAES_tax_sum_phylum)
OAES_melt<- melt(OAES_tax_sum_phylum, id.var="Phylum")
nrow(OAES_melt)
site<-rep("OAES",times=nrow(OAES_melt))
OAES_melt<-cbind(OAES_melt,site)

ONAQ_tax_sum_phylum = summarize_taxonomy(ONAQ, level = 4, report_higher_tax = FALSE)
ONAQ_tax_sum_phylum$Phylum <-row.names(ONAQ_tax_sum_phylum)
ONAQ_melt<- melt(ONAQ_tax_sum_phylum, id.var="Phylum")
nrow(ONAQ_melt)
site<-rep("ONAQ",times=nrow(ONAQ_melt))
ONAQ_melt<-cbind(ONAQ_melt,site)

ORNL_tax_sum_phylum = summarize_taxonomy(ORNL, level = 4, report_higher_tax = FALSE)
ORNL_tax_sum_phylum$Phylum <-row.names(ORNL_tax_sum_phylum)
ORNL_melt<- melt(ORNL_tax_sum_phylum, id.var="Phylum")
nrow(ORNL_melt)
site<-rep("ORNL",times=nrow(ORNL_melt))
ORNL_melt<-cbind(ORNL_melt,site)

OSBS_tax_sum_phylum = summarize_taxonomy(OSBS, level = 4, report_higher_tax = FALSE)
OSBS_tax_sum_phylum$Phylum <-row.names(OSBS_tax_sum_phylum)
OSBS_melt<- melt(OSBS_tax_sum_phylum, id.var="Phylum")
nrow(OSBS_melt)
site<-rep("OSBS",times=nrow(OSBS_melt))
OSBS_melt<-cbind(OSBS_melt,site)

PUUM_tax_sum_phylum = summarize_taxonomy(PUUM, level = 4, report_higher_tax = FALSE)
PUUM_tax_sum_phylum$Phylum <-row.names(PUUM_tax_sum_phylum)
PUUM_melt<- melt(PUUM_tax_sum_phylum, id.var="Phylum")
nrow(PUUM_melt)
site<-rep("PUUM",times=nrow(PUUM_melt))
PUUM_melt<-cbind(PUUM_melt,site)

RMNP_tax_sum_phylum = summarize_taxonomy(RMNP, level = 4, report_higher_tax = FALSE)
RMNP_tax_sum_phylum$Phylum <-row.names(RMNP_tax_sum_phylum)
RMNP_melt<- melt(RMNP_tax_sum_phylum, id.var="Phylum")
nrow(RMNP_melt)
site<-rep("RMNP",times=nrow(RMNP_melt))
RMNP_melt<-cbind(RMNP_melt,site)

SCBI_tax_sum_phylum = summarize_taxonomy(SCBI, level = 4, report_higher_tax = FALSE)
SCBI_tax_sum_phylum$Phylum <-row.names(SCBI_tax_sum_phylum)
SCBI_melt<- melt(SCBI_tax_sum_phylum, id.var="Phylum")
nrow(SCBI_melt)
site<-rep("SCBI",times=nrow(SCBI_melt))
SCBI_melt<-cbind(SCBI_melt,site)

SERC_tax_sum_phylum = summarize_taxonomy(SERC, level = 4, report_higher_tax = FALSE)
SERC_tax_sum_phylum$Phylum <-row.names(SERC_tax_sum_phylum)
SERC_melt<- melt(SERC_tax_sum_phylum, id.var="Phylum")
nrow(SERC_melt)
site<-rep("SERC",times=nrow(SERC_melt))
SERC_melt<-cbind(SERC_melt,site)

SJER_tax_sum_phylum = summarize_taxonomy(SJER, level = 4, report_higher_tax = FALSE)
SJER_tax_sum_phylum$Phylum <-row.names(SJER_tax_sum_phylum)
SJER_melt<- melt(SJER_tax_sum_phylum, id.var="Phylum")
nrow(SJER_melt)
site<-rep("SJER",times=nrow(SJER_melt))
SJER_melt<-cbind(SJER_melt,site)

SOAP_tax_sum_phylum = summarize_taxonomy(SOAP, level = 4, report_higher_tax = FALSE)
SOAP_tax_sum_phylum$Phylum <-row.names(SOAP_tax_sum_phylum)
SOAP_melt<- melt(SOAP_tax_sum_phylum, id.var="Phylum")
nrow(SOAP_melt)
site<-rep("SOAP",times=nrow(SOAP_melt))
SOAP_melt<-cbind(SOAP_melt,site)

SRER_tax_sum_phylum = summarize_taxonomy(SRER, level = 4, report_higher_tax = FALSE)
SRER_tax_sum_phylum$Phylum <-row.names(SRER_tax_sum_phylum)
SRER_melt<- melt(SRER_tax_sum_phylum, id.var="Phylum")
nrow(SRER_melt)
site<-rep("SRER",times=nrow(SRER_melt))
SRER_melt<-cbind(SRER_melt,site)

STEI_tax_sum_phylum = summarize_taxonomy(STEI, level = 4, report_higher_tax = FALSE)
STEI_tax_sum_phylum$Phylum <-row.names(STEI_tax_sum_phylum)
STEI_melt<- melt(STEI_tax_sum_phylum, id.var="Phylum")
nrow(STEI_melt)
site<-rep("STEI",times=nrow(STEI_melt))
STEI_melt<-cbind(STEI_melt,site)

STER_tax_sum_phylum = summarize_taxonomy(STER, level = 4, report_higher_tax = FALSE)
STER_tax_sum_phylum$Phylum <-row.names(STER_tax_sum_phylum)
STER_melt<- melt(STER_tax_sum_phylum, id.var="Phylum")
nrow(STER_melt)
site<-rep("STER",times=nrow(STER_melt))
STER_melt<-cbind(STER_melt,site)

TALL_tax_sum_phylum = summarize_taxonomy(TALL, level = 4, report_higher_tax = FALSE)
TALL_tax_sum_phylum$Phylum <-row.names(TALL_tax_sum_phylum)
TALL_melt<- melt(TALL_tax_sum_phylum, id.var="Phylum")
nrow(TALL_melt)
site<-rep("TALL",times=nrow(TALL_melt))
TALL_melt<-cbind(TALL_melt,site)

TOOL_tax_sum_phylum = summarize_taxonomy(TOOL, level = 4, report_higher_tax = FALSE)
TOOL_tax_sum_phylum$Phylum <-row.names(TOOL_tax_sum_phylum)
TOOL_melt<- melt(TOOL_tax_sum_phylum, id.var="Phylum")
nrow(TOOL_melt)
site<-rep("TOOL",times=nrow(TOOL_melt))
TOOL_melt<-cbind(TOOL_melt,site)

TREE_tax_sum_phylum = summarize_taxonomy(TREE, level = 4, report_higher_tax = FALSE)
TREE_tax_sum_phylum$Phylum <-row.names(TREE_tax_sum_phylum)
TREE_melt<- melt(TREE_tax_sum_phylum, id.var="Phylum")
nrow(TREE_melt)
site<-rep("TREE",times=nrow(TREE_melt))
TREE_melt<-cbind(TREE_melt,site)

UKFS_tax_sum_phylum = summarize_taxonomy(UKFS, level = 4, report_higher_tax = FALSE)
UKFS_tax_sum_phylum$Phylum <-row.names(UKFS_tax_sum_phylum)
UKFS_melt<- melt(UKFS_tax_sum_phylum, id.var="Phylum")
nrow(UKFS_melt)
site<-rep("UKFS",times=nrow(UKFS_melt))
UKFS_melt<-cbind(UKFS_melt,site)

UNDE_tax_sum_phylum = summarize_taxonomy(UNDE, level = 4, report_higher_tax = FALSE)
UNDE_tax_sum_phylum$Phylum <-row.names(UNDE_tax_sum_phylum)
UNDE_melt<- melt(UNDE_tax_sum_phylum, id.var="Phylum")
nrow(UNDE_melt)
site<-rep("UNDE",times=nrow(UNDE_melt))
UNDE_melt<-cbind(UNDE_melt,site)

WOOD_tax_sum_phylum = summarize_taxonomy(WOOD, level = 4, report_higher_tax = FALSE)
WOOD_tax_sum_phylum$Phylum <-row.names(WOOD_tax_sum_phylum)
WOOD_melt<- melt(WOOD_tax_sum_phylum, id.var="Phylum")
nrow(WOOD_melt)
site<-rep("WOOD",times=nrow(WOOD_melt))
WOOD_melt<-cbind(WOOD_melt,site)

WREF_tax_sum_phylum = summarize_taxonomy(WREF, level = 4, report_higher_tax = FALSE)
WREF_tax_sum_phylum$Phylum <-row.names(WREF_tax_sum_phylum)
WREF_melt<- melt(WREF_tax_sum_phylum, id.var="Phylum")
nrow(WREF_melt)
site<-rep("WREF",times=nrow(WREF_melt))
WREF_melt<-cbind(WREF_melt,site)

YELL_tax_sum_phylum = summarize_taxonomy(YELL, level = 4, report_higher_tax = FALSE)
YELL_tax_sum_phylum$Phylum <-row.names(YELL_tax_sum_phylum)
YELL_melt<- melt(YELL_tax_sum_phylum, id.var="Phylum")
nrow(YELL_melt)
site<-rep("YELL",times=nrow(YELL_melt))
YELL_melt<-cbind(YELL_melt,site)
```

```{r}
phylumfinalbind<-do.call("rbind",list(ABBY_melt, BARR_melt, BART_melt, BLAN_melt, BONA_melt, CLBJ_melt, CPER_melt, DCFS_melt, DEJU_melt, DELA_melt, DSNY_melt, GRSM_melt, GUAN_melt, HARV_melt, HEAL_melt, JERC_melt, JORN_melt, KONA_melt, KONZ_melt, LAJA_melt, LENO_melt, MLBS_melt, MOAB_melt, NIWO_melt, OAES_melt, ONAQ_melt, ORNL_melt, OSBS_melt, PUUM_melt, RMNP_melt, SCBI_melt, SERC_melt, SJER_melt, SOAP_melt, SRER_melt, STEI_melt, STER_melt, TALL_melt, TOOL_melt, TREE_melt, UKFS_melt, UNDE_melt, WOOD_melt, WREF_melt, YELL_melt))
```

# Latitude first by phylum
phylum graph
```{r}
positions <- c("GUAN","LAJA","PUUM","DSNY","OSBS","JERC","LENO","SRER","DELA","JORN","TALL","CLBJ","OAES","GRSM","ORNL","SOAP","SJER","MLBS","MOAB","SERC","SCBI","BLAN","UKFS","KONZ","KONA","NIWO","ONAQ","RMNP","STER","CPER","HARV","BART","YELL","TREE","STEI","ABBY","WREF","UNDE","WOOD","DCFS","HEAL","DEJU","BONA","TOOL","BARR")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_x_discrete(limits = positions)+
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Fungi Graphs by Latitude")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "fungi_site_by_latitude.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

# MAP first by phylum
phylum graph
```{r}
positions <- c("BARR", "BONA", "JORN","ONAQ","DEJU","TOOL","MOAB","CPER","SRER","HEAL","STER","DCFS","YELL","WOOD","SJER","RMNP","OAES","STEI","TREE","UNDE","LAJA","GUAN","KONA","KONZ","SOAP","CLBJ","BLAN","UKFS","NIWO","SERC","SCBI","HARV","DSNY","MLBS","OSBS","JERC","BART","ORNL","DELA","GRSM","TALL","LENO","WREF","ABBY","PUUM")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_x_discrete(limits = positions)+
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Fungi Graphs by MAP")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "fungi_site_by_MAP.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```


# MAT first by phylum
phylum graph
```{r}
positions <- c("HEAL","BARR","BONA","DEJU","TOOL","NIWO","ABBY","MOAB","SCBI","BLAN","KONZ","KONA","UKFS","PUUM","GRSM","SOAP","SERC","ORNL","OAES","JORN","SJER","TALL","CLBJ","DELA","LENO","JERC","SRER","RMNP","OSBS","DSNY","GUAN","LAJA","YELL","UNDE","STEI","TREE","DCFS","WOOD","BART","HARV","CPER","MLBS","ONAQ","WREF","STER")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_x_discrete(limits = positions)+
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Fungi Graphs by MAT")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "fungi_site_by_MAT.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
phylumfinalbind<-do.call("rbind",list(CPER_melt,STER_melt,KONZ_melt,KONA_melt,WOOD_melt,DCFS_melt,GUAN_melt,LAJA_melt,WREF_melt,ABBY_melt,TREE_melt,STEI_melt))
```

```{r}
positions <- c("CPER","STER","KONZ","KONA","WOOD","DCFS","GUAN","LAJA","WREF","ABBY","TREE","STEI")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_x_discrete(limits = positions)+
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Fungi by management")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "fungi_site_by_management.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```



***START WITH METAZOA AND FIX BELOW*** 

Filter by site
```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
input <- filter_taxa_from_input(input, taxa_to_keep = 'Metazoa')
MLBS = filter_data(input, 'Site', keep_vals = 'MLBS')
HARV = filter_data(input, 'Site', keep_vals = 'HARV')
WOOD = filter_data(input, 'Site', keep_vals = 'WOOD')
LAJA = filter_data(input, 'Site', keep_vals = 'LAJA')
UNDE = filter_data(input, 'Site', keep_vals = 'UNDE')
SRER = filter_data(input, 'Site', keep_vals = 'SRER')
DSNY = filter_data(input, 'Site', keep_vals = 'DSNY')
BONA = filter_data(input, 'Site', keep_vals = 'BONA')
GUAN = filter_data(input, 'Site', keep_vals = 'GUAN')
ABBY = filter_data(input, 'Site', keep_vals = 'ABBY')
BARR = filter_data(input, 'Site', keep_vals = 'BARR')
BART = filter_data(input, 'Site', keep_vals = 'BART')
BLAN = filter_data(input, 'Site', keep_vals = 'BLAN')
CLBJ = filter_data(input, 'Site', keep_vals = 'CLBJ')
CPER = filter_data(input, 'Site', keep_vals = 'CPER')
DCFS = filter_data(input, 'Site', keep_vals = 'DCFS')
DEJU = filter_data(input, 'Site', keep_vals = 'DEJU')
DELA = filter_data(input, 'Site', keep_vals = 'DELA')
GRSM = filter_data(input, 'Site', keep_vals = 'GRSM')
HEAL = filter_data(input, 'Site', keep_vals = 'HEAL')
JERC = filter_data(input, 'Site', keep_vals = 'JERC')
JORN = filter_data(input, 'Site', keep_vals = 'JORN')
KONA = filter_data(input, 'Site', keep_vals = 'KONA')
KONZ = filter_data(input, 'Site', keep_vals = 'KONZ')
LENO = filter_data(input, 'Site', keep_vals = 'LENO')
MOAB = filter_data(input, 'Site', keep_vals = 'MOAB')
NIWO = filter_data(input, 'Site', keep_vals = 'NIWO')
OAES = filter_data(input, 'Site', keep_vals = 'OAES')
ONAQ = filter_data(input, 'Site', keep_vals = 'ONAQ')
ORNL = filter_data(input, 'Site', keep_vals = 'ORNL')
OSBS = filter_data(input, 'Site', keep_vals = 'OSBS')
PUUM = filter_data(input, 'Site', keep_vals = 'PUUM')
RMNP = filter_data(input, 'Site', keep_vals = 'RMNP')
SCBI = filter_data(input, 'Site', keep_vals = 'SCBI')
SERC = filter_data(input, 'Site', keep_vals = 'SERC')
SJER = filter_data(input, 'Site', keep_vals = 'SJER')
SOAP = filter_data(input, 'Site', keep_vals = 'SOAP')
STEI = filter_data(input, 'Site', keep_vals = 'STEI')
STER = filter_data(input, 'Site', keep_vals = 'STER')
TALL = filter_data(input, 'Site', keep_vals = 'TALL')
TOOL = filter_data(input, 'Site', keep_vals = 'TOOL')
TREE = filter_data(input, 'Site', keep_vals = 'TREE')
UKFS = filter_data(input, 'Site', keep_vals = 'UKFS')
WREF = filter_data(input, 'Site', keep_vals = 'WREF')
YELL = filter_data(input, 'Site', keep_vals = 'YELL')
```

```{r}
library(labdsv)
otuTAB<-ABBY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ABBY$data_loaded<-t(hell)

otuTAB<-BARR$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BARR$data_loaded<-t(hell)

otuTAB<-BART$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BART$data_loaded<-t(hell)

otuTAB<-BLAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BLAN$data_loaded<-t(hell)

otuTAB<-BONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BONA$data_loaded<-t(hell)

otuTAB<-CLBJ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CLBJ$data_loaded<-t(hell)

otuTAB<-CPER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CPER$data_loaded<-t(hell)

otuTAB<-DCFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DCFS$data_loaded<-t(hell)

otuTAB<-DELA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DELA$data_loaded<-t(hell)

otuTAB<-DSNY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DSNY$data_loaded<-t(hell)

otuTAB<-GRSM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GRSM$data_loaded<-t(hell)

otuTAB<-GUAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GUAN$data_loaded<-t(hell)

otuTAB<-HARV$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HARV$data_loaded<-t(hell)

otuTAB<-HEAL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HEAL$data_loaded<-t(hell)

otuTAB<-JERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JERC$data_loaded<-t(hell)

otuTAB<-JORN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JORN$data_loaded<-t(hell)

otuTAB<-KONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONA$data_loaded<-t(hell)

otuTAB<-KONZ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONZ$data_loaded<-t(hell)

otuTAB<-LAJA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LAJA$data_loaded<-t(hell)

otuTAB<-LENO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LENO$data_loaded<-t(hell)

otuTAB<-MLBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MLBS$data_loaded<-t(hell)

otuTAB<-MOAB$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MOAB$data_loaded<-t(hell)

otuTAB<-NIWO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
NIWO$data_loaded<-t(hell)

otuTAB<-OAES$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OAES$data_loaded<-t(hell)

otuTAB<-ONAQ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ONAQ$data_loaded<-t(hell)

otuTAB<-ORNL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ORNL$data_loaded<-t(hell)

otuTAB<-OSBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OSBS$data_loaded<-t(hell)

otuTAB<-PUUM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
PUUM$data_loaded<-t(hell)

otuTAB<-RMNP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
RMNP$data_loaded<-t(hell)

otuTAB<-SCBI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SCBI$data_loaded<-t(hell)

otuTAB<-SERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SERC$data_loaded<-t(hell)

otuTAB<-SJER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SJER$data_loaded<-t(hell)

otuTAB<-SOAP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SOAP$data_loaded<-t(hell)

otuTAB<-SRER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SRER$data_loaded<-t(hell)

otuTAB<-STEI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STEI$data_loaded<-t(hell)

otuTAB<-STER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STER$data_loaded<-t(hell)

otuTAB<-TALL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TALL$data_loaded<-t(hell)

otuTAB<-TOOL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TOOL$data_loaded<-t(hell)

otuTAB<-TREE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TREE$data_loaded<-t(hell)

otuTAB<-UKFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UKFS$data_loaded<-t(hell)

otuTAB<-UNDE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UNDE$data_loaded<-t(hell)

otuTAB<-WOOD$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WOOD$data_loaded<-t(hell)

otuTAB<-WREF$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WREF$data_loaded<-t(hell)

otuTAB<-YELL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
YELL$data_loaded<-t(hell)
```


```{r}
library(reshape2)
ABBY_tax_sum_phylum = summarize_taxonomy(ABBY, level = 4, report_higher_tax = FALSE)
ABBY_tax_sum_phylum$Phylum <-row.names(ABBY_tax_sum_phylum)
ABBY_melt<- melt(ABBY_tax_sum_phylum, id.var="Phylum")
site<-rep("ABBY",times=nrow(ABBY_melt))
ABBY_melt<-cbind(ABBY_melt,site)

BARR_tax_sum_phylum = summarize_taxonomy(BARR, level = 4, report_higher_tax = FALSE)
BARR_tax_sum_phylum$Phylum <-row.names(BARR_tax_sum_phylum)
BARR_melt<- melt(BARR_tax_sum_phylum, id.var="Phylum")
nrow(BARR_melt)
site<-rep("BARR",times=nrow(BARR_melt))
BARR_melt<-cbind(BARR_melt,site)

BART_tax_sum_phylum = summarize_taxonomy(BART, level = 4, report_higher_tax = FALSE)
BART_tax_sum_phylum$Phylum <-row.names(BART_tax_sum_phylum)
BART_melt<- melt(BART_tax_sum_phylum, id.var="Phylum")
nrow(BART_melt)
site<-rep("BART",times=nrow(BART_melt))
BART_melt<-cbind(BART_melt,site)

BLAN_tax_sum_phylum = summarize_taxonomy(BLAN, level = 4, report_higher_tax = FALSE)
BLAN_tax_sum_phylum$Phylum <-row.names(BLAN_tax_sum_phylum)
BLAN_melt<- melt(BLAN_tax_sum_phylum, id.var="Phylum")
nrow(BLAN_melt)
site<-rep("BLAN",times=nrow(BLAN_melt))
BLAN_melt<-cbind(BLAN_melt,site)

BONA_tax_sum_phylum = summarize_taxonomy(BONA, level = 4, report_higher_tax = FALSE)
BONA_tax_sum_phylum$Phylum <-row.names(BONA_tax_sum_phylum)
BONA_melt<- melt(BONA_tax_sum_phylum, id.var="Phylum")
nrow(BONA_melt)
site<-rep("BONA",times=nrow(BONA_melt))
BONA_melt<-cbind(BONA_melt,site)

CLBJ_tax_sum_phylum = summarize_taxonomy(CLBJ, level = 4, report_higher_tax = FALSE)
CLBJ_tax_sum_phylum$Phylum <-row.names(CLBJ_tax_sum_phylum)
CLBJ_melt<- melt(CLBJ_tax_sum_phylum, id.var="Phylum")
nrow(CLBJ_melt)
site<-rep("CLBJ",times=nrow(CLBJ_melt))
CLBJ_melt<-cbind(CLBJ_melt,site)

CPER_tax_sum_phylum = summarize_taxonomy(CPER, level = 4, report_higher_tax = FALSE)
CPER_tax_sum_phylum$Phylum <-row.names(CPER_tax_sum_phylum)
CPER_melt<- melt(CPER_tax_sum_phylum, id.var="Phylum")
nrow(CPER_melt)
site<-rep("CPER",times=nrow(CPER_melt))
CPER_melt<-cbind(CPER_melt,site)

DCFS_tax_sum_phylum = summarize_taxonomy(DCFS, level = 4, report_higher_tax = FALSE)
DCFS_tax_sum_phylum$Phylum <-row.names(DCFS_tax_sum_phylum)
DCFS_melt<- melt(DCFS_tax_sum_phylum, id.var="Phylum")
nrow(DCFS_melt)
site<-rep("DCFS",times=nrow(DCFS_melt))
DCFS_melt<-cbind(DCFS_melt,site)

DEJU_tax_sum_phylum = summarize_taxonomy(DEJU, level = 4, report_higher_tax = FALSE)
DEJU_tax_sum_phylum$Phylum <-row.names(DEJU_tax_sum_phylum)
DEJU_melt<- melt(DEJU_tax_sum_phylum, id.var="Phylum")
nrow(DEJU_melt)
site<-rep("DEJU",times=nrow(DEJU_melt))
DEJU_melt<-cbind(DEJU_melt,site)

DELA_tax_sum_phylum = summarize_taxonomy(DELA, level = 4, report_higher_tax = FALSE)
DELA_tax_sum_phylum$Phylum <-row.names(DELA_tax_sum_phylum)
DELA_melt<- melt(DELA_tax_sum_phylum, id.var="Phylum")
nrow(DELA_melt)
site<-rep("DELA",times=nrow(DELA_melt))
DELA_melt<-cbind(DELA_melt,site)

DSNY_tax_sum_phylum = summarize_taxonomy(DSNY, level = 4, report_higher_tax = FALSE)
DSNY_tax_sum_phylum$Phylum <-row.names(DSNY_tax_sum_phylum)
DSNY_melt<- melt(DSNY_tax_sum_phylum, id.var="Phylum")
nrow(DSNY_melt)
site<-rep("DSNY",times=nrow(DSNY_melt))
DSNY_melt<-cbind(DSNY_melt,site)

GRSM_tax_sum_phylum = summarize_taxonomy(GRSM, level = 4, report_higher_tax = FALSE)
GRSM_tax_sum_phylum$Phylum <-row.names(GRSM_tax_sum_phylum)
GRSM_melt<- melt(GRSM_tax_sum_phylum, id.var="Phylum")
nrow(GRSM_melt)
site<-rep("GRSM",times=nrow(GRSM_melt))
GRSM_melt<-cbind(GRSM_melt,site)

GUAN_tax_sum_phylum = summarize_taxonomy(GUAN, level = 4, report_higher_tax = FALSE)
GUAN_tax_sum_phylum$Phylum <-row.names(GUAN_tax_sum_phylum)
GUAN_melt<- melt(GUAN_tax_sum_phylum, id.var="Phylum")
nrow(GUAN_melt)
site<-rep("GUAN",times=nrow(GUAN_melt))
GUAN_melt<-cbind(GUAN_melt,site)

HARV_tax_sum_phylum = summarize_taxonomy(HARV, level = 4, report_higher_tax = FALSE)
HARV_tax_sum_phylum$Phylum <-row.names(HARV_tax_sum_phylum)
HARV_melt<- melt(HARV_tax_sum_phylum, id.var="Phylum")
nrow(HARV_melt)
site<-rep("HARV",times=nrow(HARV_melt))
HARV_melt<-cbind(HARV_melt,site)

HEAL_tax_sum_phylum = summarize_taxonomy(HEAL, level = 4, report_higher_tax = FALSE)
HEAL_tax_sum_phylum$Phylum <-row.names(HEAL_tax_sum_phylum)
HEAL_melt<- melt(HEAL_tax_sum_phylum, id.var="Phylum")
nrow(HEAL_melt)
site<-rep("HEAL",times=nrow(HEAL_melt))
HEAL_melt<-cbind(HEAL_melt,site)

JERC_tax_sum_phylum = summarize_taxonomy(JERC, level = 4, report_higher_tax = FALSE)
JERC_tax_sum_phylum$Phylum <-row.names(JERC_tax_sum_phylum)
JERC_melt<- melt(JERC_tax_sum_phylum, id.var="Phylum")
nrow(JERC_melt)
site<-rep("JERC",times=nrow(JERC_melt))
JERC_melt<-cbind(JERC_melt,site)

JORN_tax_sum_phylum = summarize_taxonomy(JORN, level = 4, report_higher_tax = FALSE)
JORN_tax_sum_phylum$Phylum <-row.names(JORN_tax_sum_phylum)
JORN_melt<- melt(JORN_tax_sum_phylum, id.var="Phylum")
nrow(JORN_melt)
site<-rep("JORN",times=nrow(JORN_melt))
JORN_melt<-cbind(JORN_melt,site)

KONA_tax_sum_phylum = summarize_taxonomy(KONA, level = 4, report_higher_tax = FALSE)
KONA_tax_sum_phylum$Phylum <-row.names(KONA_tax_sum_phylum)
KONA_melt<- melt(KONA_tax_sum_phylum, id.var="Phylum")
nrow(KONA_melt)
site<-rep("KONA",times=nrow(KONA_melt))
KONA_melt<-cbind(KONA_melt,site)

KONZ_tax_sum_phylum = summarize_taxonomy(KONZ, level = 4, report_higher_tax = FALSE)
KONZ_tax_sum_phylum$Phylum <-row.names(KONZ_tax_sum_phylum)
KONZ_melt<- melt(KONZ_tax_sum_phylum, id.var="Phylum")
nrow(KONZ_melt)
site<-rep("KONZ",times=nrow(KONZ_melt))
KONZ_melt<-cbind(KONZ_melt,site)

LAJA_tax_sum_phylum = summarize_taxonomy(LAJA, level = 4, report_higher_tax = FALSE)
LAJA_tax_sum_phylum$Phylum <-row.names(LAJA_tax_sum_phylum)
LAJA_melt<- melt(LAJA_tax_sum_phylum, id.var="Phylum")
nrow(LAJA_melt)
site<-rep("LAJA",times=nrow(LAJA_melt))
LAJA_melt<-cbind(LAJA_melt,site)

LENO_tax_sum_phylum = summarize_taxonomy(LENO, level = 4, report_higher_tax = FALSE)
LENO_tax_sum_phylum$Phylum <-row.names(LENO_tax_sum_phylum)
LENO_melt<- melt(LENO_tax_sum_phylum, id.var="Phylum")
nrow(LENO_melt)
site<-rep("LENO",times=nrow(LENO_melt))
LENO_melt<-cbind(LENO_melt,site)

MLBS_tax_sum_phylum = summarize_taxonomy(MLBS, level = 4, report_higher_tax = FALSE)
MLBS_tax_sum_phylum$Phylum <-row.names(MLBS_tax_sum_phylum)
MLBS_melt<- melt(MLBS_tax_sum_phylum, id.var="Phylum")
nrow(MLBS_melt)
site<-rep("MLBS",times=nrow(MLBS_melt))
MLBS_melt<-cbind(MLBS_melt,site)

MOAB_tax_sum_phylum = summarize_taxonomy(MOAB, level = 4, report_higher_tax = FALSE)
MOAB_tax_sum_phylum$Phylum <-row.names(MOAB_tax_sum_phylum)
MOAB_melt<- melt(MOAB_tax_sum_phylum, id.var="Phylum")
nrow(MOAB_melt)
site<-rep("MOAB",times=nrow(MOAB_melt))
MOAB_melt<-cbind(MOAB_melt,site)

NIWO_tax_sum_phylum = summarize_taxonomy(NIWO, level = 4, report_higher_tax = FALSE)
NIWO_tax_sum_phylum$Phylum <-row.names(NIWO_tax_sum_phylum)
NIWO_melt<- melt(NIWO_tax_sum_phylum, id.var="Phylum")
nrow(NIWO_melt)
site<-rep("NIWO",times=nrow(NIWO_melt))
NIWO_melt<-cbind(NIWO_melt,site)

OAES_tax_sum_phylum = summarize_taxonomy(OAES, level = 4, report_higher_tax = FALSE)
OAES_tax_sum_phylum$Phylum <-row.names(OAES_tax_sum_phylum)
OAES_melt<- melt(OAES_tax_sum_phylum, id.var="Phylum")
nrow(OAES_melt)
site<-rep("OAES",times=nrow(OAES_melt))
OAES_melt<-cbind(OAES_melt,site)

ONAQ_tax_sum_phylum = summarize_taxonomy(ONAQ, level = 4, report_higher_tax = FALSE)
ONAQ_tax_sum_phylum$Phylum <-row.names(ONAQ_tax_sum_phylum)
ONAQ_melt<- melt(ONAQ_tax_sum_phylum, id.var="Phylum")
nrow(ONAQ_melt)
site<-rep("ONAQ",times=nrow(ONAQ_melt))
ONAQ_melt<-cbind(ONAQ_melt,site)

ORNL_tax_sum_phylum = summarize_taxonomy(ORNL, level = 4, report_higher_tax = FALSE)
ORNL_tax_sum_phylum$Phylum <-row.names(ORNL_tax_sum_phylum)
ORNL_melt<- melt(ORNL_tax_sum_phylum, id.var="Phylum")
nrow(ORNL_melt)
site<-rep("ORNL",times=nrow(ORNL_melt))
ORNL_melt<-cbind(ORNL_melt,site)

OSBS_tax_sum_phylum = summarize_taxonomy(OSBS, level = 4, report_higher_tax = FALSE)
OSBS_tax_sum_phylum$Phylum <-row.names(OSBS_tax_sum_phylum)
OSBS_melt<- melt(OSBS_tax_sum_phylum, id.var="Phylum")
nrow(OSBS_melt)
site<-rep("OSBS",times=nrow(OSBS_melt))
OSBS_melt<-cbind(OSBS_melt,site)

PUUM_tax_sum_phylum = summarize_taxonomy(PUUM, level = 4, report_higher_tax = FALSE)
PUUM_tax_sum_phylum$Phylum <-row.names(PUUM_tax_sum_phylum)
PUUM_melt<- melt(PUUM_tax_sum_phylum, id.var="Phylum")
nrow(PUUM_melt)
site<-rep("PUUM",times=nrow(PUUM_melt))
PUUM_melt<-cbind(PUUM_melt,site)

RMNP_tax_sum_phylum = summarize_taxonomy(RMNP, level = 4, report_higher_tax = FALSE)
RMNP_tax_sum_phylum$Phylum <-row.names(RMNP_tax_sum_phylum)
RMNP_melt<- melt(RMNP_tax_sum_phylum, id.var="Phylum")
nrow(RMNP_melt)
site<-rep("RMNP",times=nrow(RMNP_melt))
RMNP_melt<-cbind(RMNP_melt,site)

SCBI_tax_sum_phylum = summarize_taxonomy(SCBI, level = 4, report_higher_tax = FALSE)
SCBI_tax_sum_phylum$Phylum <-row.names(SCBI_tax_sum_phylum)
SCBI_melt<- melt(SCBI_tax_sum_phylum, id.var="Phylum")
nrow(SCBI_melt)
site<-rep("SCBI",times=nrow(SCBI_melt))
SCBI_melt<-cbind(SCBI_melt,site)

SERC_tax_sum_phylum = summarize_taxonomy(SERC, level = 4, report_higher_tax = FALSE)
SERC_tax_sum_phylum$Phylum <-row.names(SERC_tax_sum_phylum)
SERC_melt<- melt(SERC_tax_sum_phylum, id.var="Phylum")
nrow(SERC_melt)
site<-rep("SERC",times=nrow(SERC_melt))
SERC_melt<-cbind(SERC_melt,site)

SJER_tax_sum_phylum = summarize_taxonomy(SJER, level = 4, report_higher_tax = FALSE)
SJER_tax_sum_phylum$Phylum <-row.names(SJER_tax_sum_phylum)
SJER_melt<- melt(SJER_tax_sum_phylum, id.var="Phylum")
nrow(SJER_melt)
site<-rep("SJER",times=nrow(SJER_melt))
SJER_melt<-cbind(SJER_melt,site)

SOAP_tax_sum_phylum = summarize_taxonomy(SOAP, level = 4, report_higher_tax = FALSE)
SOAP_tax_sum_phylum$Phylum <-row.names(SOAP_tax_sum_phylum)
SOAP_melt<- melt(SOAP_tax_sum_phylum, id.var="Phylum")
nrow(SOAP_melt)
site<-rep("SOAP",times=nrow(SOAP_melt))
SOAP_melt<-cbind(SOAP_melt,site)

SRER_tax_sum_phylum = summarize_taxonomy(SRER, level = 4, report_higher_tax = FALSE)
SRER_tax_sum_phylum$Phylum <-row.names(SRER_tax_sum_phylum)
SRER_melt<- melt(SRER_tax_sum_phylum, id.var="Phylum")
nrow(SRER_melt)
site<-rep("SRER",times=nrow(SRER_melt))
SRER_melt<-cbind(SRER_melt,site)

STEI_tax_sum_phylum = summarize_taxonomy(STEI, level = 4, report_higher_tax = FALSE)
STEI_tax_sum_phylum$Phylum <-row.names(STEI_tax_sum_phylum)
STEI_melt<- melt(STEI_tax_sum_phylum, id.var="Phylum")
nrow(STEI_melt)
site<-rep("STEI",times=nrow(STEI_melt))
STEI_melt<-cbind(STEI_melt,site)

STER_tax_sum_phylum = summarize_taxonomy(STER, level = 4, report_higher_tax = FALSE)
STER_tax_sum_phylum$Phylum <-row.names(STER_tax_sum_phylum)
STER_melt<- melt(STER_tax_sum_phylum, id.var="Phylum")
nrow(STER_melt)
site<-rep("STER",times=nrow(STER_melt))
STER_melt<-cbind(STER_melt,site)

TALL_tax_sum_phylum = summarize_taxonomy(TALL, level = 4, report_higher_tax = FALSE)
TALL_tax_sum_phylum$Phylum <-row.names(TALL_tax_sum_phylum)
TALL_melt<- melt(TALL_tax_sum_phylum, id.var="Phylum")
nrow(TALL_melt)
site<-rep("TALL",times=nrow(TALL_melt))
TALL_melt<-cbind(TALL_melt,site)

TOOL_tax_sum_phylum = summarize_taxonomy(TOOL, level = 4, report_higher_tax = FALSE)
TOOL_tax_sum_phylum$Phylum <-row.names(TOOL_tax_sum_phylum)
TOOL_melt<- melt(TOOL_tax_sum_phylum, id.var="Phylum")
nrow(TOOL_melt)
site<-rep("TOOL",times=nrow(TOOL_melt))
TOOL_melt<-cbind(TOOL_melt,site)

TREE_tax_sum_phylum = summarize_taxonomy(TREE, level = 4, report_higher_tax = FALSE)
TREE_tax_sum_phylum$Phylum <-row.names(TREE_tax_sum_phylum)
TREE_melt<- melt(TREE_tax_sum_phylum, id.var="Phylum")
nrow(TREE_melt)
site<-rep("TREE",times=nrow(TREE_melt))
TREE_melt<-cbind(TREE_melt,site)

UKFS_tax_sum_phylum = summarize_taxonomy(UKFS, level = 4, report_higher_tax = FALSE)
UKFS_tax_sum_phylum$Phylum <-row.names(UKFS_tax_sum_phylum)
UKFS_melt<- melt(UKFS_tax_sum_phylum, id.var="Phylum")
nrow(UKFS_melt)
site<-rep("UKFS",times=nrow(UKFS_melt))
UKFS_melt<-cbind(UKFS_melt,site)

UNDE_tax_sum_phylum = summarize_taxonomy(UNDE, level = 4, report_higher_tax = FALSE)
UNDE_tax_sum_phylum$Phylum <-row.names(UNDE_tax_sum_phylum)
UNDE_melt<- melt(UNDE_tax_sum_phylum, id.var="Phylum")
nrow(UNDE_melt)
site<-rep("UNDE",times=nrow(UNDE_melt))
UNDE_melt<-cbind(UNDE_melt,site)

WOOD_tax_sum_phylum = summarize_taxonomy(WOOD, level = 4, report_higher_tax = FALSE)
WOOD_tax_sum_phylum$Phylum <-row.names(WOOD_tax_sum_phylum)
WOOD_melt<- melt(WOOD_tax_sum_phylum, id.var="Phylum")
nrow(WOOD_melt)
site<-rep("WOOD",times=nrow(WOOD_melt))
WOOD_melt<-cbind(WOOD_melt,site)

WREF_tax_sum_phylum = summarize_taxonomy(WREF, level = 4, report_higher_tax = FALSE)
WREF_tax_sum_phylum$Phylum <-row.names(WREF_tax_sum_phylum)
WREF_melt<- melt(WREF_tax_sum_phylum, id.var="Phylum")
nrow(WREF_melt)
site<-rep("WREF",times=nrow(WREF_melt))
WREF_melt<-cbind(WREF_melt,site)

YELL_tax_sum_phylum = summarize_taxonomy(YELL, level = 4, report_higher_tax = FALSE)
YELL_tax_sum_phylum$Phylum <-row.names(YELL_tax_sum_phylum)
YELL_melt<- melt(YELL_tax_sum_phylum, id.var="Phylum")
nrow(YELL_melt)
site<-rep("YELL",times=nrow(YELL_melt))
YELL_melt<-cbind(YELL_melt,site)
```

```{r}
phylumfinalbind<-do.call("rbind",list(ABBY_melt, BARR_melt, BART_melt, BLAN_melt, BONA_melt, CLBJ_melt, CPER_melt, DCFS_melt, DEJU_melt, DELA_melt, DSNY_melt, GRSM_melt, GUAN_melt, HARV_melt, HEAL_melt, JERC_melt, JORN_melt, KONA_melt, KONZ_melt, LAJA_melt, LENO_melt, MLBS_melt, MOAB_melt, NIWO_melt, OAES_melt, ONAQ_melt, ORNL_melt, OSBS_melt, PUUM_melt, RMNP_melt, SCBI_melt, SERC_melt, SJER_melt, SOAP_melt, SRER_melt, STEI_melt, STER_melt, TALL_melt, TOOL_melt, TREE_melt, UKFS_melt, UNDE_melt, WOOD_melt, WREF_melt, YELL_melt))
```

# Latitude first by phylum
phylum graph
```{r}
positions <- c("GUAN","LAJA","PUUM","DSNY","OSBS","JERC","LENO","SRER","DELA","JORN","TALL","CLBJ","OAES","GRSM","ORNL","SOAP","SJER","MLBS","MOAB","SERC","SCBI","BLAN","UKFS","KONZ","KONA","NIWO","ONAQ","RMNP","STER","CPER","HARV","BART","YELL","TREE","STEI","ABBY","WREF","UNDE","WOOD","DCFS","HEAL","DEJU","BONA","TOOL","BARR")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_x_discrete(limits = positions)+
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Phylum Graphs by Biome")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "metazoa_site_by_latitude.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

# MAP first by phylum
phylum graph
```{r}
positions <- c("BARR", "BONA", "JORN","ONAQ","DEJU","TOOL","MOAB","CPER","SRER","HEAL","STER","DCFS","YELL","WOOD","SJER","RMNP","OAES","STEI","TREE","UNDE","LAJA","GUAN","KONA","KONZ","SOAP","CLBJ","BLAN","UKFS","NIWO","SERC","SCBI","HARV","DSNY","MLBS","OSBS","JERC","BART","ORNL","DELA","GRSM","TALL","LENO","WREF","ABBY","PUUM")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_x_discrete(limits = positions)+
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Metazo Graphs by MAP")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "metazoa_site_by_MAP.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```


# MAT first by phylum
phylum graph
```{r}
positions <- c("HEAL","BARR","BONA","DEJU","TOOL","NIWO","ABBY","MOAB","SCBI","BLAN","KONZ","KONA","UKFS","PUUM","GRSM","SOAP","SERC","ORNL","OAES","JORN","SJER","TALL","CLBJ","DELA","LENO","JERC","SRER","RMNP","OSBS","DSNY","GUAN","LAJA","YELL","UNDE","STEI","TREE","DCFS","WOOD","BART","HARV","CPER","MLBS","ONAQ","WREF","STER")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_x_discrete(limits = positions)+
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Metazoa Graphs by MAT")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "metazoa_site_by_MAT.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
phylumfinalbind<-do.call("rbind",list(CPER_melt,STER_melt,KONZ_melt,KONA_melt,WOOD_melt,DCFS_melt,GUAN_melt,LAJA_melt,WREF_melt,ABBY_melt,TREE_melt,STEI_melt))
```

```{r}
positions <- c("CPER","STER","KONZ","KONA","WOOD","DCFS","GUAN","LAJA","WREF","ABBY","TREE","STEI")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_x_discrete(limits = positions)+
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Metazoa by management")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "metazoa_site_by_management.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```


Sorted by MAP, MAT, and latitude. After this do with management.

Filter by site
```{r}
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
MLBS = filter_data(input, 'Site', keep_vals = 'MLBS')
HARV = filter_data(input, 'Site', keep_vals = 'HARV')
WOOD = filter_data(input, 'Site', keep_vals = 'WOOD')
LAJA = filter_data(input, 'Site', keep_vals = 'LAJA')
UNDE = filter_data(input, 'Site', keep_vals = 'UNDE')
SRER = filter_data(input, 'Site', keep_vals = 'SRER')
DSNY = filter_data(input, 'Site', keep_vals = 'DSNY')
BONA = filter_data(input, 'Site', keep_vals = 'BONA')
GUAN = filter_data(input, 'Site', keep_vals = 'GUAN')
ABBY = filter_data(input, 'Site', keep_vals = 'ABBY')
BARR = filter_data(input, 'Site', keep_vals = 'BARR')
BART = filter_data(input, 'Site', keep_vals = 'BART')
BLAN = filter_data(input, 'Site', keep_vals = 'BLAN')
CLBJ = filter_data(input, 'Site', keep_vals = 'CLBJ')
CPER = filter_data(input, 'Site', keep_vals = 'CPER')
DCFS = filter_data(input, 'Site', keep_vals = 'DCFS')
DEJU = filter_data(input, 'Site', keep_vals = 'DEJU')
DELA = filter_data(input, 'Site', keep_vals = 'DELA')
GRSM = filter_data(input, 'Site', keep_vals = 'GRSM')
HEAL = filter_data(input, 'Site', keep_vals = 'HEAL')
JERC = filter_data(input, 'Site', keep_vals = 'JERC')
JORN = filter_data(input, 'Site', keep_vals = 'JORN')
KONA = filter_data(input, 'Site', keep_vals = 'KONA')
KONZ = filter_data(input, 'Site', keep_vals = 'KONZ')
LENO = filter_data(input, 'Site', keep_vals = 'LENO')
MOAB = filter_data(input, 'Site', keep_vals = 'MOAB')
NIWO = filter_data(input, 'Site', keep_vals = 'NIWO')
OAES = filter_data(input, 'Site', keep_vals = 'OAES')
ONAQ = filter_data(input, 'Site', keep_vals = 'ONAQ')
ORNL = filter_data(input, 'Site', keep_vals = 'ORNL')
OSBS = filter_data(input, 'Site', keep_vals = 'OSBS')
PUUM = filter_data(input, 'Site', keep_vals = 'PUUM')
RMNP = filter_data(input, 'Site', keep_vals = 'RMNP')
SCBI = filter_data(input, 'Site', keep_vals = 'SCBI')
SERC = filter_data(input, 'Site', keep_vals = 'SERC')
SJER = filter_data(input, 'Site', keep_vals = 'SJER')
SOAP = filter_data(input, 'Site', keep_vals = 'SOAP')
STEI = filter_data(input, 'Site', keep_vals = 'STEI')
STER = filter_data(input, 'Site', keep_vals = 'STER')
TALL = filter_data(input, 'Site', keep_vals = 'TALL')
TOOL = filter_data(input, 'Site', keep_vals = 'TOOL')
TREE = filter_data(input, 'Site', keep_vals = 'TREE')
UKFS = filter_data(input, 'Site', keep_vals = 'UKFS')
WREF = filter_data(input, 'Site', keep_vals = 'WREF')
YELL = filter_data(input, 'Site', keep_vals = 'YELL')
```

```{r}
library(labdsv)
otuTAB<-ABBY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ABBY$data_loaded<-t(hell)

otuTAB<-BARR$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BARR$data_loaded<-t(hell)

otuTAB<-BART$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BART$data_loaded<-t(hell)

otuTAB<-BLAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BLAN$data_loaded<-t(hell)

otuTAB<-BONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
BONA$data_loaded<-t(hell)

otuTAB<-CLBJ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CLBJ$data_loaded<-t(hell)

otuTAB<-CPER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
CPER$data_loaded<-t(hell)

otuTAB<-DCFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DCFS$data_loaded<-t(hell)

otuTAB<-DELA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DELA$data_loaded<-t(hell)

otuTAB<-DSNY$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
DSNY$data_loaded<-t(hell)

otuTAB<-GRSM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GRSM$data_loaded<-t(hell)

otuTAB<-GUAN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
GUAN$data_loaded<-t(hell)

otuTAB<-HARV$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HARV$data_loaded<-t(hell)

otuTAB<-HEAL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
HEAL$data_loaded<-t(hell)

otuTAB<-JERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JERC$data_loaded<-t(hell)

otuTAB<-JORN$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
JORN$data_loaded<-t(hell)

otuTAB<-KONA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONA$data_loaded<-t(hell)

otuTAB<-KONZ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
KONZ$data_loaded<-t(hell)

otuTAB<-LAJA$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LAJA$data_loaded<-t(hell)

otuTAB<-LENO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
LENO$data_loaded<-t(hell)

otuTAB<-MLBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MLBS$data_loaded<-t(hell)

otuTAB<-MOAB$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
MOAB$data_loaded<-t(hell)

otuTAB<-NIWO$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
NIWO$data_loaded<-t(hell)

otuTAB<-OAES$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OAES$data_loaded<-t(hell)

otuTAB<-ONAQ$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ONAQ$data_loaded<-t(hell)

otuTAB<-ORNL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
ORNL$data_loaded<-t(hell)

otuTAB<-OSBS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
OSBS$data_loaded<-t(hell)

otuTAB<-PUUM$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
PUUM$data_loaded<-t(hell)

otuTAB<-RMNP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
RMNP$data_loaded<-t(hell)

otuTAB<-SCBI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SCBI$data_loaded<-t(hell)

otuTAB<-SERC$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SERC$data_loaded<-t(hell)

otuTAB<-SJER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SJER$data_loaded<-t(hell)

otuTAB<-SOAP$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SOAP$data_loaded<-t(hell)

otuTAB<-SRER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
SRER$data_loaded<-t(hell)

otuTAB<-STEI$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STEI$data_loaded<-t(hell)

otuTAB<-STER$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
STER$data_loaded<-t(hell)

otuTAB<-TALL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TALL$data_loaded<-t(hell)

otuTAB<-TOOL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TOOL$data_loaded<-t(hell)

otuTAB<-TREE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
TREE$data_loaded<-t(hell)

otuTAB<-UKFS$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UKFS$data_loaded<-t(hell)

otuTAB<-UNDE$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
UNDE$data_loaded<-t(hell)

otuTAB<-WOOD$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WOOD$data_loaded<-t(hell)

otuTAB<-WREF$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
WREF$data_loaded<-t(hell)

otuTAB<-YELL$data_loaded
otuTAB <- as.data.frame (otuTAB)
otuTABtranspose<-t(otuTAB)
hell<-hellinger(otuTABtranspose)
YELL$data_loaded<-t(hell)
```


```{r}
library(reshape2)
ABBY_tax_sum_phylum = summarize_taxonomy(ABBY, level = 3, report_higher_tax = FALSE)
ABBY_tax_sum_phylum$Phylum <-row.names(ABBY_tax_sum_phylum)
ABBY_melt<- melt(ABBY_tax_sum_phylum, id.var="Phylum")
site<-rep("ABBY",times=nrow(ABBY_melt))
ABBY_melt<-cbind(ABBY_melt,site)

BARR_tax_sum_phylum = summarize_taxonomy(BARR, level = 3, report_higher_tax = FALSE)
BARR_tax_sum_phylum$Phylum <-row.names(BARR_tax_sum_phylum)
BARR_melt<- melt(BARR_tax_sum_phylum, id.var="Phylum")
nrow(BARR_melt)
site<-rep("BARR",times=nrow(BARR_melt))
BARR_melt<-cbind(BARR_melt,site)

BART_tax_sum_phylum = summarize_taxonomy(BART, level = 3, report_higher_tax = FALSE)
BART_tax_sum_phylum$Phylum <-row.names(BART_tax_sum_phylum)
BART_melt<- melt(BART_tax_sum_phylum, id.var="Phylum")
nrow(BART_melt)
site<-rep("BART",times=nrow(BART_melt))
BART_melt<-cbind(BART_melt,site)

BLAN_tax_sum_phylum = summarize_taxonomy(BLAN, level = 3, report_higher_tax = FALSE)
BLAN_tax_sum_phylum$Phylum <-row.names(BLAN_tax_sum_phylum)
BLAN_melt<- melt(BLAN_tax_sum_phylum, id.var="Phylum")
nrow(BLAN_melt)
site<-rep("BLAN",times=nrow(BLAN_melt))
BLAN_melt<-cbind(BLAN_melt,site)

BONA_tax_sum_phylum = summarize_taxonomy(BONA, level = 3, report_higher_tax = FALSE)
BONA_tax_sum_phylum$Phylum <-row.names(BONA_tax_sum_phylum)
BONA_melt<- melt(BONA_tax_sum_phylum, id.var="Phylum")
nrow(BONA_melt)
site<-rep("BONA",times=nrow(BONA_melt))
BONA_melt<-cbind(BONA_melt,site)

CLBJ_tax_sum_phylum = summarize_taxonomy(CLBJ, level = 3, report_higher_tax = FALSE)
CLBJ_tax_sum_phylum$Phylum <-row.names(CLBJ_tax_sum_phylum)
CLBJ_melt<- melt(CLBJ_tax_sum_phylum, id.var="Phylum")
nrow(CLBJ_melt)
site<-rep("CLBJ",times=nrow(CLBJ_melt))
CLBJ_melt<-cbind(CLBJ_melt,site)

CPER_tax_sum_phylum = summarize_taxonomy(CPER, level = 3, report_higher_tax = FALSE)
CPER_tax_sum_phylum$Phylum <-row.names(CPER_tax_sum_phylum)
CPER_melt<- melt(CPER_tax_sum_phylum, id.var="Phylum")
nrow(CPER_melt)
site<-rep("CPER",times=nrow(CPER_melt))
CPER_melt<-cbind(CPER_melt,site)

DCFS_tax_sum_phylum = summarize_taxonomy(DCFS, level = 3, report_higher_tax = FALSE)
DCFS_tax_sum_phylum$Phylum <-row.names(DCFS_tax_sum_phylum)
DCFS_melt<- melt(DCFS_tax_sum_phylum, id.var="Phylum")
nrow(DCFS_melt)
site<-rep("DCFS",times=nrow(DCFS_melt))
DCFS_melt<-cbind(DCFS_melt,site)

DEJU_tax_sum_phylum = summarize_taxonomy(DEJU, level = 3, report_higher_tax = FALSE)
DEJU_tax_sum_phylum$Phylum <-row.names(DEJU_tax_sum_phylum)
DEJU_melt<- melt(DEJU_tax_sum_phylum, id.var="Phylum")
nrow(DEJU_melt)
site<-rep("DEJU",times=nrow(DEJU_melt))
DEJU_melt<-cbind(DEJU_melt,site)

DELA_tax_sum_phylum = summarize_taxonomy(DELA, level = 3, report_higher_tax = FALSE)
DELA_tax_sum_phylum$Phylum <-row.names(DELA_tax_sum_phylum)
DELA_melt<- melt(DELA_tax_sum_phylum, id.var="Phylum")
nrow(DELA_melt)
site<-rep("DELA",times=nrow(DELA_melt))
DELA_melt<-cbind(DELA_melt,site)

DSNY_tax_sum_phylum = summarize_taxonomy(DSNY, level = 3, report_higher_tax = FALSE)
DSNY_tax_sum_phylum$Phylum <-row.names(DSNY_tax_sum_phylum)
DSNY_melt<- melt(DSNY_tax_sum_phylum, id.var="Phylum")
nrow(DSNY_melt)
site<-rep("DSNY",times=nrow(DSNY_melt))
DSNY_melt<-cbind(DSNY_melt,site)

GRSM_tax_sum_phylum = summarize_taxonomy(GRSM, level = 3, report_higher_tax = FALSE)
GRSM_tax_sum_phylum$Phylum <-row.names(GRSM_tax_sum_phylum)
GRSM_melt<- melt(GRSM_tax_sum_phylum, id.var="Phylum")
nrow(GRSM_melt)
site<-rep("GRSM",times=nrow(GRSM_melt))
GRSM_melt<-cbind(GRSM_melt,site)

GUAN_tax_sum_phylum = summarize_taxonomy(GUAN, level = 3, report_higher_tax = FALSE)
GUAN_tax_sum_phylum$Phylum <-row.names(GUAN_tax_sum_phylum)
GUAN_melt<- melt(GUAN_tax_sum_phylum, id.var="Phylum")
nrow(GUAN_melt)
site<-rep("GUAN",times=nrow(GUAN_melt))
GUAN_melt<-cbind(GUAN_melt,site)

HARV_tax_sum_phylum = summarize_taxonomy(HARV, level = 3, report_higher_tax = FALSE)
HARV_tax_sum_phylum$Phylum <-row.names(HARV_tax_sum_phylum)
HARV_melt<- melt(HARV_tax_sum_phylum, id.var="Phylum")
nrow(HARV_melt)
site<-rep("HARV",times=nrow(HARV_melt))
HARV_melt<-cbind(HARV_melt,site)

HEAL_tax_sum_phylum = summarize_taxonomy(HEAL, level = 3, report_higher_tax = FALSE)
HEAL_tax_sum_phylum$Phylum <-row.names(HEAL_tax_sum_phylum)
HEAL_melt<- melt(HEAL_tax_sum_phylum, id.var="Phylum")
nrow(HEAL_melt)
site<-rep("HEAL",times=nrow(HEAL_melt))
HEAL_melt<-cbind(HEAL_melt,site)

JERC_tax_sum_phylum = summarize_taxonomy(JERC, level = 3, report_higher_tax = FALSE)
JERC_tax_sum_phylum$Phylum <-row.names(JERC_tax_sum_phylum)
JERC_melt<- melt(JERC_tax_sum_phylum, id.var="Phylum")
nrow(JERC_melt)
site<-rep("JERC",times=nrow(JERC_melt))
JERC_melt<-cbind(JERC_melt,site)

JORN_tax_sum_phylum = summarize_taxonomy(JORN, level = 3, report_higher_tax = FALSE)
JORN_tax_sum_phylum$Phylum <-row.names(JORN_tax_sum_phylum)
JORN_melt<- melt(JORN_tax_sum_phylum, id.var="Phylum")
nrow(JORN_melt)
site<-rep("JORN",times=nrow(JORN_melt))
JORN_melt<-cbind(JORN_melt,site)

KONA_tax_sum_phylum = summarize_taxonomy(KONA, level = 3, report_higher_tax = FALSE)
KONA_tax_sum_phylum$Phylum <-row.names(KONA_tax_sum_phylum)
KONA_melt<- melt(KONA_tax_sum_phylum, id.var="Phylum")
nrow(KONA_melt)
site<-rep("KONA",times=nrow(KONA_melt))
KONA_melt<-cbind(KONA_melt,site)

KONZ_tax_sum_phylum = summarize_taxonomy(KONZ, level = 3, report_higher_tax = FALSE)
KONZ_tax_sum_phylum$Phylum <-row.names(KONZ_tax_sum_phylum)
KONZ_melt<- melt(KONZ_tax_sum_phylum, id.var="Phylum")
nrow(KONZ_melt)
site<-rep("KONZ",times=nrow(KONZ_melt))
KONZ_melt<-cbind(KONZ_melt,site)

LAJA_tax_sum_phylum = summarize_taxonomy(LAJA, level = 3, report_higher_tax = FALSE)
LAJA_tax_sum_phylum$Phylum <-row.names(LAJA_tax_sum_phylum)
LAJA_melt<- melt(LAJA_tax_sum_phylum, id.var="Phylum")
nrow(LAJA_melt)
site<-rep("LAJA",times=nrow(LAJA_melt))
LAJA_melt<-cbind(LAJA_melt,site)

LENO_tax_sum_phylum = summarize_taxonomy(LENO, level = 3, report_higher_tax = FALSE)
LENO_tax_sum_phylum$Phylum <-row.names(LENO_tax_sum_phylum)
LENO_melt<- melt(LENO_tax_sum_phylum, id.var="Phylum")
nrow(LENO_melt)
site<-rep("LENO",times=nrow(LENO_melt))
LENO_melt<-cbind(LENO_melt,site)

MLBS_tax_sum_phylum = summarize_taxonomy(MLBS, level = 3, report_higher_tax = FALSE)
MLBS_tax_sum_phylum$Phylum <-row.names(MLBS_tax_sum_phylum)
MLBS_melt<- melt(MLBS_tax_sum_phylum, id.var="Phylum")
nrow(MLBS_melt)
site<-rep("MLBS",times=nrow(MLBS_melt))
MLBS_melt<-cbind(MLBS_melt,site)

MOAB_tax_sum_phylum = summarize_taxonomy(MOAB, level = 3, report_higher_tax = FALSE)
MOAB_tax_sum_phylum$Phylum <-row.names(MOAB_tax_sum_phylum)
MOAB_melt<- melt(MOAB_tax_sum_phylum, id.var="Phylum")
nrow(MOAB_melt)
site<-rep("MOAB",times=nrow(MOAB_melt))
MOAB_melt<-cbind(MOAB_melt,site)

NIWO_tax_sum_phylum = summarize_taxonomy(NIWO, level = 3, report_higher_tax = FALSE)
NIWO_tax_sum_phylum$Phylum <-row.names(NIWO_tax_sum_phylum)
NIWO_melt<- melt(NIWO_tax_sum_phylum, id.var="Phylum")
nrow(NIWO_melt)
site<-rep("NIWO",times=nrow(NIWO_melt))
NIWO_melt<-cbind(NIWO_melt,site)

OAES_tax_sum_phylum = summarize_taxonomy(OAES, level = 3, report_higher_tax = FALSE)
OAES_tax_sum_phylum$Phylum <-row.names(OAES_tax_sum_phylum)
OAES_melt<- melt(OAES_tax_sum_phylum, id.var="Phylum")
nrow(OAES_melt)
site<-rep("OAES",times=nrow(OAES_melt))
OAES_melt<-cbind(OAES_melt,site)

ONAQ_tax_sum_phylum = summarize_taxonomy(ONAQ, level = 3, report_higher_tax = FALSE)
ONAQ_tax_sum_phylum$Phylum <-row.names(ONAQ_tax_sum_phylum)
ONAQ_melt<- melt(ONAQ_tax_sum_phylum, id.var="Phylum")
nrow(ONAQ_melt)
site<-rep("ONAQ",times=nrow(ONAQ_melt))
ONAQ_melt<-cbind(ONAQ_melt,site)

ORNL_tax_sum_phylum = summarize_taxonomy(ORNL, level = 3, report_higher_tax = FALSE)
ORNL_tax_sum_phylum$Phylum <-row.names(ORNL_tax_sum_phylum)
ORNL_melt<- melt(ORNL_tax_sum_phylum, id.var="Phylum")
nrow(ORNL_melt)
site<-rep("ORNL",times=nrow(ORNL_melt))
ORNL_melt<-cbind(ORNL_melt,site)

OSBS_tax_sum_phylum = summarize_taxonomy(OSBS, level = 3, report_higher_tax = FALSE)
OSBS_tax_sum_phylum$Phylum <-row.names(OSBS_tax_sum_phylum)
OSBS_melt<- melt(OSBS_tax_sum_phylum, id.var="Phylum")
nrow(OSBS_melt)
site<-rep("OSBS",times=nrow(OSBS_melt))
OSBS_melt<-cbind(OSBS_melt,site)

PUUM_tax_sum_phylum = summarize_taxonomy(PUUM, level = 3, report_higher_tax = FALSE)
PUUM_tax_sum_phylum$Phylum <-row.names(PUUM_tax_sum_phylum)
PUUM_melt<- melt(PUUM_tax_sum_phylum, id.var="Phylum")
nrow(PUUM_melt)
site<-rep("PUUM",times=nrow(PUUM_melt))
PUUM_melt<-cbind(PUUM_melt,site)

RMNP_tax_sum_phylum = summarize_taxonomy(RMNP, level = 3, report_higher_tax = FALSE)
RMNP_tax_sum_phylum$Phylum <-row.names(RMNP_tax_sum_phylum)
RMNP_melt<- melt(RMNP_tax_sum_phylum, id.var="Phylum")
nrow(RMNP_melt)
site<-rep("RMNP",times=nrow(RMNP_melt))
RMNP_melt<-cbind(RMNP_melt,site)

SCBI_tax_sum_phylum = summarize_taxonomy(SCBI, level = 3, report_higher_tax = FALSE)
SCBI_tax_sum_phylum$Phylum <-row.names(SCBI_tax_sum_phylum)
SCBI_melt<- melt(SCBI_tax_sum_phylum, id.var="Phylum")
nrow(SCBI_melt)
site<-rep("SCBI",times=nrow(SCBI_melt))
SCBI_melt<-cbind(SCBI_melt,site)

SERC_tax_sum_phylum = summarize_taxonomy(SERC, level = 3, report_higher_tax = FALSE)
SERC_tax_sum_phylum$Phylum <-row.names(SERC_tax_sum_phylum)
SERC_melt<- melt(SERC_tax_sum_phylum, id.var="Phylum")
nrow(SERC_melt)
site<-rep("SERC",times=nrow(SERC_melt))
SERC_melt<-cbind(SERC_melt,site)

SJER_tax_sum_phylum = summarize_taxonomy(SJER, level = 3, report_higher_tax = FALSE)
SJER_tax_sum_phylum$Phylum <-row.names(SJER_tax_sum_phylum)
SJER_melt<- melt(SJER_tax_sum_phylum, id.var="Phylum")
nrow(SJER_melt)
site<-rep("SJER",times=nrow(SJER_melt))
SJER_melt<-cbind(SJER_melt,site)

SOAP_tax_sum_phylum = summarize_taxonomy(SOAP, level = 3, report_higher_tax = FALSE)
SOAP_tax_sum_phylum$Phylum <-row.names(SOAP_tax_sum_phylum)
SOAP_melt<- melt(SOAP_tax_sum_phylum, id.var="Phylum")
nrow(SOAP_melt)
site<-rep("SOAP",times=nrow(SOAP_melt))
SOAP_melt<-cbind(SOAP_melt,site)

SRER_tax_sum_phylum = summarize_taxonomy(SRER, level = 3, report_higher_tax = FALSE)
SRER_tax_sum_phylum$Phylum <-row.names(SRER_tax_sum_phylum)
SRER_melt<- melt(SRER_tax_sum_phylum, id.var="Phylum")
nrow(SRER_melt)
site<-rep("SRER",times=nrow(SRER_melt))
SRER_melt<-cbind(SRER_melt,site)

STEI_tax_sum_phylum = summarize_taxonomy(STEI, level = 3, report_higher_tax = FALSE)
STEI_tax_sum_phylum$Phylum <-row.names(STEI_tax_sum_phylum)
STEI_melt<- melt(STEI_tax_sum_phylum, id.var="Phylum")
nrow(STEI_melt)
site<-rep("STEI",times=nrow(STEI_melt))
STEI_melt<-cbind(STEI_melt,site)

STER_tax_sum_phylum = summarize_taxonomy(STER, level = 3, report_higher_tax = FALSE)
STER_tax_sum_phylum$Phylum <-row.names(STER_tax_sum_phylum)
STER_melt<- melt(STER_tax_sum_phylum, id.var="Phylum")
nrow(STER_melt)
site<-rep("STER",times=nrow(STER_melt))
STER_melt<-cbind(STER_melt,site)

TALL_tax_sum_phylum = summarize_taxonomy(TALL, level = 3, report_higher_tax = FALSE)
TALL_tax_sum_phylum$Phylum <-row.names(TALL_tax_sum_phylum)
TALL_melt<- melt(TALL_tax_sum_phylum, id.var="Phylum")
nrow(TALL_melt)
site<-rep("TALL",times=nrow(TALL_melt))
TALL_melt<-cbind(TALL_melt,site)

TOOL_tax_sum_phylum = summarize_taxonomy(TOOL, level = 3, report_higher_tax = FALSE)
TOOL_tax_sum_phylum$Phylum <-row.names(TOOL_tax_sum_phylum)
TOOL_melt<- melt(TOOL_tax_sum_phylum, id.var="Phylum")
nrow(TOOL_melt)
site<-rep("TOOL",times=nrow(TOOL_melt))
TOOL_melt<-cbind(TOOL_melt,site)

TREE_tax_sum_phylum = summarize_taxonomy(TREE, level = 3, report_higher_tax = FALSE)
TREE_tax_sum_phylum$Phylum <-row.names(TREE_tax_sum_phylum)
TREE_melt<- melt(TREE_tax_sum_phylum, id.var="Phylum")
nrow(TREE_melt)
site<-rep("TREE",times=nrow(TREE_melt))
TREE_melt<-cbind(TREE_melt,site)

UKFS_tax_sum_phylum = summarize_taxonomy(UKFS, level = 3, report_higher_tax = FALSE)
UKFS_tax_sum_phylum$Phylum <-row.names(UKFS_tax_sum_phylum)
UKFS_melt<- melt(UKFS_tax_sum_phylum, id.var="Phylum")
nrow(UKFS_melt)
site<-rep("UKFS",times=nrow(UKFS_melt))
UKFS_melt<-cbind(UKFS_melt,site)

UNDE_tax_sum_phylum = summarize_taxonomy(UNDE, level = 3, report_higher_tax = FALSE)
UNDE_tax_sum_phylum$Phylum <-row.names(UNDE_tax_sum_phylum)
UNDE_melt<- melt(UNDE_tax_sum_phylum, id.var="Phylum")
nrow(UNDE_melt)
site<-rep("UNDE",times=nrow(UNDE_melt))
UNDE_melt<-cbind(UNDE_melt,site)

WOOD_tax_sum_phylum = summarize_taxonomy(WOOD, level = 3, report_higher_tax = FALSE)
WOOD_tax_sum_phylum$Phylum <-row.names(WOOD_tax_sum_phylum)
WOOD_melt<- melt(WOOD_tax_sum_phylum, id.var="Phylum")
nrow(WOOD_melt)
site<-rep("WOOD",times=nrow(WOOD_melt))
WOOD_melt<-cbind(WOOD_melt,site)

WREF_tax_sum_phylum = summarize_taxonomy(WREF, level = 3, report_higher_tax = FALSE)
WREF_tax_sum_phylum$Phylum <-row.names(WREF_tax_sum_phylum)
WREF_melt<- melt(WREF_tax_sum_phylum, id.var="Phylum")
nrow(WREF_melt)
site<-rep("WREF",times=nrow(WREF_melt))
WREF_melt<-cbind(WREF_melt,site)

YELL_tax_sum_phylum = summarize_taxonomy(YELL, level = 3, report_higher_tax = FALSE)
YELL_tax_sum_phylum$Phylum <-row.names(YELL_tax_sum_phylum)
YELL_melt<- melt(YELL_tax_sum_phylum, id.var="Phylum")
nrow(YELL_melt)
site<-rep("YELL",times=nrow(YELL_melt))
YELL_melt<-cbind(YELL_melt,site)
```

```{r}
phylumfinalbind<-do.call("rbind",list(ABBY_melt, BARR_melt, BART_melt, BLAN_melt, BONA_melt, CLBJ_melt, CPER_melt, DCFS_melt, DEJU_melt, DELA_melt, DSNY_melt, GRSM_melt, GUAN_melt, HARV_melt, HEAL_melt, JERC_melt, JORN_melt, KONA_melt, KONZ_melt, LAJA_melt, LENO_melt, MLBS_melt, MOAB_melt, NIWO_melt, OAES_melt, ONAQ_melt, ORNL_melt, OSBS_melt, PUUM_melt, RMNP_melt, SCBI_melt, SERC_melt, SJER_melt, SOAP_melt, SRER_melt, STEI_melt, STER_melt, TALL_melt, TOOL_melt, TREE_melt, UKFS_melt, UNDE_melt, WOOD_melt, WREF_melt, YELL_melt))
```

# Latitude first by phylum
phylum graph
```{r}
positions <- c("GUAN","LAJA","PUUM","DSNY","OSBS","JERC","LENO","SRER","DELA","JORN","TALL","CLBJ","OAES","GRSM","ORNL","SOAP","SJER","MLBS","MOAB","SERC","SCBI","BLAN","UKFS","KONZ","KONA","NIWO","ONAQ","RMNP","STER","CPER","HARV","BART","YELL","TREE","STEI","ABBY","WREF","UNDE","WOOD","DCFS","HEAL","DEJU","BONA","TOOL","BARR")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_x_discrete(limits = positions)+
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Phylum Graphs by Biome")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "phylum_site_by latitude.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

# MAP first by phylum
phylum graph
```{r}
positions <- c("BARR", "BONA", "JORN","ONAQ","DEJU","TOOL","MOAB","CPER","SRER","HEAL","STER","DCFS","YELL","WOOD","SJER","RMNP","OAES","STEI","TREE","UNDE","LAJA","GUAN","KONA","KONZ","SOAP","CLBJ","BLAN","UKFS","NIWO","SERC","SCBI","HARV","DSNY","MLBS","OSBS","JERC","BART","ORNL","DELA","GRSM","TALL","LENO","WREF","ABBY","PUUM")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_x_discrete(limits = positions)+
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Phylum Graphs by Biome")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "phylum_site_by_MAP.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```


# MAT first by phylum
phylum graph
```{r}
positions <- c("HEAL","BARR","BONA","DEJU","TOOL","NIWO","ABBY","MOAB","SCBI","BLAN","KONZ","KONA","UKFS","PUUM","GRSM","SOAP","SERC","ORNL","OAES","JORN","SJER","TALL","CLBJ","DELA","LENO","JERC","SRER","RMNP","OSBS","DSNY","GUAN","LAJA","YELL","UNDE","STEI","TREE","DCFS","WOOD","BART","HARV","CPER","MLBS","ONAQ","WREF","STER")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_x_discrete(limits = positions)+
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Phylum Graphs by Biome")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "phylum_site_by_MAT.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

```{r}
richfinalbind<-do.call("rbind",list(WREFrichdffinal,ABBYrichdffinal,TREErichdffinal,STEIrichdffinal,KONZrichdffinal,KONArichdffinal,CPERrichdffinal,STERrichdffinal,WOODrichdffinal,DCFSrichdffinal,GUANrichdffinal,LAJArichdffinal))
```

```{r}
positions <- c("CPER","STER","KONZ","KONA","WOOD","DCFS","GUAN","LAJA","WREF","ABBY","TREE","STEI")
library(ggthemes)
c<-ggplot(phylumfinalbind, aes(x = site, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_x_discrete(limits = positions)+
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(43)) + 
  ggtitle("Phylum Graphs by Biome")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}

ggsave(filename = "phylum_site_by_management.pdf",
      plot = c,
      bg = "transparent",
      width = 8, height = 6, units = "in",
      dpi = 600)
```

***END OF FIX HERE***


```{r}
library(mctoolsr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')
CLBJ = filter_data(input, 'Site', keep_vals = 'CLBJ')
CLBJ_burned = filter_data(CLBJ, 'plotID', filter_vals = 'CLBJ_033')
CLBJ_burned = filter_data(CLBJ_burned, 'plotID', filter_vals = 'CLBJ_040')
CLBJ_burned = filter_data(CLBJ_burned, 'plotID', filter_vals = 'CLBJ_032')

```

```{r}
library(vegan)
CLBJ_specrich<-specnumber(t(CLBJ_burned$data_loaded))
CLBJ_specrich<-as.data.frame(CLBJ_specrich)
colnames(CLBJ_specrich)[1] ="rich"
CLBJ_specrich<-cbind(CLBJ_specrich,CLBJ_burned$map_loaded$Year)
colnames(CLBJ_specrich)[2] ="year"
CLBJ_specrich<-cbind(CLBJ_specrich,CLBJ_burned$map_loaded$plotID)
colnames(CLBJ_specrich)[3] ="plot"
```

```{r}
CLBJ_specrich_stat<-CLBJ_specrich
CLBJ_specrich_stat$year<-as.factor(CLBJ_specrich$year)
model<-lm(data=CLBJ_specrich_stat,rich~year)
model
anova(model)
emmeans(model,pairwise~year)
```

```{r}
library(ggplot2)
CLBJ_specrich$year<-as.character(CLBJ_specrich$year)
speciesrichness<-ggplot(data=CLBJ_specrich,aes(x=year,y=rich,fill=factor(year)))+
 geom_boxplot()+
  geom_point(size=0.3)+
  #coord_flip()+
  ggtitle("CLBJ by year -> fire between 2017 + 2018")+
  xlab("Year")+
  ylab("Richness (# of species)")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
speciesrichness
```

```{r}
ggsave(filename = "speciesrichness_CLBJ_fire.pdf",
      plot = speciesrichness,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600) 
```

```{r}
library(ggplot2)
CLBJ_specrich$year<-as.numeric(CLBJ_specrich$year)
speciesrichness_ind<-ggplot(data=CLBJ_specrich,aes(x=year,y=rich,color=plot))+
 
  geom_point(size=3)+
  geom_line()+
  #coord_flip()+
  ggtitle("CLBJ by year -> fire between 2017 + 2018")+
  xlab("Year")+
  ylab("Richness (# of species)")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    
    plot.title=element_text(size=20,face="bold",vjust=2))
speciesrichness_ind
```

```{r}
ggsave(filename = "speciesrichness_CLBJ_fire_withlines.pdf",
      plot = speciesrichness_ind,
      bg = "transparent",
      width = 6, height = 6, units = "in",
      dpi = 600) 
```



```{r}
library(ggplot2)
#CLBJ_specrich$year<-as.character(CLBJ_specrich$year)
speciesrichness_overlay<-ggplot(data=CLBJ_specrich,aes(x=year,y=rich,group=year))+
  geom_boxplot()+
  geom_point(data=CLBJ_specrich,aes(x=year,y=rich,color=plot),size=3)+
  geom_line(aes(color=plot,group=plot),data=CLBJ_specrich,size=2,alpha=0.2)+
  #coord_flip()+
  ggtitle("CLBJ by year -> fire between 2017 + 2018")+
  xlab("Year")+
  ylab("Richness (# of species)")+
  theme(
    axis.text = element_text(color='black',size=10),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=8),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    legend.position = "none",
    plot.title=element_text(size=20,face="bold",vjust=2))
speciesrichness_overlay
```


```{r}
ggsave(filename = "CLBJ_speciesrichness_overlay.pdf",
      plot = speciesrichness_overlay,
      bg = "transparent",
      width = 10, height = 6, units = "in",
      dpi = 600) 
```

```{r}
tax_sum_phyla = summarize_taxonomy(CLBJ_burned, level = 3, report_higher_tax = FALSE)
tax_sum_phyla$Phylum <-row.names(tax_sum_phyla)
tax_sum_phyla_melt<- melt(tax_sum_phyla, id.var="Phylum")
library(dplyr)
CLBJ_burned$map_loaded$Well<-row.names(CLBJ_burned$map_loaded)
colnames(tax_sum_phyla_melt)[2] ="Well"
CLBJ_burned_phylum<-merge(tax_sum_phyla_melt,CLBJ_burned$map_loaded,by="Well")
```

```{r}

library(ggthemes)
c<-ggplot(CLBJ_burned_phylum, aes(x = Year, y = value, fill = Phylum)) + 
  geom_bar(stat="identity",position = "fill") + 
  scale_y_continuous(labels = scales::percent) +
  scale_fill_manual(values = colorRampPalette(solarized_pal()(8))(25)) + 
  ggtitle("CLBJ kingdoms by year -> burned between 2017 +2018")+
  coord_flip()+

  theme(
    axis.text = element_text(color='black',size=10),
    axis.title.y= element_blank(),
    axis.title.x= element_blank(),
    axis.title = element_text(color='black',size=12),
    axis.ticks = element_line(color='black'),
    legend.title = element_text(size=10),
    legend.key.size = unit(0.25, 'cm'), 
    legend.key.height = unit(0.25, 'cm'), 
    legend.key.width = unit(0.25, 'cm'),
    legend.text = element_text(size=6),
    panel.background = element_rect(fill=NA,color='black'),
    panel.grid = element_blank(),
    plot.title=element_text(size=20,face="bold",vjust=2))

c
```

```{r}
ggsave(filename = "kingdom_CLBJ_fire.pdf",
      plot = c,
      bg = "transparent",
      width = 10, height = 6, units = "in",
      dpi = 600) 
```

Correlation plots - all
```{r}
library(mctoolsr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')


```

```{r}
library(vegan)
correlations<-specnumber(t(input$data_loaded))
correlations<-as.data.frame(correlations)
colnames(correlations)[1] ="rich"
correlations<-cbind(correlations,input$map_loaded$elevation)
colnames(correlations)[2] ="elevation"
correlations<-cbind(correlations,input$map_loaded$latitude)
colnames(correlations)[3] ="latitude"
correlations<-cbind(correlations,input$map_loaded$soilTemp)
colnames(correlations)[4] ="soilTemp"
correlations<-cbind(correlations,input$map_loaded$soilMoisture)
colnames(correlations)[5] ="soilMoisture"
correlations<-cbind(correlations,input$map_loaded$soilInWaterpH)
colnames(correlations)[6] ="soilInWaterpH"
correlations<-cbind(correlations,input$map_loaded$soilInCaClpH)
colnames(correlations)[7] ="soilInCaClpH"
correlations<-cbind(correlations,input$map_loaded$d15N)
colnames(correlations)[8] ="d15N"
correlations<-cbind(correlations,input$map_loaded$organicd13C)
colnames(correlations)[9] ="organicd13C"
correlations<-cbind(correlations,input$map_loaded$nitrogenPercent)
colnames(correlations)[10] ="nitrogenPercent"
correlations<-cbind(correlations,input$map_loaded$organicCPercent)
colnames(correlations)[11] ="organicCPercent"
correlations<-cbind(correlations,input$map_loaded$kclAmmoniumNConc)
colnames(correlations)[12] ="kclAmmoniumNConc"
correlations<-cbind(correlations,input$map_loaded$kclNitrateNitriteNConc)
colnames(correlations)[13] ="kclNitrateNitriteNConc"
```

```{r}
library(Hmisc)
res <- rcorr(as.matrix(correlations)) # rcorr() accepts matrices only
round(res$r, 3)
```

```{r}
# display p-values (rounded to 3 decimals)
round(res$P, 3)
```

Fungi only
```{r}
library(mctoolsr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')

input <- filter_taxa_from_input(input, taxa_to_keep = 'Fungi')
```

```{r}
library(vegan)
correlations<-specnumber(t(input$data_loaded))
correlations<-as.data.frame(correlations)
colnames(correlations)[1] ="rich"
correlations<-cbind(correlations,input$map_loaded$elevation)
colnames(correlations)[2] ="elevation"
correlations<-cbind(correlations,input$map_loaded$latitude)
colnames(correlations)[3] ="latitude"
correlations<-cbind(correlations,input$map_loaded$soilTemp)
colnames(correlations)[4] ="soilTemp"
correlations<-cbind(correlations,input$map_loaded$soilMoisture)
colnames(correlations)[5] ="soilMoisture"
correlations<-cbind(correlations,input$map_loaded$soilInWaterpH)
colnames(correlations)[6] ="soilInWaterpH"
correlations<-cbind(correlations,input$map_loaded$soilInCaClpH)
colnames(correlations)[7] ="soilInCaClpH"
correlations<-cbind(correlations,input$map_loaded$d15N)
colnames(correlations)[8] ="d15N"
correlations<-cbind(correlations,input$map_loaded$organicd13C)
colnames(correlations)[9] ="organicd13C"
correlations<-cbind(correlations,input$map_loaded$nitrogenPercent)
colnames(correlations)[10] ="nitrogenPercent"
correlations<-cbind(correlations,input$map_loaded$organicCPercent)
colnames(correlations)[11] ="organicCPercent"
correlations<-cbind(correlations,input$map_loaded$kclAmmoniumNConc)
colnames(correlations)[12] ="kclAmmoniumNConc"
correlations<-cbind(correlations,input$map_loaded$kclNitrateNitriteNConc)
colnames(correlations)[13] ="kclNitrateNitriteNConc"
```

```{r}
library(Hmisc)
res <- rcorr(as.matrix(correlations)) # rcorr() accepts matrices only
round(res$r, 3)
```

```{r}
# display p-values (rounded to 3 decimals)
round(res$P, 3)
```


Metazoa only
```{r}
library(mctoolsr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')

input <- filter_taxa_from_input(input, taxa_to_keep = 'Metazoa')
```

```{r}
library(vegan)
correlations<-specnumber(t(input$data_loaded))
correlations<-as.data.frame(correlations)
colnames(correlations)[1] ="rich"
correlations<-cbind(correlations,input$map_loaded$elevation)
colnames(correlations)[2] ="elevation"
correlations<-cbind(correlations,input$map_loaded$latitude)
colnames(correlations)[3] ="latitude"
correlations<-cbind(correlations,input$map_loaded$soilTemp)
colnames(correlations)[4] ="soilTemp"
correlations<-cbind(correlations,input$map_loaded$soilMoisture)
colnames(correlations)[5] ="soilMoisture"
correlations<-cbind(correlations,input$map_loaded$soilInWaterpH)
colnames(correlations)[6] ="soilInWaterpH"
correlations<-cbind(correlations,input$map_loaded$soilInCaClpH)
colnames(correlations)[7] ="soilInCaClpH"
correlations<-cbind(correlations,input$map_loaded$d15N)
colnames(correlations)[8] ="d15N"
correlations<-cbind(correlations,input$map_loaded$organicd13C)
colnames(correlations)[9] ="organicd13C"
correlations<-cbind(correlations,input$map_loaded$nitrogenPercent)
colnames(correlations)[10] ="nitrogenPercent"
correlations<-cbind(correlations,input$map_loaded$organicCPercent)
colnames(correlations)[11] ="organicCPercent"
correlations<-cbind(correlations,input$map_loaded$kclAmmoniumNConc)
colnames(correlations)[12] ="kclAmmoniumNConc"
correlations<-cbind(correlations,input$map_loaded$kclNitrateNitriteNConc)
colnames(correlations)[13] ="kclNitrateNitriteNConc"
```

```{r}
library(Hmisc)
res <- rcorr(as.matrix(correlations)) # rcorr() accepts matrices only
round(res$r, 3)
```

```{r}
# display p-values (rounded to 3 decimals)
round(res$P, 3)
```

Nematoda only
```{r}
library(mctoolsr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')

input <- filter_taxa_from_input(input, taxa_to_keep = 'Nematoda')
```

```{r}
library(vegan)
correlations<-specnumber(t(input$data_loaded))
correlations<-as.data.frame(correlations)
colnames(correlations)[1] ="rich"
correlations<-cbind(correlations,input$map_loaded$elevation)
colnames(correlations)[2] ="elevation"
correlations<-cbind(correlations,input$map_loaded$latitude)
colnames(correlations)[3] ="latitude"
correlations<-cbind(correlations,input$map_loaded$soilTemp)
colnames(correlations)[4] ="soilTemp"
correlations<-cbind(correlations,input$map_loaded$soilMoisture)
colnames(correlations)[5] ="soilMoisture"
correlations<-cbind(correlations,input$map_loaded$soilInWaterpH)
colnames(correlations)[6] ="soilInWaterpH"
correlations<-cbind(correlations,input$map_loaded$soilInCaClpH)
colnames(correlations)[7] ="soilInCaClpH"
correlations<-cbind(correlations,input$map_loaded$d15N)
colnames(correlations)[8] ="d15N"
correlations<-cbind(correlations,input$map_loaded$organicd13C)
colnames(correlations)[9] ="organicd13C"
correlations<-cbind(correlations,input$map_loaded$nitrogenPercent)
colnames(correlations)[10] ="nitrogenPercent"
correlations<-cbind(correlations,input$map_loaded$organicCPercent)
colnames(correlations)[11] ="organicCPercent"
correlations<-cbind(correlations,input$map_loaded$kclAmmoniumNConc)
colnames(correlations)[12] ="kclAmmoniumNConc"
correlations<-cbind(correlations,input$map_loaded$kclNitrateNitriteNConc)
colnames(correlations)[13] ="kclNitrateNitriteNConc"
```

```{r}
library(Hmisc)
res <- rcorr(as.matrix(correlations)) # rcorr() accepts matrices only
round(res$r, 3)
```

```{r}
# display p-values (rounded to 3 decimals)
round(res$P, 3)
```


Streptophyta only
```{r}
library(mctoolsr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')

input <- filter_taxa_from_input(input, taxa_to_keep = 'Streptophyta')
```

```{r}
library(vegan)
correlations<-specnumber(t(input$data_loaded))
correlations<-as.data.frame(correlations)
colnames(correlations)[1] ="rich"
correlations<-cbind(correlations,input$map_loaded$elevation)
colnames(correlations)[2] ="elevation"
correlations<-cbind(correlations,input$map_loaded$latitude)
colnames(correlations)[3] ="latitude"
correlations<-cbind(correlations,input$map_loaded$soilTemp)
colnames(correlations)[4] ="soilTemp"
correlations<-cbind(correlations,input$map_loaded$soilMoisture)
colnames(correlations)[5] ="soilMoisture"
correlations<-cbind(correlations,input$map_loaded$soilInWaterpH)
colnames(correlations)[6] ="soilInWaterpH"
correlations<-cbind(correlations,input$map_loaded$soilInCaClpH)
colnames(correlations)[7] ="soilInCaClpH"
correlations<-cbind(correlations,input$map_loaded$d15N)
colnames(correlations)[8] ="d15N"
correlations<-cbind(correlations,input$map_loaded$organicd13C)
colnames(correlations)[9] ="organicd13C"
correlations<-cbind(correlations,input$map_loaded$nitrogenPercent)
colnames(correlations)[10] ="nitrogenPercent"
correlations<-cbind(correlations,input$map_loaded$organicCPercent)
colnames(correlations)[11] ="organicCPercent"
correlations<-cbind(correlations,input$map_loaded$kclAmmoniumNConc)
colnames(correlations)[12] ="kclAmmoniumNConc"
correlations<-cbind(correlations,input$map_loaded$kclNitrateNitriteNConc)
colnames(correlations)[13] ="kclNitrateNitriteNConc"
```

```{r}
library(Hmisc)
res <- rcorr(as.matrix(correlations)) # rcorr() accepts matrices only
round(res$r, 3)
```

```{r}
# display p-values (rounded to 3 decimals)
round(res$P, 3)
```


Arthropoda only
```{r}
library(mctoolsr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')

input <- filter_taxa_from_input(input, taxa_to_keep = 'Arthropoda')
```

```{r}
library(vegan)
correlations<-specnumber(t(input$data_loaded))
correlations<-as.data.frame(correlations)
colnames(correlations)[1] ="rich"
correlations<-cbind(correlations,input$map_loaded$elevation)
colnames(correlations)[2] ="elevation"
correlations<-cbind(correlations,input$map_loaded$latitude)
colnames(correlations)[3] ="latitude"
correlations<-cbind(correlations,input$map_loaded$soilTemp)
colnames(correlations)[4] ="soilTemp"
correlations<-cbind(correlations,input$map_loaded$soilMoisture)
colnames(correlations)[5] ="soilMoisture"
correlations<-cbind(correlations,input$map_loaded$soilInWaterpH)
colnames(correlations)[6] ="soilInWaterpH"
correlations<-cbind(correlations,input$map_loaded$soilInCaClpH)
colnames(correlations)[7] ="soilInCaClpH"
correlations<-cbind(correlations,input$map_loaded$d15N)
colnames(correlations)[8] ="d15N"
correlations<-cbind(correlations,input$map_loaded$organicd13C)
colnames(correlations)[9] ="organicd13C"
correlations<-cbind(correlations,input$map_loaded$nitrogenPercent)
colnames(correlations)[10] ="nitrogenPercent"
correlations<-cbind(correlations,input$map_loaded$organicCPercent)
colnames(correlations)[11] ="organicCPercent"
correlations<-cbind(correlations,input$map_loaded$kclAmmoniumNConc)
colnames(correlations)[12] ="kclAmmoniumNConc"
correlations<-cbind(correlations,input$map_loaded$kclNitrateNitriteNConc)
colnames(correlations)[13] ="kclNitrateNitriteNConc"
```

```{r}
library(Hmisc)
res <- rcorr(as.matrix(correlations)) # rcorr() accepts matrices only
round(res$r, 3)
```

```{r}
# display p-values (rounded to 3 decimals)
round(res$P, 3)
```

Annelida only
```{r}
library(mctoolsr)
setwd("/Users/leenavilonen/Documents/NEON")
input <- load_taxa_table("otu_table_withtax.txt", "eukDiv_mta.txt")

input <- filter_taxa_from_input(input, taxa_to_remove = 'Cercozoa:chro')
input <- filter_taxa_from_input(input, taxa_to_remove = 'Eukaryota:mito')

input <- filter_taxa_from_input(input, taxa_to_keep = 'Annelida')
```

```{r}
library(vegan)
correlations<-specnumber(t(input$data_loaded))
correlations<-as.data.frame(correlations)
colnames(correlations)[1] ="rich"
correlations<-cbind(correlations,input$map_loaded$elevation)
colnames(correlations)[2] ="elevation"
correlations<-cbind(correlations,input$map_loaded$latitude)
colnames(correlations)[3] ="latitude"
correlations<-cbind(correlations,input$map_loaded$soilTemp)
colnames(correlations)[4] ="soilTemp"
correlations<-cbind(correlations,input$map_loaded$soilMoisture)
colnames(correlations)[5] ="soilMoisture"
correlations<-cbind(correlations,input$map_loaded$soilInWaterpH)
colnames(correlations)[6] ="soilInWaterpH"
correlations<-cbind(correlations,input$map_loaded$soilInCaClpH)
colnames(correlations)[7] ="soilInCaClpH"
correlations<-cbind(correlations,input$map_loaded$d15N)
colnames(correlations)[8] ="d15N"
correlations<-cbind(correlations,input$map_loaded$organicd13C)
colnames(correlations)[9] ="organicd13C"
correlations<-cbind(correlations,input$map_loaded$nitrogenPercent)
colnames(correlations)[10] ="nitrogenPercent"
correlations<-cbind(correlations,input$map_loaded$organicCPercent)
colnames(correlations)[11] ="organicCPercent"
correlations<-cbind(correlations,input$map_loaded$kclAmmoniumNConc)
colnames(correlations)[12] ="kclAmmoniumNConc"
correlations<-cbind(correlations,input$map_loaded$kclNitrateNitriteNConc)
colnames(correlations)[13] ="kclNitrateNitriteNConc"
```

```{r}
library(Hmisc)
res <- rcorr(as.matrix(correlations)) # rcorr() accepts matrices only
round(res$r, 3)
```

```{r}
# display p-values (rounded to 3 decimals)
round(res$P, 3)
```









